<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.22">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Interactive Simulations with JSPlotly</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../../nivel/basico/jsplotly/jsplotly_bas_en.html" rel="next">
<link href="../../../nivel/superior/jsplotly/jsplotly_geral_en.html" rel="prev">
<link href="../../../avatar.png" rel="icon" type="image/png">
<script src="../../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-7b4406b7675125bc2ba204020e191172.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-4b957b30fffc5edb70387ff4d75203a6.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-S1L73VGCG5"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-S1L73VGCG5', { 'anonymize_ip': true});
</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../logo-unifal.png" alt="" class="navbar-logo">
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../index.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../erre_en.html"> 
<span class="menu-text">Reproducible Teaching</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../producao_en.html"> 
<span class="menu-text">Works</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../hp50_en.html"> 
<span class="menu-text">hp50</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../contato_en.html"> 
<span class="menu-text">Contact</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../../pt/index.html"> 
<span class="menu-text">PT</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools tools-wide">
    <a href="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-true"></i></a>
    <a href="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-true"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../nivel/superior/jsplotly/jsplotly_geral_en.html">JSPlotly: Interactive objects &amp; AI</a></li><li class="breadcrumb-item"><a href="../../../nivel/superior/jsplotly/jsptl_bioq_en.html">Biophysical chemistry</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">JSPlotly: Interactive objects &amp; AI</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jsplotly/jsplotly_geral_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">JSPlotly</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jsplotly/jsptl_bioq_en.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Biophysical chemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/jsplotly/jsplotly_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">School</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Jmol - 3D molecules</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jmol/index_jmol_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Biochemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/Curso/index_jmol_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">School</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/Jardim/index_Jardim_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Molecule Garden</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">R - Maths and charts</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/r/_book/index_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Biophysical chemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/ScriptRplot/scriptrplot_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">scriptRplot: elegant graphics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/Curso/index_r_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">School</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/R/index_OIER_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Interactivity at School</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Sisma - dynamic flows</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/sisma/index_sisma_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Biochemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/sisma/index_sisma_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">School</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#simulations" id="toc-simulations" class="nav-link active" data-scroll-target="#simulations"><span class="header-section-number">1</span> Simulations</a>
  <ul class="collapse">
  <li><a href="#acidbase-equilibrium-and-buffer-system" id="toc-acidbase-equilibrium-and-buffer-system" class="nav-link" data-scroll-target="#acidbase-equilibrium-and-buffer-system"><span class="header-section-number">1.1</span> Acid–base equilibrium and buffer system</a></li>
  <li><a href="#ionization-state-of-chemical-groups-of-biomolecules" id="toc-ionization-state-of-chemical-groups-of-biomolecules" class="nav-link" data-scroll-target="#ionization-state-of-chemical-groups-of-biomolecules"><span class="header-section-number">1.2</span> Ionization state of chemical groups of biomolecules</a></li>
  <li><a href="#net-charge-network-in-peptides" id="toc-net-charge-network-in-peptides" class="nav-link" data-scroll-target="#net-charge-network-in-peptides"><span class="header-section-number">1.3</span> Net charge network in peptides</a></li>
  <li><a href="#suggestion-2" id="toc-suggestion-2" class="nav-link" data-scroll-target="#suggestion-2">Suggestion:</a></li>
  <li><a href="#charge-ph-pka-pi-of-amino-acids-animation" id="toc-charge-ph-pka-pi-of-amino-acids-animation" class="nav-link" data-scroll-target="#charge-ph-pka-pi-of-amino-acids-animation"><span class="header-section-number">1.4</span> Charge, pH, pKa &amp; pI of amino acids (animation)</a></li>
  <li><a href="#isoelectric-point-in-proteins" id="toc-isoelectric-point-in-proteins" class="nav-link" data-scroll-target="#isoelectric-point-in-proteins"><span class="header-section-number">1.5</span> Isoelectric Point in Proteins</a></li>
  <li><a href="#hydrophobicity-in-amino-acid-sequences" id="toc-hydrophobicity-in-amino-acid-sequences" class="nav-link" data-scroll-target="#hydrophobicity-in-amino-acid-sequences"><span class="header-section-number">1.6</span> Hydrophobicity in amino acid sequences</a></li>
  <li><a href="#interaction-of-oxygen-with-myoglobin-and-hemoglobin" id="toc-interaction-of-oxygen-with-myoglobin-and-hemoglobin" class="nav-link" data-scroll-target="#interaction-of-oxygen-with-myoglobin-and-hemoglobin"><span class="header-section-number">1.7</span> Interaction of oxygen with myoglobin and hemoglobin</a></li>
  <li><a href="#bohr-effect-in-hemoglobin-ph" id="toc-bohr-effect-in-hemoglobin-ph" class="nav-link" data-scroll-target="#bohr-effect-in-hemoglobin-ph"><span class="header-section-number">1.8</span> Bohr effect in hemoglobin (pH)</a></li>
  <li><a href="#catalysis-and-enzymatic-inhibition" id="toc-catalysis-and-enzymatic-inhibition" class="nav-link" data-scroll-target="#catalysis-and-enzymatic-inhibition"><span class="header-section-number">1.9</span> Catalysis and enzymatic inhibition</a></li>
  <li><a href="#linearization-of-kinetic-data-by-lineweaverburk" id="toc-linearization-of-kinetic-data-by-lineweaverburk" class="nav-link" data-scroll-target="#linearization-of-kinetic-data-by-lineweaverburk"><span class="header-section-number">1.10</span> Linearization of kinetic data by Lineweaver–Burk</a></li>
  <li><a href="#diagnosis-of-enzymatic-inhibition-michaelismenten-and-lineweaverburk" id="toc-diagnosis-of-enzymatic-inhibition-michaelismenten-and-lineweaverburk" class="nav-link" data-scroll-target="#diagnosis-of-enzymatic-inhibition-michaelismenten-and-lineweaverburk"><span class="header-section-number">1.11</span> Diagnosis of Enzymatic Inhibition (Michaelis–Menten and Lineweaver–Burk)</a></li>
  <li><a href="#van-der-waals-equation-for-ideal-gases" id="toc-van-der-waals-equation-for-ideal-gases" class="nav-link" data-scroll-target="#van-der-waals-equation-for-ideal-gases"><span class="header-section-number">1.12</span> van der Waals equation for ideal gases</a></li>
  <li><a href="#equilibrium-of-atp-production-from-reagents-temperature-and-ph" id="toc-equilibrium-of-atp-production-from-reagents-temperature-and-ph" class="nav-link" data-scroll-target="#equilibrium-of-atp-production-from-reagents-temperature-and-ph"><span class="header-section-number">1.13</span> Equilibrium of ATP production from reagents, temperature, and pH</a></li>
  <li><a href="#variation-of-gibbs-energy-with-temperature" id="toc-variation-of-gibbs-energy-with-temperature" class="nav-link" data-scroll-target="#variation-of-gibbs-energy-with-temperature"><span class="header-section-number">1.14</span> Variation of Gibbs energy with temperature</a></li>
  <li><a href="#oxygen-consumption-by-mitochondria" id="toc-oxygen-consumption-by-mitochondria" class="nav-link" data-scroll-target="#oxygen-consumption-by-mitochondria"><span class="header-section-number">1.15</span> Oxygen consumption by mitochondria</a></li>
  <li><a href="#electrochemical-impedance-spectroscopy-eis" id="toc-electrochemical-impedance-spectroscopy-eis" class="nav-link" data-scroll-target="#electrochemical-impedance-spectroscopy-eis"><span class="header-section-number">1.16</span> Electrochemical impedance spectroscopy (EIS)</a></li>
  <li><a href="#cyclic-voltammetry" id="toc-cyclic-voltammetry" class="nav-link" data-scroll-target="#cyclic-voltammetry"><span class="header-section-number">1.17</span> Cyclic voltammetry</a></li>
  <li><a href="#diagrams-and-flowcharts" id="toc-diagrams-and-flowcharts" class="nav-link" data-scroll-target="#diagrams-and-flowcharts"><span class="header-section-number">1.18</span> Diagrams and flowcharts</a></li>
  </ul></li>
  <li><a href="#construction-of-plots" id="toc-construction-of-plots" class="nav-link" data-scroll-target="#construction-of-plots"><span class="header-section-number">2</span> Construction of plots</a>
  <ul class="collapse">
  <li><a href="#plot-with-2-y-axes" id="toc-plot-with-2-y-axes" class="nav-link" data-scroll-target="#plot-with-2-y-axes"><span class="header-section-number">2.1</span> Plot with 2 y-axes</a></li>
  <li><a href="#rangeslider" id="toc-rangeslider" class="nav-link" data-scroll-target="#rangeslider"><span class="header-section-number">2.2</span> RangeSlider</a></li>
  <li><a href="#subplots" id="toc-subplots" class="nav-link" data-scroll-target="#subplots"><span class="header-section-number">2.3</span> Subplots</a></li>
  <li><a href="#subplots-with-different-types" id="toc-subplots-with-different-types" class="nav-link" data-scroll-target="#subplots-with-different-types"><span class="header-section-number">2.4</span> Subplots with different types</a></li>
  <li><a href="#synchronized-multiple-selection" id="toc-synchronized-multiple-selection" class="nav-link" data-scroll-target="#synchronized-multiple-selection"><span class="header-section-number">2.5</span> Synchronized multiple selection</a></li>
  <li><a href="#click-event-message" id="toc-click-event-message" class="nav-link" data-scroll-target="#click-event-message"><span class="header-section-number">2.6</span> Click event &amp; message</a></li>
  <li><a href="#menu-for-selecting-types" id="toc-menu-for-selecting-types" class="nav-link" data-scroll-target="#menu-for-selecting-types"><span class="header-section-number">2.7</span> Menu for selecting types</a></li>
  <li><a href="#slider-control" id="toc-slider-control" class="nav-link" data-scroll-target="#slider-control"><span class="header-section-number">2.8</span> Slider control</a></li>
  </ul></li>
  <li><a href="#data-analysis" id="toc-data-analysis" class="nav-link" data-scroll-target="#data-analysis"><span class="header-section-number">3</span> Data analysis</a>
  <ul class="collapse">
  <li><a href="#user-data-entry" id="toc-user-data-entry" class="nav-link" data-scroll-target="#user-data-entry"><span class="header-section-number">3.1</span> User data entry</a></li>
  <li><a href="#loading-a-file-for-analysis" id="toc-loading-a-file-for-analysis" class="nav-link" data-scroll-target="#loading-a-file-for-analysis"><span class="header-section-number">3.2</span> Loading a file for analysis</a></li>
  <li><a href="#linear-fitting-of-data" id="toc-linear-fitting-of-data" class="nav-link" data-scroll-target="#linear-fitting-of-data"><span class="header-section-number">3.3</span> Linear fitting of data</a></li>
  <li><a href="#polynomial-regression" id="toc-polynomial-regression" class="nav-link" data-scroll-target="#polynomial-regression"><span class="header-section-number">3.4</span> Polynomial regression</a></li>
  <li><a href="#multiple-linear-regression" id="toc-multiple-linear-regression" class="nav-link" data-scroll-target="#multiple-linear-regression"><span class="header-section-number">3.5</span> Multiple linear regression</a></li>
  <li><a href="#multiple-linear-fitting-with-3-or-more-predictor-variables" id="toc-multiple-linear-fitting-with-3-or-more-predictor-variables" class="nav-link" data-scroll-target="#multiple-linear-fitting-with-3-or-more-predictor-variables"><span class="header-section-number">3.6</span> Multiple linear fitting with 3 or more predictor variables</a></li>
  <li><a href="#response-surface-methodology-rsm" id="toc-response-surface-methodology-rsm" class="nav-link" data-scroll-target="#response-surface-methodology-rsm"><span class="header-section-number">3.7</span> Response Surface Methodology (RSM)</a></li>
  <li><a href="#data-smoothing---spline-and-savitzky-golay-filter" id="toc-data-smoothing---spline-and-savitzky-golay-filter" class="nav-link" data-scroll-target="#data-smoothing---spline-and-savitzky-golay-filter"><span class="header-section-number">3.8</span> Data smoothing - Spline and Savitzky-Golay filter</a></li>
  <li><a href="#anova-and-tukey-hsd-test" id="toc-anova-and-tukey-hsd-test" class="nav-link" data-scroll-target="#anova-and-tukey-hsd-test"><span class="header-section-number">3.9</span> ANOVA and Tukey-HSD test</a></li>
  <li><a href="#nonlinear-fitting-of-data" id="toc-nonlinear-fitting-of-data" class="nav-link" data-scroll-target="#nonlinear-fitting-of-data"><span class="header-section-number">3.10</span> Nonlinear fitting of data</a></li>
  </ul></li>
  <li><a href="#applications" id="toc-applications" class="nav-link" data-scroll-target="#applications"><span class="header-section-number">4</span> Applications</a>
  <ul class="collapse">
  <li><a href="#prot.id---simulated-experimentation-for-characterization-of-amino-acids-and-chains" id="toc-prot.id---simulated-experimentation-for-characterization-of-amino-acids-and-chains" class="nav-link" data-scroll-target="#prot.id---simulated-experimentation-for-characterization-of-amino-acids-and-chains"><span class="header-section-number">4.1</span> prot.Id - Simulated experimentation for characterization of amino acids and chains</a></li>
  <li><a href="#zemol---3d-viewer-for-molecular-models" id="toc-zemol---3d-viewer-for-molecular-models" class="nav-link" data-scroll-target="#zemol---3d-viewer-for-molecular-models"><span class="header-section-number">4.2</span> ZeMol - 3D viewer for molecular models</a></li>
  <li><a href="#carbo.id---simulated-experimentation-for-carbohydrate-characterization" id="toc-carbo.id---simulated-experimentation-for-carbohydrate-characterization" class="nav-link" data-scroll-target="#carbo.id---simulated-experimentation-for-carbohydrate-characterization"><span class="header-section-number">4.3</span> carbo.Id - Simulated experimentation for carbohydrate characterization</a></li>
  <li><a href="#flowforces" id="toc-flowforces" class="nav-link" data-scroll-target="#flowforces"><span class="header-section-number">4.4</span> FlowForces</a></li>
  <li><a href="#gizz---digital-chalkboard" id="toc-gizz---digital-chalkboard" class="nav-link" data-scroll-target="#gizz---digital-chalkboard"><span class="header-section-number">4.5</span> Gizz - Digital chalkboard</a></li>
  </ul></li>
  <li><a href="#toolkits" id="toc-toolkits" class="nav-link" data-scroll-target="#toolkits"><span class="header-section-number">5</span> Toolkits</a>
  <ul class="collapse">
  <li><a href="#ek-toolkit" id="toc-ek-toolkit" class="nav-link" data-scroll-target="#ek-toolkit"><span class="header-section-number">5.1</span> EK-Toolkit</a></li>
  <li><a href="#leadhunt-toolkit" id="toc-leadhunt-toolkit" class="nav-link" data-scroll-target="#leadhunt-toolkit"><span class="header-section-number">5.2</span> LeadHunt-Toolkit</a></li>
  <li><a href="#lb-toolkit" id="toc-lb-toolkit" class="nav-link" data-scroll-target="#lb-toolkit"><span class="header-section-number">5.3</span> LB-Toolkit</a></li>
  <li><a href="#gh-toolkit" id="toc-gh-toolkit" class="nav-link" data-scroll-target="#gh-toolkit"><span class="header-section-number">5.4</span> GH-Toolkit</a></li>
  <li><a href="#mtherm-toolkit" id="toc-mtherm-toolkit" class="nav-link" data-scroll-target="#mtherm-toolkit"><span class="header-section-number">5.5</span> mTherm-Toolkit</a></li>
  <li><a href="#thermstab---toolkit" id="toc-thermstab---toolkit" class="nav-link" data-scroll-target="#thermstab---toolkit"><span class="header-section-number">5.6</span> ThermStab - Toolkit</a></li>
  <li><a href="#biocel-toolkit" id="toc-biocel-toolkit" class="nav-link" data-scroll-target="#biocel-toolkit"><span class="header-section-number">5.7</span> Biocel-Toolkit</a></li>
  </ul></li>
  <li><a href="#arduino-biocell" id="toc-arduino-biocell" class="nav-link" data-scroll-target="#arduino-biocell"><span class="header-section-number">6</span> Arduino &amp; Biocell</a>
  <ul class="collapse">
  <li><a href="#serial-cable" id="toc-serial-cable" class="nav-link" data-scroll-target="#serial-cable"><span class="header-section-number">6.1</span> Serial cable</a></li>
  <li><a href="#internet-of-things---iot" id="toc-internet-of-things---iot" class="nav-link" data-scroll-target="#internet-of-things---iot"><span class="header-section-number">6.2</span> Internet of Things - IoT</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../nivel/superior/jsplotly/jsplotly_geral_en.html">JSPlotly: Interactive objects &amp; AI</a></li><li class="breadcrumb-item"><a href="../../../nivel/superior/jsplotly/jsptl_bioq_en.html">Biophysical chemistry</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Interactive Simulations with JSPlotly</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div class="line-block">&nbsp;&nbsp;To illustrate the potential use of <em>JSPlotly</em> for Biochemistry and related areas, some examples follow for plots, simulations, data analysis, and applications, whose contents are frequently found in textbooks and technical-scientific sources. For better use of each theme, try to follow the code Suggestions for each object.</div>
<p><br>
</p>
<p><strong>General instructions</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Choose a theme;</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Click on the corresponding plot;</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Click on <span class="st">"add"</span>;</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Use the mouse <span class="cf">for</span> interactivity and<span class="sc">/</span>or edit the code. </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>Reminder<span class="sc">:</span> the editor uses infinite undo<span class="sc">/</span>redo <span class="cf">in</span> the <span class="fu">code</span> (Ctrl<span class="sc">+</span>Z <span class="sc">/</span> Shift<span class="sc">+</span>Ctrl<span class="sc">+</span>Z) <span class="sc">!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="simulations" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="simulations"><span class="header-section-number">1</span> Simulations</h2>
<section id="acidbase-equilibrium-and-buffer-system" class="level3" data-number="1.1">
<h3 data-number="1.1" class="anchored" data-anchor-id="acidbase-equilibrium-and-buffer-system"><span class="header-section-number">1.1</span> Acid–base equilibrium and buffer system</h3>
<section id="context" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context">Context:</h4>
<div class="line-block">&nbsp;&nbsp;The example illustrates the addition of base in a system containing a weak acid and its conjugate base. The titration equation refers to a <em>triprotic acid</em>, phosphoric acid of the homonymous buffer, but it would equally serve for many others, such as those present in the Krebs cycle (citrate, isocitrate).</div>
</section>
<section id="equation" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation">Equation:</h4>
<p><span class="math display">\[
fa= \frac{1}{1 + 10^{pKa1 - pH}} + \frac{1}{1 + 10^{pKa2 - pH}} + \frac{1}{1 + 10^{pKa3 - pH}}
\]</span> <em>Where</em>, fa = acid fraction (protonable groups)<br>
</p>
<p><a href="Eq/jsp_acbase.html" target="_blank"><img src="Eq/tit_acbase.png" class="img-fluid"></a><br>
</p>
<!---div class="jsplotly_sup-fundo"> --->
</section>
<section id="suggestion" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="st">"A. Converting the phosphate buffer curve (triprotic) to bicarbonate buffer (diprotic)"</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Change the pKa values to those of the bicarbonate buffer<span class="sc">:</span> pKa1 <span class="ot">=</span>      <span class="fl">6.1</span>, and pKa2 <span class="ot">=</span> <span class="fl">10.3</span>;</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Set a very large value <span class="cf">for</span> <span class="fu">pKa3</span> (e.g., <span class="fl">1e20</span>). </span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Click on <span class="st">"add plot"</span>.</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>Explanation<span class="sc">:</span> pKa is a term that represents the logarithm of a dissociation <span class="fu">constant</span> (<span class="sc">-</span>log[Ka]). With an extreme value, the denominator also becomes immense, nullifying the term that carries pKa3. In JavaScript and other languages, <span class="st">"e"</span> represents the notation <span class="cf">for</span> power of <span class="fl">10.</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="st">"B. Converting the bicarbonate buffer curve to acetate"</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Just <span class="cf">repeat</span> the procedure above, with pKa1 <span class="ot">=</span> <span class="fl">4.75</span>, and eliminating pKa2.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="ionization-state-of-chemical-groups-of-biomolecules" class="level3" data-number="1.2">
<h3 data-number="1.2" class="anchored" data-anchor-id="ionization-state-of-chemical-groups-of-biomolecules"><span class="header-section-number">1.2</span> Ionization state of chemical groups of biomolecules</h3>
<section id="context-1" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-1">Context:</h4>
<div class="line-block">&nbsp;&nbsp;<em>JSPlotly</em> also allows the observation of some concepts facilitated by <em>animation-based learning</em> together with graphical representations obtained by mathematical functions. In the example that follows, the interactive object illustrates the protonation state for the main chemical groups found in biomolecules.</div>
<p><br>
</p>
<p><strong>Instruction</strong></p>
<ol type="1">
<li><p>After clicking the corresponding image below, click on “add” and observe the generated acid–base titration plot. The abscissa axis (x-axis) has an extensive range of pH values, and the ordinate axis the charge that the molecule with ionization potential manifests at each pH, and as a function of its pKa value;</p></li>
<li><p>Click on “Animate” to observe the displacement of a small ball along the titration curve, up to a pH value pre-defined in the simulator script, and as a function of its pKa;</p></li>
<li><p>Modify in the <em>script</em> the type of ionizable group (in <em>“let grupo_edit =”</em>) and/or the desired final pH (<em>“let pH_final_edit =”</em>), to evaluate the ionization state of a new group;</p></li>
<li><p>Click on “Reset” and “Animate”, to observe the new animation.</p></li>
</ol>
<p><a href="Eq/jsp_pHpKa.html" target="_blank"><img src="Eq/pHpKa.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-1" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-1">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Change the pH and the type of group that is intended to be observed;</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Simulate the ionization state of different molecules, such as drugs with carboxylic <span class="fu">acid</span> (e.g., acetylsalicylic acid... aspirin), at blood pH and <span class="cf">in</span> the gastric cavity;</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Create a new ionizable group and predict its ionization state. For this, just change the information <span class="cf">in</span> <span class="dv">3</span> constants at the beginning of the script<span class="sc">:</span> groups, pKas, and types.</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  a. </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="net-charge-network-in-peptides" class="level3" data-number="1.3">
<h3 data-number="1.3" class="anchored" data-anchor-id="net-charge-network-in-peptides"><span class="header-section-number">1.3</span> Net charge network in peptides</h3>
<section id="context-2" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-2">Context:</h4>
<div class="line-block">&nbsp;&nbsp;The code refers to the net charge network present in any sequence of amino acid residues. Here <em>angiotensin II</em> is illustrated, an important peptide for regulation of blood pressure and electrolyte balance, and whose converting enzyme is associated with the mechanism of cellular invasiveness by SARS-CoV-2.</div>
</section>
<section id="equation-1" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-1">Equation:</h4>
<div class="line-block">&nbsp;&nbsp;For this simulation there is no direct equation, since the algorithm needs to decide the charge as a function of the basic or acidic nature of the residue at a given pH value (observe the <em>script</em>). Thus:</div>
<p><span class="math display">\[
q =
\begin{cases}
-\dfrac{1}{1 + 10^{pK_a - pH}} &amp; \text{(acid group)} \\
\dfrac{1}{1 + 10^{pH - pK_a}} &amp; \text{(basic group)}
\end{cases}
\]</span><br>
</p>
<p><em>Where</em>,</p>
<ul>
<li>pKa = value of the base-10 antilogarithm for the equilibrium constant of acid dissociation, Ka (or log[Ka]).</li>
</ul>
<p><a href="Eq/jsp_cargaAA3.html" target="_blank"><img src="Eq/cargaAA3.png" class="img-fluid"></a><br>
</p>
<!---div class="jsplotly_sup-fundo"> --->
</section>
</section>
<section id="suggestion-2" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-2">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Select the peptide sequence below, and observe the charge distribution<span class="sc">:</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="st">"Ala,Lys,Arg,Leu,Phe,Glu,Cys,Asp,His"</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Simulate the stomach pH <span class="fu">condition</span> (<span class="st">"const pH = 1.5"</span>), and verify the change of charges <span class="cf">in</span> the peptide. </span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Select a physiological <span class="fu">peptide</span> (oxytocin, <span class="cf">for</span> example), observe its charge <span class="cf">in</span> <span class="fu">blood</span> (pH <span class="fl">7.5</span>), and reflect on its potential <span class="cf">for</span> electrostatic interaction with cellular components.</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="st">"Cys,Tyr,Ile,Gln,Asn,Cys,Pro,Leu,Gly"</span>  <span class="sc">-</span> oxytocin</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
<section id="charge-ph-pka-pi-of-amino-acids-animation" class="level3" data-number="1.4">
<h3 data-number="1.4" class="anchored" data-anchor-id="charge-ph-pka-pi-of-amino-acids-animation"><span class="header-section-number">1.4</span> Charge, pH, pKa &amp; pI of amino acids (animation)</h3>
<section id="context-3" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-3">Context:</h4>
<div class="line-block">&nbsp;&nbsp;This object addresses a different potential for <em>JSPlotly</em>, namely, <em>animation-based learning</em>. To illustrate this approach, an animation follows for observing the charge of an amino acid based on the value of its <em>pKa</em>, and on the information of its <em>pI</em>, the isoelectric point of the compound (pH value at which the species has zero net charge).</div>
<div class="line-block">&nbsp;&nbsp;To visualize the charge value <em>q</em> (<em>“current state”</em>), click the “add” button followed by “Animate”. For another animation, just select the amino acid (<em>“let aa_edit =”</em>) and the final pH at which you wish to observe its charge (<em>“let pH_final_edit =”</em>). The animation also allows comparing the charge of a free amino acid with that of the corresponding residue in a polypeptide chain (option <em>“let contexto_edit =”</em>). Note that the animation provides a legend informing the pKa value of the compound for its ionizable groups.</div>
<p><a href="Eq/jsp_AA_pHpKapI.html" target="_blank"><img src="Eq/AA_pHpKapI.png" class="img-fluid"></a><br>
</p>
</section>
<section id="suggestion-3" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-3">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Simulate the condition of an amino acid at blood pH and at stomach pH;</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Compare the pH values of a free amino acid, with that of its residue <span class="cf">in</span> a chain;</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Change the boolean option <span class="cf">in</span> <span class="st">"const showSites"</span>, to observe the curves of each ionizable group.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="isoelectric-point-in-proteins" class="level3" data-number="1.5">
<h3 data-number="1.5" class="anchored" data-anchor-id="isoelectric-point-in-proteins"><span class="header-section-number">1.5</span> Isoelectric Point in Proteins</h3>
<section id="context-4" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-4">Context:</h4>
<div class="line-block">&nbsp;&nbsp;The <em>script</em> for this simulation is based on the distribution of the net charge network for a polyelectrolyte, and the identification of the pH value at which this network is null, that is, the <em>isoelectric (or isoeionic) point, pI</em>. The example uses the primary sequence of lysozyme, a hydrolase that acts in breaking the microbial wall.</div>
</section>
<section id="equation-2" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-2">Equation:</h4>
<p><span class="math display">\[
q_{\text{net}}(pH) = \sum_{i=1}^{N} \left[ n_i \cdot q_{B_i} + \frac{n_i}{1 + 10^{pH - pK_{a_i}}} \right]
\]</span></p>
<p><em>Where</em>,</p>
<ul>
<li>qnet = total net charge;</li>
<li>qB$_{i} = charge of the basic form for residue i (for example, +1 for Lys, 0 for Asp);</li>
<li>n<span class="math inline">\(_{i}\)</span> = number of groups of residue <em>i</em>.</li>
</ul>
<p><a href="Eq/jsp_pI.html" target="_blank"><img src="Eq/pI.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-4" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-4">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="st">"Finding the pI for other proteins"</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> One can verify the titration of any other protein or peptide sequence by simply substituting the primary sequence contained <span class="cf">in</span> the code. An assertive way to perform this substitution involves<span class="sc">:</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  a. Search <span class="cf">for</span> the <span class="st">"FASTA"</span> sequence of the protein <span class="cf">in</span> <span class="fu">NCBI</span> (<span class="st">"https://www.ncbi.nlm.nih.gov/protein/"</span>) <span class="sc">-</span> e.g.<span class="sc">:</span> <span class="st">"papain"</span>;</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  b. Click on <span class="st">"FASTA"</span> and copy the sequence obtained <span class="cf">in</span> <span class="dv">1</span>a.;</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  c. Paste the sequence into a site <span class="cf">for</span> residue <span class="fu">quantification</span> (e.g., <span class="st">"https://www.protpi.ch/Calculator/ProteinTool"</span>);</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Substitute the sequence <span class="cf">in</span> the code. </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="hydrophobicity-in-amino-acid-sequences" class="level3" data-number="1.6">
<h3 data-number="1.6" class="anchored" data-anchor-id="hydrophobicity-in-amino-acid-sequences"><span class="header-section-number">1.6</span> Hydrophobicity in amino acid sequences</h3>
<section id="context-5" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-5">Context:</h4>
<div class="line-block">&nbsp;&nbsp;Observing hydrophobicity levels of amino acids can be useful for the study and prediction of residue sequences, such as those found in the central portion of helices of transmembrane proteins, as for identification of specific sites in enzymes, such as clefts and hydrophobic pockets.</div>
<div class="line-block">&nbsp;&nbsp;These levels are classified in a relative way, giving rise to some indices such as the <em>Kyte–Doolittle scale</em>, which assigns each amino acid a hydropathy value (hydrophobicity/hydrophobicity) estimated from experimental and observational data. Positive values indicate hydrophobic tendency (e.g., Ile, Val, Leu) and negative values indicate hydrophilic tendency (e.g., Asp, Glu, Lys, Arg). The index allows highlighting stretches prone to nonpolar environments, as well as regions exposed to the aqueous solvent.</div>
</section>
<section id="equation-3" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-3">Equation:</h4>
<p><span class="math display">\[
\tilde{y}*i ;=; \frac{1}{w} \sum*{k=i-m}^{i+m} y_k,
\qquad w = 2m+1
\]</span><br>
</p>
<p><em>Where</em>,</p>
<span class="math display">\[\begin{aligned}
y_i &amp;:; \text{hydrophobicity value at position $i$ of the sequence} \
\tilde{y}_i &amp;:; \text{smoothed value (moving average) at position $i$} \
m &amp;:; \text{half-window (number of neighbors on each side)} \
w &amp;:; \text{total window width ($w=2m+1$)} \
i &amp;:; \text{amino acid position in the sequence ($1 \leq i \leq N$)}
\end{aligned}\]</span>
<p>:; <br>
m &amp;:; <br>
w &amp;:; <br>
i &amp;:; \end{aligned}</p>
<p><a href="Eq/jsp_hidrofob.html" target="_blank"><img src="Eq/hidrofob.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-5" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-5">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try changing the sequence to a known one;</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Compare a polar sequence with a nonpolar one by <span class="fu">overlay</span> (<span class="st">"add"</span>);</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Transcribe a transmembrane helix sequence of a protein by accessing its data on the PDB site</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="interaction-of-oxygen-with-myoglobin-and-hemoglobin" class="level3" data-number="1.7">
<h3 data-number="1.7" class="anchored" data-anchor-id="interaction-of-oxygen-with-myoglobin-and-hemoglobin"><span class="header-section-number">1.7</span> Interaction of oxygen with myoglobin and hemoglobin</h3>
<section id="context-6" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-6">Context:</h4>
<div class="line-block">&nbsp;&nbsp;The oxygen molecule can combine with the <em>heme</em> group of myoglobin and hemoglobin in a distinct way, as a function of the cooperativity exhibited in the latter relative to its quaternary structure. The following model exemplifies this interaction, by use of the <em>Hill equation</em>.</div>
</section>
<section id="equation-4" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-4">Equation:</h4>
<p><span class="math display">\[
Y= \frac{pO_2^{nH}}{p_{50}^{nH} + pO_2^{nH}}
\]</span><br>
</p>
<p><em>Where</em></p>
<ul>
<li>Y = degree of oxygen saturation in the protein;</li>
<li>pO<span class="math inline">\(_{2}\)</span> = oxygen pressure;</li>
<li>p<span class="math inline">\(_{50}\)</span> = oxygen pressure at 50% saturation;</li>
<li>nH = Hill coefficient for the interaction;<br>
</li>
</ul>
<p><a href="Eq/jsp_O2.html" target="_blank"><img src="Eq/O2.png" class="img-fluid"></a><br>
</p>
<!---div class="jsplotly_sup-fundo"> --->
</section>
<section id="suggestion-6" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-6">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Run the <span class="fu">application</span> (<span class="st">"add plot"</span>). See that the value of <span class="st">"nH"</span> of the Hill constant is <span class="st">"1"</span>, that is, without cooperativity effect.</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Now replace the value of <span class="st">"nH"</span> with the Hill coefficient <span class="cf">for</span> hemoglobin, <span class="fl">2.8</span>, and run again <span class="sc">!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="bohr-effect-in-hemoglobin-ph" class="level3" data-number="1.8">
<h3 data-number="1.8" class="anchored" data-anchor-id="bohr-effect-in-hemoglobin-ph"><span class="header-section-number">1.8</span> Bohr effect in hemoglobin (pH)</h3>
<section id="context-7" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-7">Context:</h4>
<div class="line-block">&nbsp;&nbsp;Some physiological or pathological conditions can alter the binding affinity of oxygen with hemoglobin, such as temperature, some metabolites (2,3-BPG), and the hydrogen ion content of the solution.</div>
</section>
<section id="equation-5" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-5">Equation:</h4>
<p><span class="math display">\[
Y(pO_2) = \frac{{pO_2^n}}{{P_{50}^n + pO_2^n}}, \quad \text{with } P_{50} = P_{50,\text{ref}} + \alpha (pH_{\text{ref}} - pH)
\]</span></p>
<p><em>Where</em>,</p>
<ul>
<li>Y = hemoglobin saturation,</li>
<li>pO<span class="math inline">\(_{2}\)</span> = partial pressure of oxygen (in mmHg),</li>
<li>P<span class="math inline">\(*{50}\)</span> = pressure of O<span class="math inline">\(*{2}\)</span> at which hemoglobin is 50% saturated,</li>
<li>P<span class="math inline">\(_{50,ref}\)</span> = 26 mmHg (standard value),</li>
<li><span class="math inline">\(\alpha\)</span> = 50 (intensity of the Bohr effect),</li>
<li>pH<span class="math inline">\(_{ref}\)</span> = 7.4,</li>
<li>n = 2.8 = Hill coefficient for hemoglobin.</li>
</ul>
<p><a href="Eq/jspl_bohr.html"><img src="Eq/jspl_bohr.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-7" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-7">Suggestion</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try altering the reference pH <span class="cf">for</span> the interaction;</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Simulate other allosteric models by changing the value of <span class="st">"n"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="catalysis-and-enzymatic-inhibition" class="level3" data-number="1.9">
<h3 data-number="1.9" class="anchored" data-anchor-id="catalysis-and-enzymatic-inhibition"><span class="header-section-number">1.9</span> Catalysis and enzymatic inhibition</h3>
<section id="context-8" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-8">Context:</h4>
<div class="line-block">&nbsp;&nbsp;The simulation that follows aims to offer a general equation for enzymatic inhibition studies, which contemplates the <em>competitive, uncompetitive and competitive (pure or mixed)</em> models, also allowing the study of enzymatic catalysis in the absence of inhibitor.</div>
</section>
<section id="equation-6" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-6">Equation:</h4>
<p><span class="math display">\[
v=\frac{Vm*S}{Km(1+\frac{I}{Kic})+S(1+\frac{I}{Kiu})}
\]</span></p>
<p><em>Where</em></p>
<ul>
<li>S = substrate content for the reaction;</li>
<li>Vm = limiting velocity of the reaction (in books, maximum velocity);</li>
<li>Km = Michaelis–Menten constant;</li>
<li>Kic = inhibitor dissociation equilibrium constant for the competitive model;</li>
<li>Kiu = inhibitor dissociation equilibrium constant for the uncompetitive model<br>
</li>
</ul>
<p><a href="Eq/jsp_kin_inib.html" target="_blank"><img src="Eq/kin_inib.png" class="img-fluid"></a><br>
</p>
<!---div class="jsplotly_sup-fundo"> --->
</section>
<section id="suggestion-8" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-8">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="st">"A. Enzymatic catalysis in the absence of inhibitor."</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Just run the application with the general equation. See that the values <span class="cf">for</span> Kic and Kiu are <span class="fu">high</span> (<span class="fl">1e20</span>). In this way, with high <span class="st">"dissociation constants"</span>, the interaction of the inhibitor with the enzyme is irrelevant, returning the model to the classic Michaelis–Menten equation.</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try changing the values of Vm and Km, comparing plots.</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Use the geographic coordinates feature of the icon <span class="fu">bar</span> (<span class="st">"Toggle Spike Lines"</span>), to consolidate the mathematical meaning of Km, as well as observe the effect of distinct values of this on the visualization of the plot.</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="st">"B. Competitive inhibition model."</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> To observe or compare the michaelian model with the competitive inhibition model, just replace the value of Kic with a consistent <span class="fu">number</span> (e.g., <span class="at">Kic=</span> <span class="dv">3</span>).</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="st">"C. Uncompetitive inhibition model."</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> The same suggestion above serves <span class="cf">for</span> the uncompetitive model, this time replacing the value <span class="cf">for</span> Kiu.</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="st">"D. Pure noncompetitive inhibition model."</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> In this model, the simulation is given by equal values <span class="cf">for</span> Kic and Kiu.</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="st">"E. Mixed noncompetitive inhibition model."</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> For this model, just allocate distinct values <span class="cf">for</span> Kic and Kiu.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="linearization-of-kinetic-data-by-lineweaverburk" class="level3" data-number="1.10">
<h3 data-number="1.10" class="anchored" data-anchor-id="linearization-of-kinetic-data-by-lineweaverburk"><span class="header-section-number">1.10</span> Linearization of kinetic data by Lineweaver–Burk</h3>
<section id="context-9" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-9">Context</h4>
<div class="line-block">&nbsp;&nbsp;One of the (many) difficulties in learning Biochemistry refers to literacy for interpreting plots. Among these, the hyperbolic nature of the michaelian model is alternatively treated by several transformations of the variables <em>S</em> and <em>v</em>, standing out in the literature the linearization by <em>double-reciprocal</em>, or <em>Lineweaver–Burk</em> transformation.</div>
<div class="line-block">&nbsp;&nbsp;The example that follows illustrates the relationship between the dependent and independent variables of the original and transformed model, using another feature of the <em>Plotly.js</em> library: the possibility of cross-selection of points.</div>
<div class="line-block">&nbsp;&nbsp;To observe this functionality, select a set of points in one of the <em>subplots</em> of the plot that follows, and observe their automatic selection in the other subplot.</div>
</section>
<section id="equation-7" class="level4 unnumered" data-number="1.10.1">
<h4 class="unnumered anchored" data-number="1.10.1" data-anchor-id="equation-7"><span class="header-section-number">1.10.1</span> Equation</h4>
<div class="line-block">&nbsp;&nbsp;The linear transformation of Lineweaver–Burk is given below:</div>
<p><span class="math display">\[
\frac{1}{v} = \frac{1}{S}*\frac{Km}{Vm} + \frac{1}{Vm}
\]</span><br>
</p>
<p><a href="Eq/jsp_LB.html" target="_blank"><img src="Eq/LB.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-9" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-9">Suggestion</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try selecting the extremes of the Michaelis–Menten <span class="fu">plot</span> (hyperbolic curve), and observe the same points <span class="cf">in</span> the double<span class="sc">-</span>reciprocal. Which values are more reliable <span class="cf">in</span> the latter, the first or the last?</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Research other forms of <span class="fu">linearization</span> (e.g., Eadie–Hofstee; <span class="st">"v/S"</span> X <span class="st">"v"</span>), and see how the transformation and point selection presents itself. For this, change the constants below<span class="sc">:</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>    const invS <span class="ot">=</span> <span class="fu">S.map</span>(<span class="at">s =</span><span class="sc">&gt;</span> <span class="dv">1</span><span class="sc">/</span>s);</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>    const invV <span class="ot">=</span> <span class="fu">V.map</span>(<span class="at">v =</span><span class="sc">&gt;</span> <span class="dv">1</span><span class="sc">/</span>v);</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="diagnosis-of-enzymatic-inhibition-michaelismenten-and-lineweaverburk" class="level3" data-number="1.11">
<h3 data-number="1.11" class="anchored" data-anchor-id="diagnosis-of-enzymatic-inhibition-michaelismenten-and-lineweaverburk"><span class="header-section-number">1.11</span> Diagnosis of Enzymatic Inhibition (Michaelis–Menten and Lineweaver–Burk)</h3>
<section id="context-10" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-10">Context</h4>
<div class="line-block">&nbsp;&nbsp;As presented above and present in texts of the area, it is possible to linearize the Michaelis–Menten curve for a simpler observation for diagnosing enzymatic inhibition models. The object below proposes this interpretation. For this, keep the <em>control enzyme</em> without addition of inhibitor in the code (<em>I=0</em>). To seek an inhibition model, insert a value for <em>I</em> and change the inhibitor dissociation equilibrium constant(s), <em>kic</em> and/or <em>kiu</em>.</div>
<p><a href="Eq/jsp_inib_diag.html" target="_blank"><img src="Eq/inib_diag.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-10" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-10">Suggestion</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Simulate the conditions <span class="cf">for</span> a competitive inhibition, inserting a value <span class="cf">for</span> the inhibitor <span class="fu">content</span> (<span class="st">"I"</span>) and changing the value of <span class="st">"Km"</span>;</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Do the same <span class="cf">for</span> an uncompetitive model, although changing the value of <span class="st">"Vm"</span>;</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Try the pure noncompetitive model, changing <span class="st">"Km"</span> and <span class="st">"Vm"</span> to the same value;</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Test mixed competitive inhibition, inserting distinct values <span class="cf">for</span> <span class="st">"Km"</span> and <span class="cf">for</span> <span class="st">"Vm."</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="van-der-waals-equation-for-ideal-gases" class="level3" data-number="1.12">
<h3 data-number="1.12" class="anchored" data-anchor-id="van-der-waals-equation-for-ideal-gases"><span class="header-section-number">1.12</span> van der Waals equation for ideal gases</h3>
<section id="context-11" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-11">Context:</h4>
<div class="line-block">&nbsp;&nbsp;An adaptation that relates the thermodynamic quantities of pressure, volume and temperature for ideal gases, is the <em>van der Waals equation</em>. In this, coefficients are computed that estimate the existence of a volume and inter-particle interactions. Thus, the van der Waals equation corrects the ideal gas equation considering a term to compensate intermolecular forces (a/V<span class="math inline">\(^{2}\)</span>) and the available volume, this discounting the volume occupied by the gas molecules themselves.</div>
<div class="line-block">&nbsp;&nbsp;In the simulation an additional interactivity is offered by the presence of <em>sliders</em> (sliding controls) for temperature, and for the coefficients of finite volume (<em>b</em>) and interaction between particles (<em>a</em>).</div>
</section>
<section id="equation-8" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-8">Equation:</h4>
<p><span class="math display">\[
P = \frac{RT}{V - b} - \frac{a}{V^2}
\]</span></p>
<ul>
<li>P = gas pressure (atm);</li>
<li>V = molar volume (L);</li>
<li>T = temperature (K);</li>
<li>R = 0.0821 = ideal gas constant (L·atm/mol·K);</li>
<li>a = intermolecular attraction constant (L<span class="math inline">\(^{2}\)</span>·atm/mol$^{2})</li>
<li>b = excluded volume constant (L/mol)</li>
</ul>
<p><a href="Eq/jsp_vanderWaals.html" target="_blank"><img src="Eq/vanderWaals.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-11" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-11">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try varying the parameters of the equation by means of the <span class="st">"slider"</span> <span class="cf">for</span> temperature, as well as <span class="cf">for</span> the coefficients <span class="st">"a and b"</span>.</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Discuss which of the coefficients has the greatest effect on the curve profile, and the reason <span class="cf">for</span> this.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="equilibrium-of-atp-production-from-reagents-temperature-and-ph" class="level3" data-number="1.13">
<h3 data-number="1.13" class="anchored" data-anchor-id="equilibrium-of-atp-production-from-reagents-temperature-and-ph"><span class="header-section-number">1.13</span> Equilibrium of ATP production from reagents, temperature, and pH</h3>
<section id="context-12" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-12">Context:</h4>
<div class="line-block">&nbsp;&nbsp;Intracellular ATP production involves the classic relationship of chemical equilibrium between reagents and products as a function of reaction temperature, and adjusted for a given pH value. By varying one and/or another reagent content, or physicochemical parameter, it is possible to quantify the product by the thermodynamic reaction that follows:</div>
</section>
<section id="equation-9" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-9">Equation:</h4>
<p><span class="math display">\[
\Delta G = \Delta G^{\circ'} + RT \ln\left(\frac{[\text{ADP}] \cdot [\text{P}_i]}{[\text{ATP}]}\right) + 2{,}303 \cdot RT \cdot n_H \cdot \text{pH}
\]</span></p>
<p><em>Where</em>,</p>
<ul>
<li><span class="math inline">\(\Delta\)</span>G = Gibbs energy of the reaction (positive for spontaneously unfavorable synthesis, kJ/mol);</li>
<li><span class="math inline">\(\Delta\)</span>G<span class="math inline">\(^{o'}\)</span> = 30.5 kJ/mol biological standard Gibbs energy for ATP synthesis;</li>
<li>R = 8.314 J/mol/K (general gas constant);</li>
<li>T=310 K (physiological temperature);</li>
<li>nH<span class="math inline">\(^{+}\)</span> = 1 (number of protons involved in the reaction);</li>
<li>[ADP], [Pi], [ATP] = molar concentrations of reagents and product</li>
</ul>
<p><a href="Eq/jsp_ATP.html" target="_blank"><img src="Eq/ATP.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-12" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-12">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Change the quantities involved <span class="cf">in</span> the expression, and compare with previous visualizations. For example, temperature, pH, and ADP and Pi contents.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="variation-of-gibbs-energy-with-temperature" class="level3" data-number="1.14">
<h3 data-number="1.14" class="anchored" data-anchor-id="variation-of-gibbs-energy-with-temperature"><span class="header-section-number">1.14</span> Variation of Gibbs energy with temperature</h3>
<section id="context-13" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-13">Context:</h4>
<div class="line-block">&nbsp;&nbsp;The Gibbs–Helmholtz relationship predicts that Gibbs energy varies in a non-linear way with temperature, in reactions that involve a change in the heat capacity of the system (<span class="math inline">\(\Delta\)</span>Cp). This expanded form of the Gibbs equation for variable heat capacity is presented in several biochemical phenomena, such as in phase transition of biomembrane structure subjected to a compound challenge, or in the conformational change that accompanies protein structure under heating. \</div>
</section>
<section id="equation-10" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-10">Equation:</h4>
<p><span class="math display">\[
\Delta G(T) = \Delta H^\circ - T,\Delta S^\circ + \Delta C_p \left(T - T_0 - T \ln\left(\frac{T}{T_0}\right)\right)
\]</span></p>
<p><em>Where</em>,</p>
<ul>
<li><span class="math inline">\(\Delta\)</span>G(T) = Gibbs energy of the reaction at each temperature value, kJ/mol);</li>
<li><span class="math inline">\(\Delta\)</span>H<span class="math inline">\(^{o}\)</span> = standard enthalpy of the reaction at T<span class="math inline">\(_{0}\)</span>, usually 298 K (J/mol);</li>
<li><span class="math inline">\(\Delta\)</span>S<span class="math inline">\(^{o}\)</span> = standard entropy of the reaction at T<span class="math inline">\(_{0}\)</span>;</li>
<li><span class="math inline">\(\Delta\)</span>Cp = change in heat capacity of the reaction (J/mol·K), assumed constant with temperature;</li>
<li>T = temperature of interest (K);</li>
<li>T<span class="math inline">\(_{0}\)</span> = reference temperature, generally 298 K.</li>
<li>R = 8.314 J/mol/K (general gas constant);</li>
</ul>
<p><a href="Eq/jsp_gibbs.html" target="_blank"><img src="Eq/gibbs.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-13" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-13">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try varying one or more parameters of the expression;</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Test the behavior of the Gibbs curve at an elevated reference <span class="fu">temperature</span> (simulation <span class="cf">for</span> an extremophile organism);</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Simulate the situation where the heat capacity change is null</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="oxygen-consumption-by-mitochondria" class="level3" data-number="1.15">
<h3 data-number="1.15" class="anchored" data-anchor-id="oxygen-consumption-by-mitochondria"><span class="header-section-number">1.15</span> Oxygen consumption by mitochondria</h3>
<section id="context-14" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-14">Context:</h4>
<div class="line-block">&nbsp;&nbsp;<em>Oxygraphs</em> are equipment that monitor the content of dissolved oxygen in solution. Different from <em>oximeters</em>, based on measuring hemoglobin oxygenation by a direct <em>clip</em> on the tip of a finger, oxygraphs are widely used in studies of living cells and suspensions of mitochondria.</div>
<div class="line-block">&nbsp;&nbsp;The example that follows illustrates an <em>animation</em> for the content of <span class="math inline">\(O_{2}\)</span> measured in an oxygraph for a suspension of mitochondria, and whose consumption rates are varied by the use of modifying metabolites (e.g., pyruvate, ADP, azide).</div>
<p><br>
</p>
<p><a href="Eq/jsp_O2_mitocondria.html" target="_blank"><img src="Eq/O2_mitocondria.png" class="img-fluid"></a><br>
</p>
</section>
<section id="suggestion-14" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-14">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Note that the appearance of metabolites and the resulting oxygen consumption rates are defined by the <span class="cf">function</span> <span class="st">"oxigenio"</span>. Try changing them and observe the effect <span class="cf">in</span> the animation;</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Realize that the metabolites and their action, as well as the measured signal, O2 content, are easily adapted <span class="cf">for</span> any other metabolic measurement <span class="cf">in</span> the code. Try, <span class="cf">for</span> example, what the plot would look like <span class="cf">in</span> a simulation <span class="cf">for</span> the glycolytic pathway;</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> This code was designed <span class="cf">for</span> an animation. Go to the <span class="st">"Applications"</span> section below, and try <span class="st">"FlowForces"</span> <span class="cf">for</span> a similar purpose, although with full interactive adaptation.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="electrochemical-impedance-spectroscopy-eis" class="level3" data-number="1.16">
<h3 data-number="1.16" class="anchored" data-anchor-id="electrochemical-impedance-spectroscopy-eis"><span class="header-section-number">1.16</span> Electrochemical impedance spectroscopy (EIS)</h3>
<section id="context-15" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-15">Context:</h4>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Electrochemical impedance spectroscopy (EIS) is an analytical technique that measures the response of an electrochemical system to a small-amplitude sinusoidal perturbation (e.g., 5–10 mV), applied over a wide frequency range (e.g., 100 kHz–1 mHz). EIS allows obtaining information about interfacial processes, such as charge transfer, ion diffusion and formation of passive films, through analysis of the complex impedance of the system. It is a non-destructive technique, widely used in areas such as corrosion, batteries, fuel cells, biosensors and protective coatings.</div>
</section>
<section id="equation-11" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-11">Equation:</h4>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;An electrochemical impedance (EIS) model with typical equivalent-circuit components is the modified Randles: series resistor <em>Rs</em>, a resistor <em>Rp</em> in parallel with a <em>CPE</em> (Constant Phase Element), and a <em>Warburg</em> element. A common representation for EIS is the <em>Nyquist</em> plot, in which the real axis represents resistance (Zre) and the imaginary axis the negative reactance (−Zim). The formalism involved is in the equation below:</div>
<p><span class="math display">\[
Z_{\text{total}}(\omega) = R_s + \left[ \left( \frac{1}{R_p} + Q (j\omega)^n \right)^{-1} \right] + \frac{\sigma}{\sqrt{\omega}} (1 - j)
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>Z<span class="math inline">\(_{total}\)</span> = total impedance at a given frequency ;</li>
<li><span class="math inline">\(\omega\)</span> = angular frequency (rad/s), <span class="math inline">\(\omega\)</span>=2πf or <span class="math inline">\(\omega\)</span>=2πf;</li>
<li>R<span class="math inline">\(_{p}\)</span> = ohmic resistance (resistance of the electrolyte solution, wires, contacts etc.) ;</li>
<li>R<span class="math inline">\(_{p}\)</span> = polarization resistance (associated with charge-transfer processes, such as electrochemical reactions);</li>
<li><em>Q</em> = constant associated with the constant phase element (CPE), replaces an ideal capacitor to represent non-ideal behaviors ;</li>
<li><em>n</em> = CPE exponent, between 0 and 1; defines the degree of ideality of the capacitive behavior (n = 1: ideal capacitor; n &lt; 1: dispersion);</li>
<li><span class="math inline">\(\sigma\)</span> = Warburg coefficient, associated with ion diffusion in the electrochemical system ;</li>
<li><em>j</em> = imaginary unit, j<span class="math inline">\(^{2}\)</span> = −1;</li>
</ul>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;An electrochemical impedance (EIS) model with typical equivalent-circuit components is the modified Randles: series resistor <em>Rs</em>, a resistor <em>Rp</em> in parallel with a <em>CPE</em> (Constant Phase Element), and a <em>Warburg</em> element. A common representation for EIS is the <em>Nyquist</em> plot, in which the real axis represents resistance (Zre) and the imaginary axis the negative reactance (−Zim).</div>
<p><a href="Eq/jsp_EIS.html" target="_blank"><img src="Eq/EIS.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-15" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-15">Suggestion</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Check the effect of Rs on the plot, nullifying its <span class="fu">value</span> (solution resistance);</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Observe the deformation of the semicircle by varying the values of the constant phase <span class="fu">element</span> (e.g., <span class="at">Q =</span> <span class="fl">1e-3</span>; <span class="at">n =</span> <span class="fl">0.6</span> <span class="sc">-</span> dispersion of the capacitive behavior); </span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try combining other values of the parameters <span class="cf">in</span> the code header, to highlight common situations <span class="cf">in</span> electroanalysis<span class="sc">:</span> Rs, Rp, Q, n, and sigma;</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Reduce the Warburg<span class="sc">-</span>with<span class="sc">-</span>constant<span class="sc">-</span>phase<span class="sc">-</span>element model to a simple Randles model, formed only by <span class="dv">2</span> resistors <span class="cf">in</span> series, the second <span class="cf">in</span> parallel with an ideal capacitor, and without Warburg <span class="fu">diffusion</span> (<span class="at">sigma =</span> <span class="dv">0</span>; <span class="at">n =</span> <span class="dv">1</span>).</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="cyclic-voltammetry" class="level3" data-number="1.17">
<h3 data-number="1.17" class="anchored" data-anchor-id="cyclic-voltammetry"><span class="header-section-number">1.17</span> Cyclic voltammetry</h3>
<section id="context-16" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-16">Context:</h4>
<div class="line-block">&nbsp;&nbsp;Cyclic voltammetry (CV) is an electroanalytical technique for characterization of interfacial redox processes (electrode-solution). When a standard analyte in quantitative equilibrium between oxidized and reduced forms is used, the resulting plot acquires an unusual profile, named <em>duck shape</em>. This results from the peculiar format under a bidirectional potentiometric sweep over a set of electrodes by a <em>potentiostat</em>, resulting in oxidation cycles and subsequent reduction (application of positive potentials followed by negative ones). The shape, value and distance between the resulting peaks allow identifying and characterizing the analyte, as well as its redox processes involved in the experiment.</div>
</section>
<section id="equation-12" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-12">Equation:</h4>
<div class="line-block">&nbsp;&nbsp;Although the formalism for cyclic voltammetry is complex and involves solving <em>ordinary differential equations</em> (as do electroanalytical techniques in general), being far from the objective of this material, one can summarize the resulting current flow by the <em>Buttler–Volmer equation</em> that follows:</div>
<p><span class="math display">\[
j = j_0 \left[ \exp\left(\frac{\alpha n F (E - E^0)}{RT}\right) - \exp\left(\frac{-(1 - \alpha) n F (E - E^0)}{RT}\right) \right]
\]</span> <em>Where:</em></p>
<ul>
<li>j = current density;</li>
<li>j<span class="math inline">\(_{0}\)</span> = exchange current;</li>
<li><span class="math inline">\(\alpha\)</span> = charge transfer coefficient;</li>
<li>E = potential applied to the electrode;</li>
<li>E<span class="math inline">\(^{0}\)</span> = standard electrode potential;</li>
<li>n = number of electrons;</li>
<li>F = Faraday constant (96485 C·mol⁻¹)</li>
<li>R = general gas constant (8.314 J·mol⁻¹·K⁻¹);</li>
<li>T = temperature<br>
</li>
</ul>
<div class="line-block">&nbsp;&nbsp;For a model of diffusion of electroanalyte and accumulation on the electrode surface, the general equation for the boundary condition where the redox conversion rate is proportional to the diffusion current, can be represented by the <em>equation of the 2nd Fick’s law</em> for mass transport:</div>
<p><span class="math display">\[
\frac{\partial C(x,t)}{\partial t} = D \frac{\partial^2 C(x,t)}{\partial x^2}
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>C(x,t) = concentration of the electroanalytical species (mol/cm³), as a function of position <em>x</em> and time <em>t</em>;</li>
<li>D = diffusion coefficient of the species (cm²/s);</li>
<li>t = time (s);</li>
<li>x = distance from the electrode surface (cm);</li>
<li><span class="math inline">\(\delta\)</span> = notation for partial derivative.<br>
</li>
</ul>
<div class="line-block">&nbsp;&nbsp;&nbsp;The chemical equilibrium of the redox reaction is given by the ratio between the concentrations of oxidized and reduced form at the electrode/solution interface when the system is in equilibrium (or near-equilibrium), according to the <em>Nernst equation</em>:</div>
<p><span class="math display">\[
E = E^0 + \frac{RT}{nF} \ln \left( \frac{[\text{Ox}]}{[\text{Red}]} \right)
\]</span></p>
<div class="line-block">&nbsp;&nbsp;The resulting current is obtained by <em>Faraday’s law</em>:</div>
<p><span class="math display">\[
i(t) = n F A D \left. \frac{\partial C(x,t)}{\partial x} \right|_{x=0}
\]</span> <em>Where:</em></p>
<ul>
<li>A = electrode surface area (cm²);</li>
<li>i(t) = current at time <em>t</em> (A, Ampere).</li>
</ul>
</section>
<section id="cyclic-voltammetry-simulator" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="cyclic-voltammetry-simulator">Cyclic Voltammetry Simulator</h4>
<div class="line-block">&nbsp;&nbsp;To illustrate the potential of the <em>JSPlotly/GSPlotly</em> platform for the development of complex <em>virtual experiments</em>, a <em>cyclic voltammetry simulator</em> follows resulting from the procedures above (as well as from a few hours of prompt engineering!!).</div>
<div class="line-block">&nbsp;&nbsp;The example illustrates a typical behavior in cyclic voltammetry (and the curious <em>duck shape</em>), and includes a diffusion and accumulation modeling of electroactive species near the electrode surface. The code uses a <em>numerical solution</em> to obtain current density over time, considering the effects of cyclic scanning of the applied potential, mass transport, and reaction kinetics.</div>
<div class="line-block">&nbsp;&nbsp;In sum, it numerically solves Fick’s equation with a dynamic boundary condition, applies the Nernst equation at the boundary (electrode surface) to determine conversion between species, uses the finite-difference method or an explicit scheme to advance in time, and obtains the current by applying Faraday’s law, by the derivative of the flux at the interface.</div>
<div class="line-block">&nbsp;&nbsp;Found it complex? Then how about: the code unites Fick (transport) and Nernst (local redox equilibrium) to accurately model the voltammogram shape, obtaining a physical simulation with significant realism for cyclic voltammetry.</div>
<p><br>
</p>
<p><a href="Eq/jsp_cv.html" target="_blank"><img src="Eq/cv.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-16" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-16">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Observe the number of values at the beginning of the code, tangible to a <span class="st">"parametric manipulation"</span>. Try to know what they represent, and try to vary their values consciously, aiming to add value to learning the simulation. This is the soul of <span class="st">"parametric manipulation"</span> that involves <span class="st">"reproducible teaching"</span> <span class="sc">!!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="diagrams-and-flowcharts" class="level3" data-number="1.18">
<h3 data-number="1.18" class="anchored" data-anchor-id="diagrams-and-flowcharts"><span class="header-section-number">1.18</span> Diagrams and flowcharts</h3>
<section id="context-diagram" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-diagram">Context: Diagram</h4>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;By the nature of the <em>Plotly.js</em> library, it is also possible to generate codes in <em>JSPlotly</em> without involving an equation, such as for representing diagrams and flowcharts, among others. The example below illustrates a representation of the Krebs cycle, and the next an experimental flowchart.</div>
<p><a href="Eq/jsp_krebs.html" target="_blank"><img src="Eq/krebs.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-17" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-17">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try repositioning enzymes and metabolites better, just by clicking and dragging the terms;</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try replacing the names that are <span class="cf">in</span> the code to produce another metabolic cycle, such as the urea cycle.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="context-flowchart" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-flowchart">Context: Flowchart</h4>
<p><a href="Eq/jsp_fluxograma.html" target="_blank"><img src="Eq/fluxograma.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-18" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-18">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Reposition terms and connectors by mouse dragging;</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> For a flowchart different <span class="cf">in</span> content, modify the terms <span class="cf">in</span> the code;</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> For a flowchart different <span class="cf">in</span> format, change the font and connector characteristics <span class="cf">in</span> the <span class="st">"annotations"</span> constant.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
</section>
<section id="construction-of-plots" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="construction-of-plots"><span class="header-section-number">2</span> Construction of plots</h2>
<div class="line-block">&nbsp;&nbsp;Although a large part of representative plots involves only lines and/or points on a plane, the <code>Plotly.js</code> library is prodigious in elaborating a significant set of varied types, as represented in its version for the <a href="https://www.r-project.org/">R</a> language on the <a href="https://plotly.com/r/">developer site</a>.</div>
<div class="line-block">&nbsp;&nbsp;In addition, even for a simple plot of points and/or lines, it is possible to have a large number of distinct configurations and interactive <em>widgets</em>, some of which are represented below.</div>
<section id="plot-with-2-y-axes" class="level3" data-number="2.1">
<h3 data-number="2.1" class="anchored" data-anchor-id="plot-with-2-y-axes"><span class="header-section-number">2.1</span> Plot with 2 y-axes</h3>
<p><a href="Eq/jsp_2eixos.html" target="_blank"><img src="Eq/2eixos.png" class="img-fluid"></a><br>
</p>
</section>
<section id="rangeslider" class="level3" data-number="2.2">
<h3 data-number="2.2" class="anchored" data-anchor-id="rangeslider"><span class="header-section-number">2.2</span> RangeSlider</h3>
<p><a href="Eq/jsp_rangeslider.html" target="_blank"><img src="Eq/rangeslider.png" class="img-fluid"></a></p>
</section>
<section id="subplots" class="level3" data-number="2.3">
<h3 data-number="2.3" class="anchored" data-anchor-id="subplots"><span class="header-section-number">2.3</span> Subplots</h3>
<p><a href="Eq/jsp_subplots.html" target="_blank"><img src="Eq/subplots.png" class="img-fluid"></a></p>
</section>
<section id="subplots-with-different-types" class="level3" data-number="2.4">
<h3 data-number="2.4" class="anchored" data-anchor-id="subplots-with-different-types"><span class="header-section-number">2.4</span> Subplots with different types</h3>
<p><a href="Eq/jsp_subplots_tipos.html" target="_blank"><img src="Eq/subplots_tipos.png" class="img-fluid"></a></p>
</section>
<section id="synchronized-multiple-selection" class="level3" data-number="2.5">
<h3 data-number="2.5" class="anchored" data-anchor-id="synchronized-multiple-selection"><span class="header-section-number">2.5</span> Synchronized multiple selection</h3>
<p><a href="Eq/jsp_subplots_cruzados.html" target="_blank"><img src="Eq/subplots_cruzados.png" class="img-fluid"></a></p>
</section>
<section id="click-event-message" class="level3" data-number="2.6">
<h3 data-number="2.6" class="anchored" data-anchor-id="click-event-message"><span class="header-section-number">2.6</span> Click event &amp; message</h3>
<p><a href="Eq/jsp_cliqueMensagem.html" target="_blank"><img src="Eq/cliqueMensagem.png" class="img-fluid"></a></p>
</section>
<section id="menu-for-selecting-types" class="level3" data-number="2.7">
<h3 data-number="2.7" class="anchored" data-anchor-id="menu-for-selecting-types"><span class="header-section-number">2.7</span> Menu for selecting types</h3>
<p><a href="Eq/jsp_menu.tipos.html" target="_blank"><img src="Eq/menu.tipos.png" class="img-fluid"></a></p>
</section>
<section id="slider-control" class="level3" data-number="2.8">
<h3 data-number="2.8" class="anchored" data-anchor-id="slider-control"><span class="header-section-number">2.8</span> Slider control</h3>
<p><a href="Eq/jsp_slider.Hill.html" target="_blank"><img src="Eq/slider.Hill.png" class="img-fluid"></a></p>
</section>
</section>
<section id="data-analysis" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="data-analysis"><span class="header-section-number">3</span> Data analysis</h2>
<section id="user-data-entry" class="level3" data-number="3.1">
<h3 data-number="3.1" class="anchored" data-anchor-id="user-data-entry"><span class="header-section-number">3.1</span> User data entry</h3>
<section id="context-17" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-17">Context</h4>
<div class="line-block">&nbsp;&nbsp;Sometimes it is useful to work with one’s own discrete data, seeking a trend relationship—mathematical or simply visual—for the relationship between variables. Below is an example of an interactive plot with user data.</div>
<p><br>
</p>
<p><a href="Eq/jsp_dados.html" target="_blank"><img src="Eq/dados.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-19" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-19">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try changing the inserted data, overlaying the plot or not;</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try changing the data representation <span class="cf">in</span> <span class="st">"mode"</span> and <span class="st">"type"</span>, to points, lines, points<span class="sc">+</span>lines, bars.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="loading-a-file-for-analysis" class="level3" data-number="3.2">
<h3 data-number="3.2" class="anchored" data-anchor-id="loading-a-file-for-analysis"><span class="header-section-number">3.2</span> Loading a file for analysis</h3>
<section id="context---csv-file" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context---csv-file">Context - CSV file</h4>
<div class="line-block">&nbsp;&nbsp;Sometimes, there is a need to analyze results referring to a dataset present in a spreadsheet. For this situation, the following example illustrates loading two data vectors from a <em>CSV</em> (<em>comma separated value</em>) file, a format commonly used for export/import in electronic spreadsheets, and its resulting plot.</div>
<p><a href="Eq/jsp_csv.html" target="_blank"><img src="Eq/csv.png" class="img-fluid"></a></p>
<p><strong>Instructions:</strong></p>
<ul>
<li>1 Click <em>add plot</em> and select a CSV file using the upper <em>browse</em> button that is shown. Note: variable <em>X</em> in the 1st column of the file, and variable <em>Y</em> in the 2nd column;</li>
<li><ol start="2" type="1">
<li>Click <em>add plot</em> again to view the resulting plot.</li>
</ol></li>
</ul>
</section>
<section id="suggestion-20" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-20">Suggestion</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try other CSV files;</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Vary aspects of the plot, such as type, color, marker size, etc.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="linear-fitting-of-data" class="level3" data-number="3.3">
<h3 data-number="3.3" class="anchored" data-anchor-id="linear-fitting-of-data"><span class="header-section-number">3.3</span> Linear fitting of data</h3>
<section id="context-18" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-18">Context:</h4>
<div class="line-block">&nbsp;&nbsp;In many situations in experimental research, it is necessary to correlate the data of a dependent variable with a predictor variable, according to the equation below.</div>
</section>
<section id="equation-13" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-13">Equation:</h4>
<p><span class="math display">\[
y = \alpha x + \beta + \varepsilon
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>y = dependent variable;</li>
<li>x = independent variable;</li>
<li><span class="math inline">\(\alpha\)</span> = slope of the fitted line (<em>slope</em>);</li>
<li><span class="math inline">\(\beta\)</span> = intercept of the fitted line;</li>
<li><span class="math inline">\(\epsilon\)</span> = measurement error.</li>
</ul>
<div class="line-block">&nbsp;&nbsp;The parameters <em>slope and intercept</em> can be obtained by some procedures, such as <em>least squares</em>, <em>matrix algebra</em>, or by simple <em>summations</em>. In the latter case, the computation of parameters and error can be calculated as follows:</div>
<p><span class="math display">\[
\alpha = \frac{n \sum x_i y_i - \sum x_i \sum y_i}{n \sum x_i^2 - \left( \sum x_i \right)^2}
\]</span></p>
<p><span class="math display">\[
\beta = \frac{\sum y_i - \alpha \sum x_i}{n}
\]</span></p>
<p><span class="math display">\[
\hat{y}_i = \alpha x_i + \beta
\]</span></p>
<p><span class="math display">\[
\varepsilon_i = \left| y_i - \hat{y}_i \right|
\]</span></p>
<div class="line-block">&nbsp;&nbsp;A representation for linear fitting with arbitrary data is provided below:</div>
<p><br>
</p>
<p><a href="Eq/jsp_ajLinR2.html" target="_blank"><img src="Eq/ajLin.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-21" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-21">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Show the points overlaid or not with the fitted line. For overlay, choose <span class="st">"mostrarPontos = true"</span> and <span class="st">"mostrarReta = true"</span>;</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Change the data and perform a new fit, to obtain other line parameters.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="polynomial-regression" class="level3" data-number="3.4">
<h3 data-number="3.4" class="anchored" data-anchor-id="polynomial-regression"><span class="header-section-number">3.4</span> Polynomial regression</h3>
<section id="context-19" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-19">Context:</h4>
<div class="line-block">&nbsp;&nbsp;In some situations, the linear fit does not provide an adequate trend for modeling experimental data, which calls for the use of other models, such as <em>exponential, hyperbolic, logarithmic, power, and polynomial</em>. Polynomial trends are common in natural phenomena, with mathematical treatment of the data occurring through different methods, such as simple summations, least squares, or linear algebra. Below is an example of polynomial regression with degrees adjustable in the code.</div>
</section>
<section id="equation-14" class="level4" data-number="3.4.1">
<h4 data-number="3.4.1" class="anchored" data-anchor-id="equation-14"><span class="header-section-number">3.4.1</span> Equation</h4>
<div class="line-block">&nbsp;&nbsp;Error minimization for the polynomial curve is obtained by a generalized <em>least-squares regression</em>, and application of the <em>Vandermonde matrix</em>, as follows.</div>
<div class="line-block">&nbsp;&nbsp;Let the vectors x=[x<span class="math inline">\(*{1}\)</span>,x<span class="math inline">\(*{2}\)</span>,…,x<span class="math inline">\(*{n}\)</span>] and y=[y<span class="math inline">\(*{1}\)</span>,y<span class="math inline">\(_{2}\)</span>,…,y<span class="math inline">\({_n}\)</span>], and we want to fit the polynomial:</div>
<p><span class="math display">\[
y = \beta_0 + \beta_1 x + \beta_2 x^2 + \cdots + \beta_g x^g
\]</span> | Thus, the Vandermonde matrix is built as:</p>
<p><span class="math display">\[
X =
\begin{bmatrix}
1 &amp; x_1 &amp; x_1^2 &amp; \cdots &amp; x_1^g \\
1 &amp; x_2 &amp; x_2^2 &amp; \cdots &amp; x_2^g \\
\vdots &amp; \vdots &amp; \vdots &amp; &amp; \vdots \\
1 &amp; x_n &amp; x_n^2 &amp; \cdots &amp; x_n^g
\end{bmatrix}
\]</span></p>
<div class="line-block">&nbsp;&nbsp;To find the <em>coefficients <span class="math inline">\(\beta\)</span></em> that minimize the squared error, the linear system is solved in sequence:</div>
<p><span class="math display">\[
\boldsymbol{\beta} = (X^T X)^{-1} X^T \mathbf{y}
\]</span><br>
</p>
<p><em>Where:</em></p>
<ul>
<li><em>T</em> represents the transpose matrix<br>
</li>
</ul>
<p><a href="Eq/jsp_polinomial.html" target="_blank"><img src="Eq/polinomial.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-22" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-22">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try degree <span class="dv">1</span> <span class="cf">for</span> the polynomial, i.e., reduce the treatment to a linear fit;</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Change the formatting of labels, colors, sizes, etc., <span class="cf">in</span> the code;</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Overlay some fits, edit and reposition the legend;</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Test the code with another data vector.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="multiple-linear-regression" class="level3" data-number="3.5">
<h3 data-number="3.5" class="anchored" data-anchor-id="multiple-linear-regression"><span class="header-section-number">3.5</span> Multiple linear regression</h3>
<section id="context-20" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-20">Context:</h4>
<div class="line-block">&nbsp;&nbsp;Multiple linear fitting is very useful when one wants to predict a behavior as a function of two or more predictor variables. The procedure involves a <em>design matrix</em> (or <em>projection operator</em>) similar to the <em>Vandermonde matrix</em>.</div>
</section>
<section id="equation-15" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-15">Equation:</h4>
<div class="line-block">&nbsp;&nbsp;The analytical solution for the design matrix involves solving a coefficient matrix <span class="math inline">\(\beta\)</span> for a set of predictor values of the response vector <em>y</em>, as follows:</div>
<p><span class="math display">\[
y_i = \beta_0 + \beta_1 x_{1i} + \beta_2 x_{2i} + \cdots + \beta_p x_{pi} + \varepsilon_i
\]</span></p>
<p><span class="math display">\[
\mathbf{y} = \mathbf{X} \boldsymbol{\beta} + \boldsymbol{\varepsilon}
\]</span></p>
<p><span class="math display">\[
\hat{\boldsymbol{\beta}} = (\mathbf{X}^\top \mathbf{X})^{-1} \mathbf{X}^\top \mathbf{y}
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li><span class="math inline">\(\beta\)</span> = vector of coefficients;</li>
<li><em>y</em> = response vector;</li>
<li><em>X</em> = design matrix;</li>
<li><span class="math inline">\(\epsilon\)</span> = random noise</li>
</ul>
<div class="line-block">&nbsp;&nbsp;Thus, predicted values and associated error are obtained by:</div>
<p><span class="math display">\[
\hat{\mathbf{y}} = \mathbf{X} \hat{\boldsymbol{\beta}}
\]</span> <span class="math display">\[
\mathbf{e} = \mathbf{y} - \hat{\mathbf{y}}
\]</span></p>
<div class="line-block">&nbsp;&nbsp;Obviously, the graphical representation for a multiple linear fit is limited to two independent variables, <em>x1</em> and <em>x2</em>, in addition to the response variable (<em>3D plot</em>), although the matrix calculation allows multiple predictor variables (hyperdimensional space). With respect to the first situation, an example follows.</div>
<p><strong>Instructions</strong></p>
<ol type="1">
<li><p>Note that there is a boolean flag (<em>mostrarAjuste</em>) at the beginning of the code: <em>false</em> for data only, and <em>true</em> for the fit;</p></li>
<li><p>You can click <em>add plot</em> to view the data with the flag set to <em>false</em>, followed by another <em>add plot</em> with the flag set to <em>true</em>.<br>
</p></li>
</ol>
<p><a href="Eq/jsp_multilinear.html" target="_blank"><img src="Eq/multilinear.png" class="img-fluid"></a><br>
</p>
<div class="line-block">&nbsp;&nbsp;Rotate the plot and observe the surface around the points. Unlike a linear fit, the points are distributed non-continuously in space.</div>
</section>
<section id="suggestion-23" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-23">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> <span class="fu">QSAR</span> (<span class="st">"Quantitative Structure-Activity Relationship"</span>) results use multiple linear fitting analysis to identify the strength of predictor variables. Search online <span class="cf">for</span> a dataset that has <span class="dv">2</span> <span class="fu">variables</span> (e.g., concentration, pH, compound A, B, etc.)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> For more predictor variables, see the code that follows <span class="sc">!!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="multiple-linear-fitting-with-3-or-more-predictor-variables" class="level3" data-number="3.6">
<h3 data-number="3.6" class="anchored" data-anchor-id="multiple-linear-fitting-with-3-or-more-predictor-variables"><span class="header-section-number">3.6</span> Multiple linear fitting with 3 or more predictor variables</h3>
<section id="context-21" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-21">Context</h4>
<div class="line-block">&nbsp;&nbsp;Although <em>JSPlotly</em> was designed for producing interactive plots and maps using the <em>Plotly.js</em> library on the shoulders of the <em>JavaScript</em> language, it extends the possibilities of application beyond graphical representations. An example of this is the code below, for a <em>multiple linear fit</em> for a situation with 4 predictor variables.</div>
<p><a href="Eq/jsp_multilin4Var.html" target="_blank"><img src="Eq/multilin4Var.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-24" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-24">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try varying the number of <span class="fu">predictors</span> (xi). To do this<span class="sc">:</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  a. If you want to reduce, just remove the desired <span class="fu">vector</span>(s), and correct their quantity <span class="cf">in</span> the line<span class="sc">:</span> <span class="st">"const X = x1.map((_, i)"</span>;</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  b. If you want to increase, add a new vector <span class="cf">in</span> the corresponding line, and update the quantity <span class="cf">in</span> the mapping of <span class="st">"const X = x1.map((_, i)"</span>.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="response-surface-methodology-rsm" class="level3" data-number="3.7">
<h3 data-number="3.7" class="anchored" data-anchor-id="response-surface-methodology-rsm"><span class="header-section-number">3.7</span> Response Surface Methodology (RSM)</h3>
<section id="context-22" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-22">Context</h4>
<div class="line-block">&nbsp;&nbsp;A variation of multiple linear fitting involves <em>response surface methodology</em>, commonly linear (plane) or quadratic (curvilinear), applied in modeling and optimization of experimental processes with multiple independent variables. Its goal is to fit a mathematical equation that represents the relationship between experimental factors and the response variable, allowing prediction of behaviors and identification of optimal conditions.</div>
<div class="line-block">&nbsp;&nbsp;For experiments with two factors, a second-order quadratic model is commonly used, which allows representing surfaces with curvature. The central idea of the experiment is to select values and their codes (-1,0,1), representing, respectively, a quantity below, a medium quantity, and one above, for a given predictor variable.</div>
</section>
<section id="equation-16" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-16">Equation:</h4>
<div class="line-block">&nbsp;&nbsp;The general quadratic model for 2 factors is represented by:</div>
<p><span class="math display">\[
y = \beta_0 + \beta_1 x_1 + \beta_2 x_2 + \beta_{11} x_1^2 + \beta_{22} x_2^2 + \beta_{12} x_1 x_2 + \varepsilon
\]</span></p>
<div class="line-block">&nbsp;&nbsp;With the matrix form of the model represented by:</div>
<p><span class="math display">\[
\mathbf{y} = \mathbf{X} \boldsymbol{\beta} + \boldsymbol{\varepsilon}
\]</span></p>
<div class="line-block">&nbsp;&nbsp;And estimation of the coefficients <span class="math inline">\(\beta\)</span> by least squares, similarly to multiple linear fitting, by:</div>
<p><span class="math display">\[
\hat{\boldsymbol{\beta}} = (\mathbf{X}^\top \mathbf{X})^{-1} \mathbf{X}^\top \mathbf{y}
\]</span><br>
</p>
<p><a href="Eq/jsp_msr.html" target="_blank"><img src="Eq/msr.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-25" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-25">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> As suggested earlier, try using data from the literature or another source whose responses are already known, replacing the respective vectors at the beginning of the code. This allows comparing the effectiveness of using the presented tool.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="data-smoothing---spline-and-savitzky-golay-filter" class="level3" data-number="3.8">
<h3 data-number="3.8" class="anchored" data-anchor-id="data-smoothing---spline-and-savitzky-golay-filter"><span class="header-section-number">3.8</span> Data smoothing - Spline and Savitzky-Golay filter</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;In experimental research, signal detection occasionally occurs in the presence of significant levels of noise, as happens in continuous spectral scanning under variation of wavelength, or in electrical signal detection over time. In this case, there is no analytical equation that describes the behavior of the signal, although some digital processing tools can produce a trend curve in data treatment, significantly reducing the noise level, such as by a <em>spline curve</em> and <em>Savitzky-Golay filter</em>.</div>
<section id="context---cubic-spline" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context---cubic-spline">Context - Cubic spline</h4>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A very common interpolation technique refers to using cubic spline by degree-3 polynomials interconnecting data windows. The treatment consists of applying the polynomial functions in subintervals of the points, promoting smoothing between the data windows.</div>
</section>
<section id="equation-17" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-17">Equation</h4>
<p>Mathematically, the spline is applied such that:</p>
<p><span class="math display">\[
S_i(x) = a_i + b_i(x - x_i) + c_i(x - x_i)^2 + d_i(x - x_i)^3
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>S<span class="math inline">\(_{i}\)</span>(x) = cubic spline function in the interval <em>[xi,xi+1]</em>;</li>
<li>a<span class="math inline">\(*{i}\)</span>, b<span class="math inline">\(*{i}\)</span>, c<span class="math inline">\(*{i}\)</span>, d<span class="math inline">\(*{i}\)</span> = coefficients specific to segment <em>i</em>;</li>
<li>x<span class="math inline">\(_{i}\)</span> = starting point of the interval;</li>
<li><em>x</em> = independent variable.</li>
</ul>
</section>
<section id="context---savitzky-golay-filter" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context---savitzky-golay-filter">Context - Savitzky-Golay filter</h4>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The Savitzky-Golay smoothing technique applies a local polynomial regression to a moving window, by least squares and a predetermined degree, preserving signals of peaks, valleys, and bands. Unlike cubic spline, the filter produces a global continuous curve, not a set of interconnected curves.</div>
</section>
<section id="equation-18" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-18">Equation</h4>
<p>Mathematically:</p>
<p><span class="math display">\[
\tilde{y}*i = \sum*{j=-m}^{m} c_j \cdot y_{i+j}
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>y<span class="math inline">\(_{i}\)</span> = smoothed value at position <em>i</em>;</li>
<li>y<span class="math inline">\(_{i+j}\)</span> = real series values within the window;</li>
<li>c<span class="math inline">\(_{i}\)</span> = filter coefficients derived from a polynomial regression;</li>
<li><em>m</em> = number of points on each side of the central window.</li>
</ul>
<div class="line-block">&nbsp;&nbsp;Below is an example of code for <em>JSplotly</em> that illustrates the use of both interpolation methods.</div>
<p><a href="Eq/jsp_splineGolay.html" target="_blank"><img src="Eq/splineGolay.png" class="img-fluid"></a><br>
</p>
<p><strong>Instructions:</strong></p>
<ul>
<li><ol type="1">
<li>The code has two <em>true/false flags</em>, one for “usarGolay” and another for “usarSpline”;</li>
</ol></li>
<li><ol start="2" type="1">
<li>View the raw data adopting <em>false</em> for both flags;</li>
</ol></li>
<li><ol start="3" type="1">
<li>To overlay a cubic spline, change its constant to <em>true</em>;</li>
</ol></li>
<li><ol start="4" type="1">
<li>To overlay the Savitzky-Golay filter, change its constant to <em>true</em>;</li>
</ol></li>
<li><ol start="5" type="1">
<li>To tune the filter, modify its parameters in the constants <em>janela</em> (moving window) and/or <em>grau</em> (polynomial degree).</li>
</ol></li>
</ul>
</section>
<section id="suggestion-26" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-26">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Compare the smoothing effects of the two polynomial interpolation treatments;</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Apply only the filter, tuning moving<span class="sc">-</span>window and polynomial<span class="sc">-</span>degree parameters;</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Try interpolation with other raw data <span class="cf">in</span> the code, replacing the constants <span class="st">"x_values"</span> and <span class="st">"y_raw"</span> with numeric vectors;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="anova-and-tukey-hsd-test" class="level3" data-number="3.9">
<h3 data-number="3.9" class="anchored" data-anchor-id="anova-and-tukey-hsd-test"><span class="header-section-number">3.9</span> ANOVA and Tukey-HSD test</h3>
<section id="context-23" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-23">Context:</h4>
<div class="line-block">&nbsp;&nbsp;In research, it is common to analyze a dataset by comparing the variance between groups with that obtained within each group, i.e., <em>Analysis of Variance (ANOVA)</em>. Once the statistical information is obtained that there is a difference between groups greater than the internal variance, a <em>post-hoc</em> test is performed for comparison of means, such as the <em>Tukey-HSD</em> (<em>Honest Significant Difference</em>) test.</div>
</section>
<section id="equation-19" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-19">Equation:</h4>
<div class="line-block">&nbsp;&nbsp;The equation that characterizes a Tukey-HSD test is provided below (correction for unbalanced groups):</div>
<p><span class="math display">\[
\text{HSD}*{ij} = q*{\alpha} \cdot \sqrt{\frac{\text{MS}_{\text{within}}}{2} \left( \frac{1}{n_i} + \frac{1}{n_j} \right)}
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>HSD = minimum value of the difference between two means for it to be considered significant;</li>
<li>q<span class="math inline">\(_{\alpha}\)</span> = critical value of the studentized <em>q</em> distribution (Tukey distribution), depending on the number of groups and the significance level <span class="math inline">\(\alpha\)</span>;</li>
<li>MS<span class="math inline">\(_{within}\)</span> = mean square within groups (error) obtained from ANOVA;</li>
<li>n: number of observations per group (<em>i</em> and <em>j</em> if there are groups of different sizes).<br>
</li>
</ul>
<p><a href="Eq/jsp_tukey.html" target="_blank"><img src="Eq/tukey.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-27" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-27">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> You can click and drag the minimum<span class="sc">-</span>difference <span class="fu">markers</span> (a,b,etc) above the plot <span class="cf">for</span> better visual adjustment, as well as the ANOVA table and the statistical test result;</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Change the vector values, redo the analysis and the plot simultaneously, and notice the difference <span class="cf">in</span> the obtained values. Note that, as stated on the lower border, it is necessary to refresh the site page to clear previous data retained <span class="cf">in</span> cache memory;</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Enter other data <span class="cf">in</span> the respective vectors, yours or from another source;</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Insert new data vectors, or delete one, <span class="cf">for</span> a customized calculation and plot.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="nonlinear-fitting-of-data" class="level3" data-number="3.10">
<h3 data-number="3.10" class="anchored" data-anchor-id="nonlinear-fitting-of-data"><span class="header-section-number">3.10</span> Nonlinear fitting of data</h3>
<div class="line-block">&nbsp;&nbsp;Modeling is an area of common use in the natural sciences, when one wants prediction or mathematical correlation of experimental data. Models may be linear or nonlinear. When nonlinear, they may be applied empirically to the data (e.g., quadratic, logarithmic, exponential, or power functions), or based on an explanatory mathematical function of the model.</div>
<div class="line-block">&nbsp;&nbsp;In this case, techniques are used for nonlinear fitting of the data to the model, such as <em>Simplex</em>, <em>Gauss</em>, <em>Newton-Raphson</em>, <em>Levenberg-Marquadt</em>, or <em>least squares</em>, for example. Whatever the method, nonlinear fitting differs from linear fitting in assumptions and data treatment (see a quick explanation <a href="https://bioquanti.netlify.app/pt/nivel/superior/r/enzimas#ajuste-n%C3%A3o-linear">here</a>.</div>
<section id="context-ligand-protein-interaction-grid-search-algorithm" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-ligand-protein-interaction-grid-search-algorithm">Context: Ligand-protein interaction (<em>grid search</em> algorithm)</h4>
<div class="line-block">&nbsp;&nbsp;The following example illustrates applying a nonlinear fit to molecular interaction data by a Langmuir model, and treating it by <em>minimizing squared error by manual scanning</em>. In this treatment, the <em>sum of squared errors (RMSE)</em> is computed for combinations of <em>a</em> and <em>b</em>, selecting the pair with the lowest computed error. In this case, for simplification, although <em>JSPlotly</em> works with more specific libraries, the nonlinear regression is done by least squares, but performed by <em>grid search</em>, not by classic iterative methods. Thus, a set of values is tested for the pair (a, b) on a previously defined grid (combination table), allowing identification of the parameters that best describe the experimental data according to the minimum-error criterion.</div>
<p><br>
</p>
</section>
<section id="equation-20" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-20">Equation:</h4>
<div class="line-block">&nbsp;&nbsp;The Langmuir equation is represented below:</div>
<p><span class="math display">\[
Y = \frac{aX}{K_d + X}
\]</span> <em>Where:</em></p>
<ul>
<li>Y = fraction of occupied sites;</li>
<li>X = concentration of free ligand;</li>
<li><em>a</em> = saturation value (sites of the same affinity fully occupied);</li>
<li><em>Kd</em> = equilibrium dissociation constant of the complex.</li>
</ul>
<div class="line-block">&nbsp;&nbsp;In this case, the nonlinear least-squares function is:</div>
<p><span class="math display">\[
SSR(a, K_d) = \sum_{i=1}^n \left( y_i - \frac{a x_i}{K_d + x_i} \right)^2
\]</span></p>
<p><em>Where:</em></p>
<p><em>SSR</em> = sum of squared residuals (inversely proportional to fit quality)<br>
</p>
<p><a href="Eq/jsp_ajNonLin.html" target="_blank"><img src="Eq/ajNonLin.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-28" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-28">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> As <span class="cf">in</span> the previous example, alternatively show the points and the fitted curve using the boolean <span class="fu">operation</span> (<span class="st">"mostrarPontos = true/false"</span>; <span class="st">"mostrarCurva = true/false"</span>);</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Change the data and perform a new fit, to obtain other equation <span class="fu">parameters</span> (Kd, SSR, R², adjusted R²). Note<span class="sc">:</span> adjusted R² corresponds to the R² value corrected <span class="cf">for</span> the number of model <span class="fu">parameters</span> (<span class="cf">in</span> this case, <span class="st">"2"</span>), directly affecting the <span class="st">"degrees of freedom"</span> <span class="cf">for</span> the fit.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
<section id="context---competitive-inhibition-in-enzyme-kinetics-gauss-newton-algorithm" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context---competitive-inhibition-in-enzyme-kinetics-gauss-newton-algorithm">Context - Competitive inhibition in enzyme kinetics (Gauss-Newton algorithm)</h4>
<div class="line-block">&nbsp;&nbsp;The <em>Gauss-Newton</em> method iteratively fits nonlinear models to experimental datasets. It is especially efficient when the model is <em>almost linear</em> with respect to the parameters and the residuals are moderately small. The goal is to find the parameters <span class="math inline">\(\theta\)</span> that minimize the sum of squared residuals between the observed data <em>yi</em> and the predicted values by the model f(xi,<span class="math inline">\(\theta\)</span>. Unlike the full Newton-Raphson method, Gauss-Newton avoids explicit calculation of the Hessian, using an approximation based on the Jacobian of the residuals.</div>
</section>
<section id="equation-21" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-21">Equation:</h4>
<div class="line-block">&nbsp;&nbsp;This approximation results in a linear system that can be solved at each iteration to update the model parameters. The central formula of the method consists of multiplying the transposed Jacobian <em>JT</em> by the Jacobian itself <em>J</em>, forming a symmetric matrix that approximates the Hessian. The residual vector <em>r</em> is then used to determine the update direction <span class="math inline">\(\Delta\theta\)</span>, with the following equation:</div>
<p><span class="math display">\[
\theta_{k+1} = \theta_k + \left(J^T J\right)^{-1} J^T \cdot r
\]</span></p>
<div class="line-block">&nbsp;&nbsp;To illustrate the potential of the app in iterative nonlinear fitting, an example follows for enzyme kinetics using a <em>competitive inhibition model</em>.</div>
<div class="line-block">&nbsp;&nbsp;The code allows fitting the <em>seeds</em>, <em>tolerance</em>, and number of <em>iterations</em>, and provides the plot (points/fitted line, manually), and statistical regression parameters.</div>
<p><br>
</p>
<p><strong>Instructions</strong></p>
<ul>
<li><ol type="1">
<li>Notice that the code has a <em>true/false</em> flag in the constant <em>ajuste</em>. Leave it <em>false</em> for representing the data only (vectors <em>x</em> and <em>y</em>, or <em>S</em> and <em>v</em>);</li>
</ol></li>
<li><ol start="2" type="1">
<li>In the code’s <em>Plot</em> area, set the constant <em>trace</em> so that it uses the data vectors as well;</li>
</ol></li>
<li><ol start="3" type="1">
<li>Click <em>add plot</em> to view the experimental points;</li>
</ol></li>
<li><ol start="4" type="1">
<li>To overlay the fitted curve, update the constants <em>ajuste</em> and <em>trace</em> (vectors <em>S_fit</em> and <em>v_fit</em>), and choose <em>lines</em> in <em>mode</em>.</li>
</ol></li>
</ul>
<p><a href="Eq/jsp_ajNonLinGauss3.html" target="_blank"><img src="Eq/ajNonLinGauss2.png" class="img-fluid"></a><br>
</p>
</section>
</section>
</section>
<section id="applications" class="level2" data-number="4">
<h2 data-number="4" class="anchored" data-anchor-id="applications"><span class="header-section-number">4</span> Applications</h2>
<section id="prot.id---simulated-experimentation-for-characterization-of-amino-acids-and-chains" class="level3" data-number="4.1">
<h3 data-number="4.1" class="anchored" data-anchor-id="prot.id---simulated-experimentation-for-characterization-of-amino-acids-and-chains"><span class="header-section-number">4.1</span> prot.Id - Simulated experimentation for characterization of amino acids and chains</h3>
<section id="context-24" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-24">Context:</h4>
<div class="line-block">&nbsp;&nbsp;<em>JSPlotly</em> makes it possible to work with a set of interactions not necessarily tied to equations and their graphical representations. The example below illustrates its potential for simulating a classic Biochemistry experiment: <em>the characterization of amino acids, peptides, and proteins through chromogenic reactions in test tubes</em>. Since the scope of this material is to present and enable the use of varied <em>OIERs</em> (<em>interactive objects for reproducible teaching</em>), users are encouraged to seek information sources related to the reactions briefly described below.</div>
</section>
<section id="simulator-instructions" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="simulator-instructions">Simulator instructions</h4>
<div class="line-block">&nbsp;&nbsp;The simulator offers two options: a) discover the nature of a sample or b) discover the reaction color pattern for a selected sample.</div>
<p><br>
</p>
<p><strong>Option A: discovering the nature of a sample:</strong></p>
<div class="line-block">&nbsp;&nbsp;Here it is possible to <em>“play”</em> to check whether you are facing a protein/peptide (chain) sample, an amino acid (AA), a specific mixture of AAs, or AA(s) present in a chain. To do that:</div>
<ol type="1">
<li><p>Click the image below and press <em>“add”</em>. A set of 7 colored bars will be generated, representing the colors obtained in test tubes for 7 distinct reactions used for protein characterization: ninhydrin, biuret, Pauly, xanthoproteic, Sakaguchi, Millon, and sulfur (Cys);</p></li>
<li><p>Determine the sample type based on the color pattern presented, making a guess. The guess must be by type (in the <em>“Guess (type)”</em> menu—AA, chain, or AA mixture) and by profile (in the <em>“Guess (profile)”</em> menu—random, specific AA(s) or specific mixture);</p></li>
<li><p>After selecting both menus, click <em>“Check”</em>. If correct, a <em>“Scoreboard”</em> right below will count the number of correct guesses/number of samples <em>“played”</em>;</p></li>
<li><p>If you miss, try again or look for the explanation of the color pattern in <em>“Explain”</em>;</p></li>
</ol>
<ol start="5" type="1">
<li><p>There are two other possibilities: <em>“Hard mode”</em>, which hides the reaction names while keeping their sequence, and <em>“Shuffle”</em>, which changes the order of the tubes;</p></li>
<li><p>For a new sample, just click <em>“New sample”</em>;</p></li>
</ol>
<p><strong>Option B: discovering the reaction color pattern for a selected sample:</strong></p>
<ol type="1">
<li>Select a sample in <em>“Generate”</em>, and click <em>“New sample”</em>. A set of tubes will be generated with the expected reaction pattern for that sample.</li>
</ol>
<p><a href="Eq/jsp_Id-Prot-7.html" target="_blank"><img src="Eq/Id-Prot-7.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-29" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-29">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try clicking <span class="st">"New sample"</span> to see how often possibilities occur;</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try both operating options of the <span class="st">"game"</span> (yes, it is a game...after all, it has a scoreboard<span class="sc">!</span>);</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Try <span class="st">"hard mode"</span> (without reaction labels <span class="cf">in</span> sequence) or <span class="st">"shuffle"</span> (permutes the reaction order). Obviously, however, avoid using both at the same time<span class="sc">!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="zemol---3d-viewer-for-molecular-models" class="level3" data-number="4.2">
<h3 data-number="4.2" class="anchored" data-anchor-id="zemol---3d-viewer-for-molecular-models"><span class="header-section-number">4.2</span> ZeMol - 3D viewer for molecular models</h3>
<section id="context-25" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-25">Context:</h4>
<div class="line-block">&nbsp;&nbsp;Due to the combined characteristics of using <em>JavaScript</em> and the <em>Plotly.js</em> library, it is possible to extend the potential applications of <em>JSPlotly</em> to the construction of true HTML applications. A clear example of this reach is shown below: a molecular viewer similar in nature to some pre-existing tools used in teaching and research (<em>Jmol</em>, <em>Pymol</em>, etc.).</div>
<div class="line-block">&nbsp;&nbsp;Although it is obviously much smaller in functionality compared to other applications, the so-called <em>“ZeMol”</em> uses less than 70 kB of memory and provides features such as:</div>
<ul>
<li>models loaded online from the <a href="https://www.rcsb.org/">PDB</a> or <a href="https://pubchem.ncbi.nlm.nih.gov/">PubChem</a> sites;</li>
<li>models loaded from the device’s physical storage (PDB, MOL, SDF, XYZ, TXT formats);</li>
<li>identification information for the PDB file (or PubChem file name);</li>
<li>header containing counts of chains, ligands, ions, and disulfide bonds;</li>
<li>identification of only the <span class="math inline">\(\alpha\)</span> carbons of each residue in the polypeptide backbone, facilitating model visualization;</li>
<li>automatic visualization of disulfide bonds;</li>
<li>identification of residues, ligands, and ions by mouse hover over the model;</li>
<li>checkbox to show protein secondary structure (helix, <span class="math inline">\(\beta\)</span>-sheet);</li>
<li>checkbox for residue polarity index;</li>
<li>sliders for CA carbon size (PDB) or atoms in general, and model opacity (useful to highlight clefts, secondary structure, carbon chain, ligands, ions);</li>
<li>hide/show distinct parts of the model by clicking its legend (chain, ions, helices/sheets, ligands);</li>
<li>highlighting amino acids (single, range, all of one type, predefined groups in the <em>script</em>);</li>
<li>self-contained HTML when saving, <em>“freezing”</em> the model in the characteristics you want to present;</li>
<li><em>Zoom</em> on mobile devices (one finger fixed on the screen and another dragging the model) and <em>Pan</em> (moving the model with two fingers on the screen)—actions also present in the icons above the model;</li>
<li>optional adjustments in the <em>script</em> itself (sizes, colors, thicknesses, AA groups, line type and thickness, for example).</li>
</ul>
<p><a href="Eq/ZeMol.html" target="_blank"><img src="Eq/ZeMol.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-30" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-30">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Load a model with a PDB code, or a molecule from the PubChem site;</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> If it is a protein, try the separate and combined effects of the sliders <span class="cf">for</span> CA size and opacity;</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Observe residue polarity <span class="cf">in</span> the polarity checkbox;</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Try zooming <span class="cf">in</span><span class="sc">/</span>out or moving it on the screen with the mouse, or doing it on a smartphone as instructed above; </span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="fl">5.</span> Show or hide each chain or all chains, ligands, and ions by clicking the corresponding legend terms;</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="fl">6.</span> Select residues of interest <span class="cf">in</span> the model, such as a catalytic site, coenzyme<span class="sc">-</span>binding site, or regulatory residues. To do this, type <span class="cf">in</span> the corresponding field and click <span class="st">"highlight"</span>. Some suggestions<span class="sc">:</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  a. Single<span class="sc">:</span> Arg15, ASP117, <span class="dv">15</span>, A<span class="sc">:</span><span class="dv">15</span>, A<span class="sc">:</span>ARG15; </span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>  b. Ranges<span class="sc">:</span> <span class="dv">15-30</span>, A<span class="sc">:</span><span class="dv">15-30</span>, Arg15<span class="sc">-</span>Asp30; </span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>  c. Multiple separated by commas<span class="sc">:</span> <span class="st">"Arg15, Asp117"</span>;</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>  d. Groups<span class="sc">:</span> <span class="st">"aromatico"</span>, <span class="st">"polar"</span>, <span class="st">"apolar"</span>, <span class="st">"basico"</span>, <span class="st">"acido"</span>, <span class="st">"small"</span>.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="carbo.id---simulated-experimentation-for-carbohydrate-characterization" class="level3" data-number="4.3">
<h3 data-number="4.3" class="anchored" data-anchor-id="carbo.id---simulated-experimentation-for-carbohydrate-characterization"><span class="header-section-number">4.3</span> carbo.Id - Simulated experimentation for carbohydrate characterization</h3>
<section id="context-26" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-26">Context:</h4>
<div class="line-block">&nbsp;&nbsp;As presented for <em>amino acid and chain characterization</em> above, here is a simulated experimentation for carbohydrate characterization. The logic is the same: given the chromogenic results for specific reactions in test tubes (bar chart), the user tries to characterize the carbohydrate. This is done with respect to the reactions of:</div>
<ol type="1">
<li><p>Molisch test – detects carbohydrates (purple ring); reaction with <span class="math inline">\(\alpha\)</span>-naphthol producing furfural or hydroxymethylfurfural;</p></li>
<li><p>Lugol test (I<span class="math inline">\(_{2}\)</span>/KI) – detects polysaccharides (dark blue tube);</p></li>
<li><p>Benedict test – detects reducing monosaccharides (and also lactose); tube with dark orange/brick precipitate;</p></li>
<li><p>Seliwanoff test (resorcinol) – differentiates ketoses (red tube) from aldoses (faint pink tube);</p></li>
<li><p>Barfoed test – differentiates mono (tube with red precipitate – cuprous oxide) from oligosaccharides (yellowish tube).</p></li>
</ol>
</section>
<section id="simulator-instructions-1" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="simulator-instructions-1">Simulator instructions</h4>
<p><strong>Option A: discovering the nature of a sample:</strong></p>
<div class="line-block">&nbsp;&nbsp;To discover the nature of a sample, proceed as in the <em>amino acid and chain characterization</em> app above. To do that:</div>
<ol type="1">
<li><p>Click the image below and press <em>“add”</em>. A set of 5 colored bars will be generated, representing the colors obtained in test tubes for the 5 distinct reactions used for carbohydrate characterization: Molisch, Lugol, Benedict, Seliwanoff, and Barfoed.</p></li>
<li><p>Determine the sample type based on the tube color pattern presented, making a guess. The guess must be by type (in the <em>“Guess (type)”</em> menu—monosaccharide, di-oligosaccharide, polysaccharide, water) and by profile (in the <em>“Guess (profile)”</em> menu—reducing, non-reducing, aldose, ketose, monosaccharide, di-oligosaccharide, and polysaccharide);</p></li>
<li><p>After selecting both menus, click <em>“Check”</em>. If correct, a <em>“Scoreboard”</em> right below will count the number of correct guesses/number of samples <em>“played”</em>;</p></li>
<li><p>If you miss, try again or look for the explanation of the color pattern in <em>“Explain”</em>;</p></li>
</ol>
<ol start="5" type="1">
<li><p>There are two other possibilities: <em>“Hard mode”</em>, which hides the reaction names while keeping their sequence, and <em>“Shuffle”</em>, which changes the order of the tubes;</p></li>
<li><p>For a new sample, just click <em>“New sample”</em>;</p></li>
</ol>
<p><strong>Option B: discovering the reaction color pattern for a selected sample:</strong></p>
<ol type="1">
<li>Select a sample in <em>“Generate”</em>, and click <em>“New sample”</em>. A set of tubes will be generated with the expected reaction pattern for that sample.</li>
</ol>
<p><a href="Eq/jsp_Id-CHO.html" target="_blank"><img src="Eq/Id_CHO.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-31" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-31">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try clicking <span class="st">"New sample"</span> to see how often possibilities occur;</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try both operating options of the <span class="st">"simulator/app/game"</span><span class="er">)</span>;</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Try <span class="st">"hard mode"</span> (without reaction labels <span class="cf">in</span> sequence) or <span class="st">"shuffle"</span> (permutes the reaction order). Obviously, however, avoid using both at the same time<span class="sc">!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="flowforces" class="level3" data-number="4.4">
<h3 data-number="4.4" class="anchored" data-anchor-id="flowforces"><span class="header-section-number">4.4</span> FlowForces</h3>
<section id="context-27" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-27">Context:</h4>
<div class="line-block">&nbsp;&nbsp;The name above alludes to the Nobel laureate Ilya Prigogine and self-organizing processes, such as those found in living cells. In this case, the quantities of certain reactants and products configure the forces, while their transformations point to the flows.</div>
<div class="line-block">&nbsp;&nbsp;With respect to both, it is sometimes interesting to study the effect of a modifier, such as an activator or inhibitor, in a metabolic pathway or any biochemical reaction—both in isolation and in an organelle, cell, or even the organism. In this sense, the application simulates such effects in a biological system where the user enters metabolites into a continuous logging flow for an arbitrary signal.</div>
<div class="line-block">&nbsp;&nbsp;For example, the addition of metabolic modifiers to a cell suspension can be monitored by the oxygen level of the sample, as measured by an <em>oxygraph</em>, and simulated below.</div>
<p><br>
</p>
<p><a href="Eq/jsp_flowforces.html" target="_blank"><img src="Eq/flowforces.png" class="img-fluid"></a><br>
</p>
</section>
<section id="suggestion-32" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-32">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try changing one or more compounds with biochemical justification, <span class="cf">for</span> example <span class="cf">for</span> a metabolic pathway;</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Test variations of <span class="st">"delay"</span> <span class="cf">for</span> the appearance of a modifier effect by clicking the <span class="st">"Advanced"</span> checkbox;</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Try increasing the noise <span class="cf">in</span> <span class="st">"const noise"</span>, raising it <span class="dv">100</span>x<span class="sc">:</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"const noise = noiseOn ? (Math.random()-0.5)*20"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="gizz---digital-chalkboard" class="level3" data-number="4.5">
<h3 data-number="4.5" class="anchored" data-anchor-id="gizz---digital-chalkboard"><span class="header-section-number">4.5</span> Gizz - Digital chalkboard</h3>
<section id="context-28" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-28">Context:</h4>
<div class="line-block">&nbsp;&nbsp;The nature of the object below needs no introduction. It is yet another among many applications that simulate a chalkboard or digital whiteboard. Unlike its competitors, however:</div>
<ol type="1">
<li>It can be exported as a self-contained HTML file via a button on the board itself, allowing sharing of the generated image and continued editing;</li>
<li>It allows customization of the code that produces it, making it possible to insert/change pens, thicknesses, and different actions not provided in the source code.<br>
</li>
</ol>
<p><a href="Eq/lousa.html" target="_blank"><img src="Eq/lousa.png" class="img-fluid"></a><br>
</p>
<div class="line-block">&nbsp;&nbsp;In terms of its features, the digital board allows:</div>
<ul>
<li>Writing with 7 colors and 3 different thicknesses;</li>
<li>Accessing 4 background planes with different coloration;</li>
<li>Overlaying the background with a <em>grid</em>;</li>
<li>Using an eraser (<em>erase</em>);</li>
<li>Using a <em>laser pointer</em> to transiently locate what you want to point out;</li>
<li>Saving the image as PNG, indicating date and time;</li>
<li>Commands for undo (<em>undo</em>) and cleaning (<em>clean</em>) the board;</li>
<li>Keyboard shortcuts: E=eraser; L=laser; Z undo, X clear, H save HTML</li>
</ul>
</section>
<section id="suggestion-33" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-33">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> You can change pen thickness <span class="cf">in</span> <span class="st">"var baseWidth = 1;"</span> ;</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> The <span class="dv">1</span>x, <span class="dv">2</span>x, <span class="dv">4</span>x buttons are multipliers; edit the button creation lines to change label<span class="sc">/</span><span class="fu">multiplier</span> (e.g., <span class="st">"addSize('3x', 3, true)"</span>).</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
</section>
<section id="toolkits" class="level2" data-number="5">
<h2 data-number="5" class="anchored" data-anchor-id="toolkits"><span class="header-section-number">5</span> Toolkits</h2>
<div class="line-block">&nbsp;&nbsp;<em>JSPlotly</em> can operate as a complete application for simulation in varied situations. The <em>“toolboxes”</em> below illustrate interactive panels (<em>dashboards</em>) for <em>simulated experimentation and analysis</em> in biophysical chemistry, focusing on themes of <em>kinetics, interaction, and stability for protein targets</em>. Each <em>toolkit</em> allows an appreciation of methods used, applications, modeling, and parameter interpretation. As a whole, the <em>toolkits</em> enable an integrated study of kinetics and thermodynamics for application in biotechnology, biocatalysis, and drug discovery.</div>
<p><strong>Important notes</strong>:</p>
<ol type="1">
<li><p>The <em>Toolkits</em> contain internal code for styles and panel assembly that can interfere with the <em>Bootstrap</em> of the web page builder used for <em>Bioquanti</em> (the <em>Quarto</em> package for <em>R</em>). In this sense, the clickable images for each panel point to the individual <em>scripts</em> for loading each <em>Toolkit</em> into <em>JSPlotly</em>.</p></li>
<li><p>Each panel is displayed below the JSPlotly plotting screen. This is necessary to maximize the area available for displaying the panels.<br>
</p></li>
</ol>
<p>Individually, the toolkits include:</p>
<ol type="1">
<li><strong>EK-Toolkit</strong> - modeling in enzyme kinetics and inhibition;</li>
<li><strong>LeadHunt</strong> - lead-compound prospecting via enzyme inhibition;</li>
<li><strong>LB-Toolkit</strong> - analysis of ligand–protein interaction;</li>
<li><strong>GH-Toolkit</strong> - thermodynamic stability of biopolymers;</li>
<li><strong>mTherm-Toolkit</strong> - stability under perturbing conditions;</li>
<li><strong>ThermStab-Toolkit</strong> - enzyme thermostability;</li>
<li><strong>Biocel-Toolkit</strong> - microbial fuel cell performance</li>
</ol>
<section id="ek-toolkit" class="level3" data-number="5.1">
<h3 data-number="5.1" class="anchored" data-anchor-id="ek-toolkit"><span class="header-section-number">5.1</span> EK-Toolkit</h3>
<p><strong>General instructions</strong></p>
<div class="line-block">&nbsp;&nbsp;The <em>EK-Toolkit</em> is a toolbox for simulating enzyme kinetics and inhibition using diagnostic plots. The kit generates three graphs side by side:</div>
<ol type="1">
<li>Michaelis–Menten: v × [S];</li>
<li>Lineweaver–Burk (LB) — 1/v vs 1/[S];</li>
<li>Replot - LB <em>slope</em> or <em>intercept</em>, or Dixon ordinate (1/v)</li>
</ol>
<div class="line-block">&nbsp;&nbsp;Additional features include <em>HTML</em> export, <em>CSV</em> export and load, editable annotations, draggable windows, and model comparison (<em>A/B</em>).</div>
<p><br>
</p>
<p><a href="Eq/toolkit/ek-toolkit.js" download="ek-toolkit.js"><img src="Eq/toolkit/ek-toolkit.png" class="img-fluid"></a></p>
<p><strong>Features</strong></p>
<ul>
<li><em>Models:</em> Michaelis–Menten; Substrate excess; Irreversible inhibition; Competitive inhibition; Non-competitive inhibition (pure); Mixed inhibition; Uncompetitive inhibition; Bi-substrate (Random Sequential, Ordered Sequential, Ping-Pong); Hill (cooperativity); Enzyme activation (Vmax-up); <em>User</em> (free equation);</li>
<li><em>Synchronized plots:</em> v×[S], LB (1/v vs 1/[S]) and Replot (slope, intercept or Dixon 1/v vs modifier);</li>
<li><em>A/B comparison</em> with consistent color palettes across plots;</li>
<li><em>Panels:</em> models, equations, parameters (range of [S], number of simulated points, error, modifiers, parameters;</li>
<li>Configurable <em>Replot</em> with editable <em>X-axis label</em>.</li>
<li><em>Fits:</em> weighted linear fit in LB and replots - WLS (<em>weighted least squares</em>, weight: 1/y²); OLS (<em>ordinary least squares</em>) in Dixon.</li>
<li><em>Ki estimation in Dixon</em>; vertical marker at (x = -K_i).</li>
<li><em>HTML export</em> with interactive plots and equations; <em>PT-BR CSV</em> (‘;’ separator, ‘,’ decimal).</li>
<li><em>CSV load</em> to reconstruct plots and diagnostics; <em>‘Clear CSV’</em> to return to simulation.</li>
<li><em>Draggable modals</em>; clickable/editable annotations; <em>global legend ON/OFF</em>.<br>
</li>
</ul>
<section id="models" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="models">Models</h4>
<p><strong>Michaelis–Menten</strong> <span class="math display">\[ v = \dfrac{V_{max} S}{K_m + S} \]</span></p>
<p><strong>Substrate excess</strong> <span class="math display">\[ v = \dfrac{V_{max} S}{K_m + S + \dfrac{S^2}{K_{is}}} \]</span></p>
<p><strong>Irreversible inhibition</strong> <span class="math display">\[ v = \dfrac{V_{max}(1-f) S}{K_m + S} \]</span></p>
<p><strong>Competitive</strong> <span class="math display">\[ v = \dfrac{V_{max} S}{K_m \left(1 + \dfrac{I}{K_i}\right) + S} \]</span></p>
<p><strong>Non-competitive (pure)</strong> <span class="math display">\[ v = \dfrac{V_{max}}{1 + \dfrac{I}{K_i}} \cdot \dfrac{S}{K_m + S} \]</span></p>
<p><strong>Mixed</strong> <span class="math display">\[ v = \dfrac{V_{max}}{1 + \dfrac{I}{K_{i2}}} \cdot \dfrac{S}{K_m \left(1 + \dfrac{I}{K_{i1}}\right) + S} \]</span></p>
<p><strong>Uncompetitive</strong> <span class="math display">\[ v = \dfrac{V_{max} S}{K_m + S\left(1 + \dfrac{I}{K_i}\right)} \]</span></p>
<p><strong>BiBi Random Sequential</strong> <span class="math display">\[ v = \dfrac{V_{max} A B}{K_a K_b + K_a B + K_b A + A B} \]</span></p>
<p><strong>BiBi Ordered Sequential</strong> <span class="math display">\[ v = \dfrac{V_{max} A B}{K_a K_b + K_b A + A B} \]</span></p>
<p><strong>BiBi Ping-Pong</strong> <span class="math display">\[ v = \dfrac{V_{max} A B}{K_b A + K_a B + A B} \]</span></p>
<p><strong>Hill (cooperativity)</strong> <span class="math display">\[ v = \dfrac{V_{max} S^n}{K^n + S^n} \]</span></p>
<p><strong>Activator (Vmax-up)</strong> <span class="math display">\[ v = \dfrac{V_{max}\left(1 + \dfrac{A}{K_{aA}}\right) S}{K_m + S} \]</span></p>
<p><em>Note:</em></p>
<ul>
<li><em>a (intercept):</em> value of 1/v when 1/[S] = 0.</li>
<li><em>b (slope):</em> LB line slope = (K_m/V_{max}).</li>
<li><em>K_i:</em> inhibition constant (Dixon: (x = -K_i)).</li>
</ul>
</section>
<section id="suggestion-34" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-34">Suggestion</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try comparing the pure non<span class="sc">-</span>competitive model to the mixed model; <span class="cf">for</span> this, simply set the inhibition constants <span class="fu">equal</span> (pure) or not <span class="fu">equal</span> (mixed);</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try a user<span class="sc">-</span>introduced <span class="fu">model</span> (<span class="st">"user"</span>). To do this, select Models <span class="sc">-</span><span class="ot">-&gt;</span> User, enter the equation, the parameters and their values. <span class="fu">Example</span> (partial uncompetitive activation)<span class="sc">:</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  Equation<span class="sc">:</span> (alfa<span class="sc">*</span>Vmax<span class="sc">*</span>S)<span class="sc">/</span>(Km<span class="sc">+</span>S<span class="sc">*</span>(<span class="dv">1</span><span class="sc">+</span>I<span class="sc">/</span>Ki))</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  Parameters<span class="sc">:</span> alfa;Km;Vmax;Ki</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  Values<span class="sc">:</span> <span class="fl">1.7</span>;<span class="dv">2</span>;<span class="dv">10</span>;<span class="dv">5</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Models → User, enter Equation (JS), Parameters (names;) and Values (;).</p>
<ol type="1">
<li>Marangoni, Alejandro G. Enzyme kinetics: a modern approach. John Wiley &amp; Sons, 2003.</li>
<li>Purich, Daniel L. Enzyme kinetics: catalysis and control: a reference of theory and best-practice methods. Elsevier, 2010.</li>
<li>Leone, Francisco A. Fundamentos de cinética enzimática. Editora Appris, 2021.</li>
<li>Segel, Irwin H. Enzyme kinetics: behavior and analysis of rapid equilibrium and steady state enzyme systems. Vol. 115. New York: Wiley, 1975.<br>
</li>
</ol>
</section>
</section>
<section id="leadhunt-toolkit" class="level3" data-number="5.2">
<h3 data-number="5.2" class="anchored" data-anchor-id="leadhunt-toolkit"><span class="header-section-number">5.2</span> LeadHunt-Toolkit</h3>
<div class="line-block">&nbsp;&nbsp;The example shown refers to the use of <em>JSPlotly</em> as a dashboard for simulated prospecting of the potential of a fictional screening molecule, aiming at selecting a lead compound as an inhibitor of an enzyme target.</div>
<p><br>
</p>
<p><a href="Eq/toolkit/lh-toolkit.js" download="lh-toolkit.js"><img src="Eq/toolkit/lh-toolkit.png" class="img-fluid"></a></p>
<p><strong>General instructions</strong></p>
<ol type="1">
<li>At the top of the app there are 4 buttons:</li>
</ol>
<ul>
<li><em>reset</em> - renews the simulation with new data and plots;</li>
<li><em>html</em> - saves the plots and data as an interactive file;</li>
<li><em>diagnostic</em> - opens a popup window with inhibition model diagnosis and parameter values;</li>
<li><em>parameters</em> - opens a window for simulation based on user parameters.</li>
</ul>
<ol start="2" type="1">
<li>There are two study modes for LeadHunt: entering kinetic and thermodynamic parameters to observe the outputs in graphs (<em>parameters</em> button), or characterizing parameters from the graphs (<em>diagnostic</em> button).</li>
</ol>
<div class="line-block">&nbsp;&nbsp;The plots simulate results for the <em>Michaelis–Menten, Lineweaver–Burk, dose–response curve, and van’t Hoff plot</em>.</div>
<p><br>
</p>
<section id="parameters" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="parameters">Parameters</h4>
<div class="line-block">&nbsp;&nbsp;Regardless of the mode, to identify the compound potential, the parameters are:</div>
<ul>
<li><em>Ki</em>, inhibitor dissociation equilibrium constant (desirable &lt; 10<span class="math inline">\(\mu\)</span>M);</li>
<li>Vmax, limiting enzyme reaction rate;</li>
<li>Km, Michaelis–Menten constant;</li>
<li><span class="math inline">\(\Delta\)</span>H, enthalpy change upon inhibitor binding;</li>
<li><span class="math inline">\(\Delta\)</span>S, entropy change upon inhibitor binding;</li>
<li><span class="math inline">\(\Delta\)</span>Cp, heat capacity change upon inhibitor binding;</li>
</ul>
<div class="line-block">&nbsp;&nbsp;In addition to these, the fictional molecule has some <em>descriptors</em>, namely:</div>
<ul>
<li>MW, molecular weight;</li>
<li>HAC, number of heavy atoms (all except hydrogen);</li>
<li>HBD, number of hydrogen bond donors;</li>
<li>HBA, number of hydrogen bond acceptors;</li>
<li>logP, logarithm of the partition coefficient</li>
</ul>
<div class="line-block">&nbsp;&nbsp;These descriptors allow computation of other criteria for lead selection:</div>
<ul>
<li><p>LE, ligand efficiency:</p>
<p><span class="math display">\[
LE = \frac{\Delta G}{HAC};
\]</span><br>
</p></li>
</ul>
<p><span class="math display">\[
LE = 1.37 \frac{pIC_{50}}{HAC}
\]</span><br>
</p>
<p><em>Where</em> pIC<span class="math inline">\(*{50}\)</span> = -log(IC<span class="math inline">\(*{50}\)</span>);</p>
<ul>
<li>LLE, lipophilic ligand efficiency (values &gt;5 are considered good),</li>
</ul>
<p><span class="math display">\[
LLE=pIC_{50}-logP
\]</span></p>
<ul>
<li>BEI, binding efficiency index,</li>
</ul>
<p><span class="math display">\[
BEI=\frac{pIC_{50}}{MW}
\]</span></p>
<p><em>Where:</em> MW, kDa; typical ranges by Lipinski’s “rule of five”: MW&lt;=500, HBD&lt;=5, HBA&lt;=10, cLogP&lt;=5. Also LE &gt;= 0.3; LLE &gt;= 5.</p>
<div class="line-block">&nbsp;&nbsp;The inhibitor dose–response curve also allows determination of:</div>
<ul>
<li>IC<span class="math inline">\(_{50}\)</span>, inhibitor concentration giving 50% response (adequate when &lt; 10<span class="math inline">\(\mu\)</span>M);</li>
<li>nH, Hill constant (midpoint slope; cooperativity index)<br>
</li>
</ul>
</section>
<section id="using-leadhunt-to-identify-a-lead-compound" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="using-leadhunt-to-identify-a-lead-compound">Using LeadHunt to identify a lead compound</h4>
<div class="line-block">&nbsp;&nbsp;From the visualization of plots and descriptors of the fictional molecule, it is possible to determine:</div>
<ol type="1">
<li><p>The type of reversible inhibition (competitive, uncompetitive, pure non-competitive, mixed non-competitive);</p></li>
<li><p>The <em>Ki</em> value by constructing various secondary plots from the parameter-entry table (<em>Ki table</em>)—plot inhibitor concentration <em>I</em> versus apparent Km, Vmax, or Km/Vm. Alternatively, plot <em>I</em> against <em>(<span class="math inline">\(\alpha\)</span>-1)</em>, where <span class="math inline">\(\alpha\)</span> represents <em>“1+I/Ki”</em>;</p></li>
<li><p>The IC<span class="math inline">\(_{50}\)</span> value and <em>nH</em> (midpoint slope) from the dose–response curve (cooperativity estimate);</p></li>
<li><p>The <span class="math inline">\(\Delta\)</span>H and <span class="math inline">\(\Delta\)</span>S values from the van’t Hoff plot. If the plot is curvilinear, also the <span class="math inline">\(\Delta\)</span>Cp value;</p></li>
<li><p>The <em>LE</em> and <em>LLE</em> values;</p></li>
<li><p>Evaluate the candidate against Lipinski’s rules.</p></li>
</ol>
<div class="line-block">&nbsp;&nbsp;Thus, <em>LeadHunt</em> can be used for a meaningful simulation of screening an inhibitor as a lead compound for an enzyme target.</div>
</section>
<section id="suggestion-35" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-35">Suggestion</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Compare enzyme inhibition types by visual inspection of the Michaelis–Menten and Lineweaver–Burk plots, emphasizing whether <span class="st">"Km"</span> and <span class="st">"Vmax"</span> do or do not change between them;</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Compute <span class="st">"Ki"</span> by different methods using the field <span class="cf">for</span> <span class="st">"replot"</span> by inhibitor <span class="fu">level</span> (table of <span class="st">"Apparent parameters"</span>)<span class="sc">:</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>    a<span class="er">)</span> by obtaining <span class="st">"1/Ki"</span> from the I <span class="fu">x</span> (alpha<span class="dv">-1</span>) <span class="fu">plot</span> (line slope) via the <span class="st">"alpha-replot"</span> button;</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>  b<span class="er">)</span> by intercept <span class="st">"-Ki"</span> from plots of I x Vmax_app, Km_app, or Km<span class="sc">/</span><span class="fu">Vm_app</span> (depending on the inhibition model);</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  c<span class="er">)</span> by <span class="st">"slope"</span> or <span class="st">"intercept"</span> from replots of Lineweaver–Burk against I</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Try the <span class="st">"parameters"</span> mode with an <span class="st">"nH"</span> value different from <span class="dv">1</span> (e.g., <span class="fl">2.8</span>) to evaluate dose–response curve behavior under cooperativity effects;</span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Try the <span class="st">"parameters"</span> mode with a non<span class="sc">-</span>zero heat capacity <span class="fu">value</span> (e.g., <span class="sc">-</span><span class="dv">2000</span>) to evaluate the possibility of secondary binding effects as a <span class="cf">function</span> of temperature, such as enzyme conformational transition;</span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a><span class="fl">5.</span> Test curvature of the van’t Hoff plot <span class="cf">for</span> a heat<span class="sc">-</span>capacity change by providing <span class="fu">positive</span> (e.g., <span class="dv">2000</span>) and negative <span class="fu">values</span> (e.g., <span class="sc">-</span><span class="dv">2000</span>)<span class="sc">:</span> <span class="cf">if</span> concave up, negative heat<span class="sc">-</span>capacity change; <span class="cf">if</span> concave down, positive heat<span class="sc">-</span>capacity change.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="references" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="references">References:</h4>
<ol type="1">
<li>Marangoni, Alejandro G. Enzyme kinetics: a modern approach. John Wiley &amp; Sons, 2003.</li>
<li>Copeland, Robert A. Evaluation of enzyme inhibitors in drug discovery: a guide for medicinal chemists and pharmacologists. John Wiley &amp; Sons, 2013.</li>
<li>Copeland, Robert A. Enzymes: a practical introduction to structure, mechanism, and data analysis. John Wiley &amp; Sons, 2023.</li>
<li>Leone, Francisco A. Fundamentos de cinética enzimática. Editora Appris, 2021.<br>
</li>
</ol>
</section>
</section>
<section id="lb-toolkit" class="level3" data-number="5.3">
<h3 data-number="5.3" class="anchored" data-anchor-id="lb-toolkit"><span class="header-section-number">5.3</span> LB-Toolkit</h3>
<p><strong>General instructions</strong></p>
<p>The <strong>LB-Toolkit</strong> is an interactive panel for analyzing ligand–receptor binding assays (protein, enzyme, nucleic acid, glycan). It allows visualization and fitting of classic binding curves (Binding, Scatchard, Benesi–Hildebrand, and Hill), and supports interactive HTML export of data and plots.</p>
<p><a href="Eq/toolkit/lb-toolkit.js" download="lb-toolkit.js"><img src="Eq/toolkit/lb-toolkit.png" class="img-fluid"></a></p>
<p><strong>Features</strong></p>
<div class="line-block">&nbsp;&nbsp;The <em>LB-Toolkit</em> enables study and graphical visualization of a 2 × 2 panel representing classic bimolecular interaction models: Langmuir, Hill (positive and negative cooperativity), site heterogeneity (<em>two-site</em>), and conformational transition (R/T). HTML export is self-contained, and CSV import/export uses <code>L,v</code> vectors.</div>
<div class="line-block">&nbsp;&nbsp;The main functions of the top bar are: RESET (restore default values), HTML (export standalone viewer), DIAGNOSTIC (show model summary), PARAMETERS (open control panel), EQUATIONS (show formulas), and CSV/LOAD (export or import data as <code>L,v</code>.</div>
<section id="adjustable-parameters" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="adjustable-parameters">Adjustable parameters</h4>
<ul>
<li><code>n</code>, <code>Kd</code>, <code>nH</code>, <code>sigma error</code>, <code>points</code>, <code>Lmin</code>, <code>Lmax</code></li>
<li>Two-site: <code>n1</code>, <code>n2</code>, <code>Kd1</code>, <code>Kd2</code></li>
<li>Conformational (R/T): <code>Kt</code>, <code>KdR</code>, <code>KdT</code></li>
<li><em>Toggles</em>: <em>draw model only</em>, <em>results below</em>, <em>Hill 10–90%</em>, <em>zoom sliders</em>, <em>autoplot</em></li>
</ul>
</section>
<section id="models-1" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="models-1">Models</h4>
<p><strong>Langmuir</strong></p>
<p><span class="math display">\[
v = n,\frac{[L]}{K_d + [L]}
\]</span><br>
</p>
<p><strong>Hill</strong></p>
<p><span class="math display">\[
v = n,\frac{[L]^{n_H}}{K_d^{n_H} + [L]^{n_H}}
\]</span></p>
<p><em>Linearization:</em></p>
<p><span class="math display">\[
\log!\left(\frac{v}{n - v}\right) = n_H \log[L] - n_H \log K_d
\]</span><br>
</p>
<p><strong>Two-site (site heterogeneity)</strong></p>
<p><span class="math display">\[
v = n_1,\frac{[L]}{K_{d1} + [L]} + n_2,\frac{[L]}{K_{d2} + [L]}
\]</span><br>
</p>
<p><strong>Conformational (R/T)</strong></p>
<p><span class="math display">\[
v = n,\frac{L/K_{dR} + K_t L/K_{dT}}{1 + L/K_{dR} + K_t(1 + L/K_{dT})}
\]</span><br>
</p>
<section id="classic-linearizations" class="level5 unnumbered">
<h5 class="unnumbered anchored" data-anchor-id="classic-linearizations">Classic linearizations</h5>
<p><br>
</p>
<div class="line-block">&nbsp;&nbsp;Fits are displayed as <strong>smooth spline lines</strong>, because polynomial regression parameters do not have direct physical meaning.</div>
<p><strong>Scatchard</strong></p>
<p><span class="math display">\[
\frac{v}{[L]} = \frac{n}{K_d} - \frac{1}{K_d} v
\]</span><br>
</p>
<p><strong>Benesi–Hildebrand</strong></p>
<p><span class="math display">\[
\frac{1}{v} = a,\frac{1}{[L]} + b
\]</span></p>
</section>
</section>
<section id="fits-and-estimates" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="fits-and-estimates">Fits and estimates</h4>
<ul>
<li><em>Binding (n,Kd)</em>: non-linear fit (Gauss–Newton).</li>
<li><em>Scatchard and Benesi:</em> illustrative spline interpolation.</li>
<li><em>Hill:</em> linear regression on the 10–90% range.</li>
</ul>
<section id="plots" class="level5" data-number="5.3.0.1">
<h5 data-number="5.3.0.1" class="anchored" data-anchor-id="plots"><span class="header-section-number">5.3.0.1</span> Plots</h5>
<ol type="1">
<li><em>Binding: v vs [L]</em> – theoretical curve + experimental points.</li>
<li><em>Scatchard: v/L vs v</em> – spline line fit to data.</li>
<li><em>Benesi–Hildebrand: 1/[L] vs 1/v</em> – illustrative spline.</li>
<li><em>Hill: log(v/(n–v)) vs log[L]</em> – linear regression with display of nH and Kd.</li>
</ol>
</section>
</section>
<section id="suggestion-36" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-36">Suggestion</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Use the <span class="fu">slider</span> (zoom bar) to enlarge<span class="sc">/</span>reduce the plots;</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try customizing the graphical representations. For example, click the <span class="st">"Parameters"</span> button and change the model to negative cooperativity with two sites, assigning distinct values to <span class="fu">them</span> (optional).</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="references-1" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="references-1">References</h4>
<ol type="1">
<li>Carey, Jannette. Ligand-Binding Basics: Evaluating Intermolecular Affinity, Specificity, Stoichiometry, and Cooperativity. John Wiley &amp; Sons, 2025.</li>
<li>Roque, Ana Cecília A. “Ligand-Macromolecular interactions in drug discovery.” Clifton, NJ 2010; p 572 (2010).</li>
<li>Mannhold, Raimund, Hugo Kubinyi, and Gerd Folkers. Protein-ligand interactions: from molecular recognition to drug design. John Wiley &amp; Sons, 2006.</li>
<li>Klotz, Irving. Introduction to biomolecular energetics: including ligand–receptor interactions. Elsevier, 2012.</li>
</ol>
</section>
</section>
<section id="gh-toolkit" class="level3" data-number="5.4">
<h3 data-number="5.4" class="anchored" data-anchor-id="gh-toolkit"><span class="header-section-number">5.4</span> GH-Toolkit</h3>
<p><strong>General instructions</strong></p>
<div class="line-block">&nbsp;&nbsp;This <em>toolkit</em> presents some Gibbs–Helmholtz and van’t Hoff relationships observed in the study of protein thermodynamics and macromolecular interactions involving <em>folding</em>, <em>unfolding</em>, and <em>ligand-binding</em> with <em>constant</em> (ΔC_p) (and a simple variant with (ΔΔC_p)). It includes useful forms for simulation from parameters at <em>(T_0)</em> or at <em>(T_m)</em>, as well as <em>diagnostic linearizations</em> for direct parameter reading.</div>
<div class="line-block">&nbsp;&nbsp;The simulator lets you work with fundamental equations (forms in (T_0) and centered at (T_m)) with <em>constant</em> (ΔC_p) and a linear-variation variant. Interpretation of the <em>fits (regressions)</em> is shown graphically from the selected <em>Parameters</em>. It also includes some <em>Scenarios</em> (mesophile, psychrophile, thermophile) and <em>Presets</em> for thermodynamic quantities. The app can simulate (ΔG(T)) directly from ((T_m,,ΔH_m,,ΔC_p)), and exports plots and data (CSV/HTML) in selectable units (kJ/mol ↔︎ kcal/mol; J/mol/K ↔︎ cal/mol/K).</div>
<p><br>
</p>
<p><a href="Eq/toolkit/gh-toolkit.js" download="gh-toolkit.js"><img src="Eq/toolkit/gh-toolkit.png" class="img-fluid"></a></p>
<section id="parameters-1" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="parameters-1">Parameters</h4>
<p><strong>1. Reference form at (T_0)</strong> (integrals with constant (ΔC_p))</p>
<p><em>Enthalpy</em> <span class="math display">\[
ΔH(T)=ΔH^\circ + ΔC_p,(T-T_0)
\]</span></p>
<p><em>Entropy</em> <span class="math display">\[
ΔS(T)=ΔS^\circ + ΔC_p\ln!\left(\frac{T}{T_0}\right)
\]</span></p>
<p><em>Gibbs energy (integrated Gibbs–Helmholtz)</em> <span class="math display">\[
ΔG(T)=ΔH^\circ - T,ΔS^\circ + ΔC_p\Big[(T-T_0)-T\ln!\left(\frac{T}{T_0}\right)\Big]
\]</span></p>
<p><em>Equilibrium constant</em> <span class="math display">\[
\ln K(T)= -\frac{ΔG(T)}{R,T}
\]</span></p>
<p><strong>2. Form centered at (T_m)</strong> (stability curve via (ΔH_m) and (ΔC_p))</p>
<p>This form is convenient when the equilibrium temperature (T_m) (where (ΔG=0)), the enthalpy change at (T_m) ((ΔH_m)), and (ΔC_p) are known:</p>
<p><span class="math display">\[
ΔG(T)=ΔH_m!\left(1-\frac{T}{T_m}\right) + ΔC_p\Big[(T-T_m)-T\ln!\left(\frac{T}{T_m}\right)\Big].
\]</span></p>
<div class="line-block">&nbsp;&nbsp;From this expression it is possible to recover parameters at (T_0) consistent with the curve:</div>
<span class="math display">\[\begin{aligned}
ΔH^\circ(T_0) &amp;= ΔH_m + ΔC_p,(T_0-T_m),[4pt]
ΔS^\circ(T_0) &amp;= \frac{ΔH_m}{T_m} + ΔC_p\ln!\left(\frac{T_0}{T_m}\right).
\end{aligned}\]</span>
<p>In the app, the button <em>“generate by Eq. (T_m)”</em> applies these relations.</p>
<p><strong>3. Heat capacity with linear variation</strong> (term (ΔΔC_p))</p>
<div class="line-block">&nbsp;&nbsp;To explore small linear variations of heat capacity around a reference (T_r):</div>
<p><span class="math display">\[
C_p(T)=C_{p,r}+ΔΔC_p,(T-T_r).
\]</span></p>
<p>In the app, checking <strong>“Compare two proteins (use (ΔΔC_p)”</strong> plots protein B with (ΔC_{p,B}=ΔC_{p,A}+ΔΔC_p) and shows the consequence on the curvature of (ΔG(T)) and the shift of (T_m).</p>
</section>
<section id="scenarios" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="scenarios">Scenarios</h4>
<div class="line-block">&nbsp;&nbsp;The <em>Scenario</em> chooses the nature of the modeled process and initializes coherent signs and magnitudes:</div>
<ul>
<li><em>Exothermic folding</em>: (ΔH^&lt;0); enthalpy-dominant stabilization at (T_0).</li>
<li><em>Endothermic folding</em>: (ΔH^&gt;0) and (ΔS^&gt;0) (entropy gain compensates).</li>
<li><em>Unfolding</em>: inverts folding signs.</li>
<li><em>Exothermic / endothermic binding</em>: typical settings for ligand–receptor interaction.</li>
</ul>
</section>
<section id="suggestion-37" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-37">Suggestion</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Note<span class="sc">:</span> T_m occurs when ΔG<span class="ot">=</span><span class="fl">0.</span> In that case, a dashed vertical line is shown <span class="cf">in</span> the ΔG x T plot;</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Choose a didactic <span class="st">"Preset"</span> and compare its effect on stability curves<span class="sc">:</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span> None<span class="sc">:</span> coherent random draw.</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span> term ↑ΔG global<span class="sc">:</span> makes ΔS more <span class="fu">negative</span> (raises ΔG globally).</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span> term ↓ΔC_p<span class="sc">:</span> flattens curvature <span class="cf">in</span> Δ<span class="fu">G</span>(T).</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span> psychro ↑ΔC_p<span class="sc">:</span> accentuates curvature.</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span> binder ΔC_p<span class="sc">&lt;</span><span class="dv">0</span><span class="sc">:</span> uses negative Δ<span class="fu">C_p</span> (cases of reduced hydrophobic binding etc.).</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Curvature of Δ<span class="fu">G</span>(T) grows with <span class="sc">|</span>ΔC_p<span class="sc">|</span>; thus, compare A vs B, and see the difference <span class="cf">in</span> the graphical <span class="fu">profile</span> (<span class="st">"shape"</span>) and T_m shift;</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Note that<span class="sc">:</span></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span> the van’t Hoff representation recovers ΔH and Δ<span class="fu">S</span> (when Δ<span class="at">C_p =</span> <span class="dv">0</span>).</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span> slopes<span class="sc">/</span>intercepts serve as consistency checks between plots</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span> from ΔH x T<span class="sc">:</span> slope gives ΔC_p.</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span> from ΔS x ln T<span class="sc">:</span> slope gives ΔC_p, and intercept is compatible with ΔS and T_0.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="references-2" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="references-2">References</h4>
<ol type="1">
<li>Atkins, Peter William, et al.&nbsp;Physical chemistry for the life sciences. Oxford University Press, 2023.</li>
<li>Hammes, Gordon G. Thermodynamics and kinetics for the biological sciences. John Wiley &amp; Sons, 2000.</li>
<li>Cooper, Alan. Biophysical chemistry. No.&nbsp;24. Royal Society of Chemistry, 2011.</li>
<li>Creighton, T. The Biophysical Chemistry of Nucleic Acids &amp; Proteins. Helvetian Press, 2010.</li>
</ol>
</section>
</section>
<section id="mtherm-toolkit" class="level3" data-number="5.5">
<h3 data-number="5.5" class="anchored" data-anchor-id="mtherm-toolkit"><span class="header-section-number">5.5</span> mTherm-Toolkit</h3>
<p><strong>General instructions</strong></p>
<div class="line-block">&nbsp;&nbsp;This <em>toolkit</em> is an educational app to simulate protein denaturation profiles (e.g., urea, GdnHCl), with support for multimeric proteins and an empirical term for osmolytes. Data are plotted as a function of an experimental signal (absorbance/fluorescence), followed by the calculation of conformational fractions and effective Gibbs energy as a function of denaturant. The simulator also allows comparison between proteins or experimental conditions (A/B).</div>
<div class="line-block">&nbsp;&nbsp;The app uses a <em>two-state model</em> between a “native” set (F) and a “denatured” set (U). The denaturant effect is linear on the standard free energy of the monomer (LEA – Linear Extrapolation Assumption). To simulate an osmolyte effect (stabilizing or not), an empirical multiplicative term is applied to the slope (m).</div>
<p><br>
</p>
<p><a href="Eq/toolkit/mtherm-toolkit.js" download="mtherm-toolkit.js"><img src="Eq/toolkit/mtherm-toolkit.png" class="img-fluid"></a></p>
<p><br>
</p>
<div class="line-block">&nbsp;&nbsp;In addition, the simulator allows export of interactive figures and diagnostics (HTML), CSV export of denaturant, signal, and energy data together with diagnostics, as well as JSON interface presets for scenarios (save/load).</div>
<section id="parameters-2" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="parameters-2">Parameters</h4>
<p><strong>Standard free energy vs.&nbsp;([D])</strong> — monomer (LEA)</p>
<p><span class="math display">\[
\Delta G^\circ(D)=\Delta G^\circ(\mathrm{H_2O})-m,D
\tag{2.1}
\]</span></p>
<p><strong>Multimer (F_n nU)</strong> — mass action (solve (f_U))</p>
<p><span class="math display">\[
\Delta G^\circ + RT\ln!\left(\frac{n,f_U^{,n}}{P_t^{,n-1}(1-f_U)}\right)=0
\tag{2.2}
\]</span><br>
</p>
<ul>
<li>(n): number of monomers in (F_n);</li>
<li>(P_t): <em>monomer-equivalent</em> concentration (in M);</li>
<li>Eq. (2.2) is solved numerically (bisection) for (0&lt;f_U&lt;1) at each (D);</li>
</ul>
<p><strong>Effective free energy</strong></p>
<p><span class="math display">\[
\Delta G_{\mathrm{eff}}(D)=-RT\ln!\left(\frac{f_U}{1-f_U}\right)
\tag{2.3}
\]</span></p>
<p><strong>Effective slope (m_{}) (dependent on (D))</strong></p>
<p><span class="math display">\[
m_{\mathrm{eff}}(D) = \frac{m}{,n + (1-n), f_U(D),}\quad
\left(\approx \frac{m}{n}\ \text{on the native side; } \approx m \text{ on the denatured side}\right)
\tag{2.4}
\]</span></p>
<p><strong>Effective intercept in water</strong> (approximation for multimers)</p>
<p>$$ G_{}() </p>
<ul>
<li><p>n</p></li>
<li><p>RT!(1-1n)!P_t $$<br>
</p></li>
</ul>
<p><strong>Osmolytes</strong></p>
<div class="line-block">&nbsp;&nbsp;Osmolytes tend to increase stability against denaturant challenge. The simulator allows observation of the effect of compatible osmolytes (e.g., glycine, sorbitol, sucrose, betaine) on the resulting plots. In the app, a simple empirical effect is modeled as a scaling of the slope:</div>
<p><span class="math display">\[
m_{\mathrm{eff}} = m ,\bigl(1 + k_{\mathrm{osm}},[\mathrm{osm}]\bigr)
\tag{6.1}
\]</span></p>
<p>where (k_{}) is an adjustable strength parameter (typically 0.05–0.2 M<span class="math inline">\(^{-1}\)</span>). This term shifts the curve without imposing a specific mechanism, enabling didactic exploration.</p>
</section>
<section id="fits-and-parameters" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="fits-and-parameters">Fits and parameters</h4>
<ul>
<li><em>Temperature (T)</em> (K) and <em>(P_t)</em> (µM, monomer equivalent);</li>
<li><em>Energy in water</em> (()G<span class="math inline">\(^{H_2O}\)</span>) and <em>slope (m)</em>;</li>
<li><em>Multiplicity (n)</em>: (1 n ) for A and, independently, for B;</li>
<li><em>Baseline signals (S_F, S_U) and noise</em>;</li>
<li><em>A/B comparison</em>: second protein can have different (n), (m), ()G<span class="math inline">\(^{o}\)</span>, and osmolyte settings.</li>
</ul>
</section>
<section id="suggestion-38" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-38">Suggestion</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Observe curvature <span class="cf">in</span> the ΔG x [D] plot. Values of n<span class="sc">&gt;</span><span class="dv">1</span> suggest a multimeric state <span class="cf">for</span> the biopolymer;</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Note that the midpoint C_m, where f_U is approximately <span class="fl">0.5</span>, is marked <span class="cf">in</span> the diagnostic <span class="cf">for</span> reference;</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Compare osmolyte effects by comparing proteins or conditions A and B, changing osmolyte concentration and its empirical constant <span class="st">"k_osm"</span>.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="references-3" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="references-3">References</h4>
<ol type="1">
<li>Park, Chiwook, and Susan Marqusee. “Analysis of the stability of multimeric proteins by effective ΔG and effective m‐values.” Protein science 13.9 (2004): 2553-2558.</li>
<li>Holthauzen, Luis Marcelo F., et al.&nbsp;“Protein stability in the presence of cosolutes.” Methods in Enzymology. Vol. 492. Academic Press, 2011. 61-125.</li>
<li>Scholtz, J. Martin, Gerald R. Grimsley, and C. Nick Pace. “Solvent denaturation of proteins and interpretations of the m value.” Methods in enzymology. Vol. 466. Academic Press, 2009. 549-565.</li>
</ol>
</section>
</section>
<section id="thermstab---toolkit" class="level3" data-number="5.6">
<h3 data-number="5.6" class="anchored" data-anchor-id="thermstab---toolkit"><span class="header-section-number">5.6</span> ThermStab - Toolkit</h3>
<p><strong>General instructions</strong></p>
<div class="line-block">&nbsp;&nbsp;This <em>toolkit</em> is presented as an interactive didactic panel to explore thermokinetic stability parameters related to enzyme activity. It covers some Michaelis–Menten (MM), Eyring, Arrhenius relations, and half-life curves (t<span class="math inline">\(*{1/2}\)</span>). Like the other kits, it supports two proteins (A/B), exports HTML with plots and CSV of the state, and also allows estimation of Km and k<span class="math inline">\(*{cat}\)</span> values by non-linear fitting.</div>
<p><br>
</p>
<p><a href="Eq/toolkit/ts-toolkit.js" download="ts-toolkit.js"><img src="Eq/toolkit/ts-toolkit.png" class="img-fluid"></a></p>
<section id="parameters-3" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="parameters-3">Parameters</h4>
<p><strong>Michaelis–Menten (MM)</strong></p>
<p><span class="math display">\[
v = \frac{V_{\max}[S]}{K_m + [S]}
\]</span></p>
<p><span class="math display">\[
V_{\max} = k_{\text{cat}} \cdot E_0 \qquad (E_0 = 1.0\ \mu\text{M})
\]</span><br>
</p>
<div class="line-block">&nbsp;&nbsp;Associated thermodynamic relations:</div>
<p><span class="math display">\[
\Delta G^{\ddagger}_{ES} = -RT \ln \left(\frac{1}{K_m}\right)
\]</span></p>
<p><span class="math display">\[
\Delta G^{\ddagger}*{ET} = -RT \ln \left(\frac{k*{\text{cat}}}{K_m}\right)
\]</span></p>
<p><strong>Eyring equation</strong></p>
<p><span class="math display">\[
\ln \left(\frac{k h}{k_B T}\right) = -\frac{\Delta H^{\ddagger}}{RT} + \frac{\Delta S^{\ddagger}}{R}
\]</span></p>
<p><span class="math display">\[
\Delta G^{\ddagger} = \Delta H^{\ddagger} - T\Delta S^{\ddagger}
\]</span><br>
</p>
<p><strong>Arrhenius equation</strong></p>
<p><span class="math display">\[
\ln k = -\frac{E_a}{RT} + \ln A
\]</span><br>
</p>
<p><strong>Enzyme half-life</strong></p>
<p><span class="math display">\[
t_{1/2} = \frac{\ln 2}{k_d}
\]</span><br>
</p>
<p><strong>Binding equilibrium</strong></p>
<p><span class="math display">\[
K_d = \frac{[E][S]}{[ES]}
\]</span><br>
</p>
<p><span class="math display">\[
\ln K_d(T) = -\frac{\Delta H^\circ}{RT} + \frac{\Delta S^\circ}{R}
\]</span></p>
<p><strong>Kinetic relations</strong></p>
<p><span class="math display">\[
K_m \approx \frac{k_{-1} + k_{\text{cat}}}{k_1}
\]</span><br>
</p>
<p><span class="math display">\[
K_d \approx \frac{k_{-1}}{k_1}
\]</span><br>
</p>
<p><span class="math display">\[
k_{-1}(T) \approx k_1,K_d(T)
\]</span><br>
</p>
<p><strong>Relation between stability and catalysis</strong></p>
<p><span class="math display">\[
k_d(T) \approx k_1 K_d(T) - k_{\text{cat}}(T)
\]</span><br>
</p>
<p><span class="math display">\[
t_{1/2}(T) = \frac{\ln 2}{k_d(T)}
\]</span><br>
</p>
<p><strong>Didactic approximation</strong></p>
<p><span class="math display">\[
\ln t_{1/2} \approx \frac{\Delta H^\circ}{R}\left(\frac{1}{T}\right)* \frac{\Delta S^\circ}{R} + \text{constant}
\]</span><br>
</p>
</section>
<section id="suggestion-39" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-39">Suggestion</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Use the RESET button and discuss the shape of the MM curve and the influence of Km and kcat;</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Explore Eyring and Arrhenius relations to obtain ΔH‡, ΔS‡, ΔG‡ and Ea from slopes and intercepts of the plots;</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Explore the effect of temperature and compare with free <span class="fu">mode</span> (t₁<span class="sc">/</span>₂) <span class="cf">in</span> class.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="references-4" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="references-4">References</h4>
<ol type="1">
<li>Bhatti, Haq Nawaz, et al.&nbsp;“Effect of aniline coupling on kinetic and thermodynamic properties of Fusarium solani glucoamylase.” Applied microbiology and biotechnology 73.6 (2007): 1290-1298.</li>
<li>Lonhienne, Thierry, Charles Gerday, and Georges Feller. “Psychrophilic enzymes: revisiting the thermodynamic parameters of activation may explain local flexibility.” Biochimica et Biophysica Acta (BBA)-Protein Structure and Molecular Enzymology 1543.1 (2000): 1-10.</li>
</ol>
</section>
</section>
<section id="biocel-toolkit" class="level3" data-number="5.7">
<h3 data-number="5.7" class="anchored" data-anchor-id="biocel-toolkit"><span class="header-section-number">5.7</span> Biocel-Toolkit</h3>
<p><strong>General instructions</strong></p>
<div class="line-block">&nbsp;&nbsp;This <em>toolkit</em> covers a simulator that visualizes the operation of a microbial fuel cell (MFC) under a simplified <em>Thevenin</em> model. Its goal is to help understand the physical and thermodynamic performance of a fuel cell (MFC, or a simple battery) from the relationships between <em>voltage</em>, <em>current</em>, <em>power</em>, and the cell’s <em>internal resistance</em>.</div>
<div class="line-block">&nbsp;&nbsp;The simulator also includes the impact of biochemical modifiers and experimental variations such as biofilm maturation, substrate depletion, and <em>overshooting</em> observed under real operating conditions.</div>
<p><br>
</p>
<p><a href="Eq/toolkit/biocel-toolkit.js" download="biocel-toolkit.js"><img src="Eq/toolkit/biocel-toolkit.png" class="img-fluid"></a></p>
<div class="line-block">&nbsp;&nbsp;As with the previous <em>toolkits</em>, part of the simulator’s interactivity allows exporting and reimporting experimental data (CSV/JSON), generating a self-contained file with interactive plots and diagnostics (Save HTML), and reinitializing to a new configuration (Reset).</div>
<p><br>
</p>
<section id="parameters-4" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="parameters-4">Parameters</h4>
<p><strong>Thevenin model</strong></p>
<div class="line-block">&nbsp;&nbsp;The equivalent circuit of the microbial cell is modeled as a voltage source (E_{OCV}) (open-circuit voltage) associated with an internal resistance (R_{int}) and an external resistance (R_{ext}), according to Thevenin’s theorem.</div>
<p>Current (I) and the voltage across the load resistor (V) are given by:</p>
<p><span class="math display">\[
I = \frac{E_{OCV}}{R_{int} + R_{ext}}, \qquad
V = I , R_{ext}
\]</span><br>
</p>
<p><strong>Electrical power and maximum condition</strong></p>
<div class="line-block">&nbsp;&nbsp;The instantaneous power dissipated in the load is:</div>
<p><span class="math display">\[
P = E_{OCV} , I - R_{int} I^2
\]</span><br>
</p>
<div class="line-block">&nbsp;&nbsp;Substituting (I) from the Thevenin model:</div>
<p><span class="math display">\[
P = \frac{E_{OCV}^2 , R_{ext}}{(R_{int} + R_{ext})^2}
\]</span><br>
</p>
<p>The <em>maximum power point</em> occurs when (R_{ext} = R_{int}), known as the <em>Jacobi condition</em>, yielding:</p>
<p><span class="math display">\[
P_{\text{max}} = \frac{E_{OCV}^2}{4 R_{int}}
\]</span><br>
</p>
<p><strong>Activation and concentration overpotentials</strong></p>
<div class="line-block">&nbsp;&nbsp;To represent electrochemical activation and mass-transport phenomena, the observed potential (V_{pol}) is reduced by two loss terms:</div>
<p><span class="math display">\[
V_{pol} = V_{lin} - \eta_{act} - \eta_{conc}
\]</span></p>
<p>where:</p>
<p><span class="math display">\[
\eta_{act} = A \ln!\big(1 + \frac{I}{I_0}\big), \qquad
\eta_{conc} = C , \frac{(I/I_{lim})^2}{1 - I/I_{lim}}
\]</span><br>
</p>
<div class="line-block">&nbsp;&nbsp;These relations simulate voltage drops associated with electrochemical kinetics and limitations in transport of ions and substrates.</div>
<p><strong>Time dynamics</strong></p>
<div class="line-block">&nbsp;&nbsp;During operation, the cell shows changes in potential and internal resistance over time, represented by:</div>
<p><span class="math display">\[
E_{OCV}(t) = E_{\max}\big(1 - e^{-k_{act}t}\big) s(f)
\]</span><br>
</p>
<p><span class="math display">\[
R_{int}(t) = R_0 \big(1 + \beta (1 - f)\big)
\]</span><br>
</p>
<p><em>Where:</em></p>
<p><span class="math display">\[
s(f) = 0.40 + 0.60f, \quad f = \frac{Q}{Q_{theor}}
\]</span></p>
<div class="line-block">&nbsp;&nbsp;These functions describe progressive biofilm activation and resistive modulation due to substrate depletion.</div>
<p><br>
</p>
<p><strong>Charges and efficiencies</strong></p>
<div class="line-block">&nbsp;&nbsp;The theoretical charge available in the biocell is:</div>
<p><span class="math display">\[
Q_{theor} = n_{\text{gluc}} , n_e , F
\]</span><br>
</p>
<p><em>Where:</em> (F) is Faraday’s constant and (n_e = 24) is the number of electrons per mole of glucose.</p>
<div class="line-block">&nbsp;&nbsp;Charge efficiency ((_Q)) and energy efficiency ((_E)) are:</div>
<p><span class="math display">\[
\eta_Q = 100 \times \frac{\int I,dt}{Q_{theor}} \quad [%]
\]</span></p>
<p><span class="math display">\[
\eta_E = 100 \times \frac{\int IV,dt}{Q_{theor} E_{\max}} \quad [%]
\]</span></p>
</section>
<section id="interface-and-features" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="interface-and-features">Interface and features</h4>
<p>The BioFuelCell interface is divided into three plot panels and a parameter section:</p>
<ol type="1">
<li><em>Time × Potential (t × V)</em> — shows output potential evolution over time for each external resistor;</li>
<li><em>Polarization curve (V × I)</em> — shows voltage decay as a function of current, with or without overpotentials;</li>
<li><em>Power curve (I × P)</em> — computes power dissipated in the load and highlights the maximum power point.</li>
</ol>
<div class="line-block">&nbsp;&nbsp;Below the plots, automatically calculated parameters are displayed:</div>
<ul>
<li>(R_{int,lin}) — internal resistance by linear fit of the real curve;</li>
<li>(R_{int,ideal}) — theoretical internal resistance (Thevenin model);</li>
<li>(P_{}) and (R_{P}) — maximum power and corresponding resistance;</li>
<li>(_Q) and (_E) — charge and energy efficiencies.</li>
</ul>
</section>
<section id="parameters-and-modifiers" class="level4" data-number="5.7.1">
<h4 data-number="5.7.1" class="anchored" data-anchor-id="parameters-and-modifiers"><span class="header-section-number">5.7.1</span> Parameters and modifiers</h4>
<div class="line-block">&nbsp;&nbsp;The user can adjust:</div>
<ul>
<li><em>Resistor range</em> — minimum and maximum (R_{ext}) (in Ω), distributed logarithmically;</li>
<li><em>Modifier intensity</em> — from 0% to 300%;</li>
<li><em>Log scale for power</em> — toggles current axis to log scale;</li>
<li><em>Experimental presets</em> — predefined simulated operating conditions.</li>
</ul>
</section>
<section id="suggestion-40" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-40">Suggestion</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Observe the intersection point between the polarization curve and the ideal parabola. This point indicates the balance between electron supply and consumption;</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Note the initial slope of the V×I curve, which represents the internal <span class="fu">resistance</span> (R_int) of the biocell;</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Check the system’s internal efficiency by observing the position of the peak <span class="cf">in</span> the power plot;</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Try some of the experimental <span class="st">"Presets"</span> (mature biofilm, substrate excess, etc.) and evaluate how conditions affect the electrochemical performance shown <span class="cf">in</span> plots and results.</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>The potential vs time plot should appear.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
<section id="references-5" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="references-5">References</h4>
<ol type="1">
<li>Logan, B. E. Microbial Fuel Cells. John Wiley &amp; Sons, 2008.</li>
<li>Bard, A. J., &amp; Faulkner, L. R. Electrochemical Methods: Fundamentals and Applications. Wiley, 2022;</li>
<li>Silveira, G., &amp; Schneedorf, J. M. “Evaluation of kefir as a new anodic biocatalyst consortium for microbial fuel cell.” Applied Biochemistry and Biotechnology, 185(4), 1118–1131 (2018).<br>
</li>
</ol>
</section>
</section>
</section>
<section id="arduino-biocell" class="level2" data-number="6">
<h2 data-number="6" class="anchored" data-anchor-id="arduino-biocell"><span class="header-section-number">6</span> Arduino &amp; Biocell</h2>
<div class="line-block">&nbsp;&nbsp;To illustrate the potential of <em>JSPlotly</em> beyond the virtual world, the example below presents an experimental application of <em>JSPlotly</em> adapted for data acquisition from a <em>microbial fuel cell</em> using microcontroller boards such as Arduino and ESP32, as virtually exemplified in the <em>Biocel-Toolkit</em> above.</div>
<div class="line-block">&nbsp;&nbsp;Interfacing can be obtained via a WebSerial connection to an Arduino through a USB serial cable (below), as well as via <em>IoT</em> (“Internet of Things”) features through <em>Bluetooth/Wi-Fi</em>.</div>
<p><br>
</p>
<!-- [script18.js](Eq/script18.js){download="script18.js"} -->
<p><img src="Eq/Arduino/arduino.png" class="img-fluid"><br>
</p>
<section id="serial-cable" class="level3" data-number="6.1">
<h3 data-number="6.1" class="anchored" data-anchor-id="serial-cable"><span class="header-section-number">6.1</span> Serial cable</h3>
<div class="line-block">&nbsp;&nbsp;The code is adapted for standalone use: just connect the Arduino board with a USB serial cable to the notebook/PC, then press <em>Connect</em>. Below is the list of files needed to reproduce the experiment:</div>
<ol type="1">
<li><a href="Eq/Arduino/arduino.ino" download="arduino.ino">Arduino sketch</a> ;</li>
<li><a href="Eq/Arduino/arduino.html" target="_blank">HTML software</a> ;</li>
<li><a href="Eq/Arduino/protocolo.pdf" target="_blank">Experimental protocol</a> ;</li>
</ol>
<div class="line-block">&nbsp;&nbsp;The procedures file details how to connect the Arduino by <em>serial cable</em> to notebook/desktop, both with Arduino firmware and with the HTML software, and also via <em>Wi-Fi (Android/iOS)</em>.</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="st">"** QUICK TIP ** "</span><span class="sc">:</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  <span class="fl">1.</span> Connect the Arduino board;</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  <span class="fl">2.</span> Open a CMD or Terminal and type<span class="sc">:</span></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>   <span class="st">"python3 -m http.server"</span> ;</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>  <span class="fl">3.</span> Upload the sketch;</span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>  <span class="fl">4.</span> Open the software <span class="cf">in</span> <span class="fu">Chrome</span> (Firefox is problematic)</span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>  <span class="fl">5.</span> Connect to the Arduino port <span class="cf">in</span> the software</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="internet-of-things---iot" class="level3" data-number="6.2">
<h3 data-number="6.2" class="anchored" data-anchor-id="internet-of-things---iot"><span class="header-section-number">6.2</span> Internet of Things - IoT</h3>
<div class="line-block">&nbsp;&nbsp;The experiment illustrated above can also be conducted using <em>Bluetooth</em> or <em>Wi-Fi</em> modules present on some microcontroller boards, allowing even broader practical applications of the simulator and the <code>Plotly.js</code> library for <em>tablets and phones</em>. The image below illustrates an experiment conducted with a microbial fuel cell, interfaced via <em>Wi-Fi</em> to an Arduino Uno R4 WiFi board.</div>
<p><br>
</p>
<p><img src="Eq/Arduino/arduino_wifi.png" class="img-fluid"><br>
</p>
<div class="line-block">&nbsp;&nbsp;For <em>Wi-Fi</em> use, it is necessary to use this specific <a href="Eq/Arduino/arduino_wifi.ino" download="arduino_wifi.ino">Arduino-WiFi sketch</a> for the biocell.</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="st">"** QUICK TIP ** "</span><span class="sc">:</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  <span class="fl">1.</span> Connect the Arduino board;</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="fl">2.</span> Upload the sketch;</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="fl">3.</span> Open Serial Monitor;</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>  <span class="fl">4.</span> Optional<span class="sc">:</span> press Reset on the Arduino <span class="cf">for</span> <span class="sc">~</span><span class="dv">2</span>s;</span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>  <span class="fl">5.</span> Access the specific network on the phone</span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>   In Wi<span class="sc">-</span>Fi, choose Biocel_no<span class="fl">.1</span></span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>   Password<span class="sc">:</span> <span class="dv">12345678</span>  ;</span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>  <span class="fl">6.</span> Open the link http<span class="sc">:</span><span class="er">//</span><span class="dv">192</span>.<span class="dv">168</span>.<span class="fl">4.1</span></span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>The potential vs time plot should appear.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>
</section>

<a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../../../nivel/superior/jsplotly/jsplotly_geral_en.html" class="pagination-link" aria-label="JSPlotly">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">JSPlotly</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../../nivel/basico/jsplotly/jsplotly_bas_en.html" class="pagination-link" aria-label="School">
        <span class="nav-page-text">School</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>