<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.22">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Interactive Simulations for Biophysical Chemistry with JSPlotly &amp; GSPlotly</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../../nivel/basico/jsplotly/jsplotly_bas_en.html" rel="next">
<link href="../../../nivel/superior/jsplotly/gsplotly_en.html" rel="prev">
<link href="../../../avatar.png" rel="icon" type="image/png">
<script src="../../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-7b4406b7675125bc2ba204020e191172.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-4b957b30fffc5edb70387ff4d75203a6.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-S1L73VGCG5"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-S1L73VGCG5', { 'anonymize_ip': true});
</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../logo-unifal.png" alt="" class="navbar-logo">
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../index.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../erre_en.html"> 
<span class="menu-text">Reproducible Teaching</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-cheatsheets" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Cheatsheets</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-cheatsheets">    
        <li>
    <a class="dropdown-item" href="../../../cheatsheet/plotly.pdf">
 <span class="dropdown-text">Plotly</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../cheatsheet/jmol_cheat.pt.pdf">
 <span class="dropdown-text">Jmol</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../cheatsheet/r_cheat.pt.pdf">
 <span class="dropdown-text">R</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../../../producao_en.html"> 
<span class="menu-text">Works</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../contato_en.html"> 
<span class="menu-text">Contact</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../../pt/index.html"> 
<span class="menu-text">PT</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools tools-wide">
    <a href="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-true"></i></a>
    <a href="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-true"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../nivel/superior/jsplotly/jsplotly_geral_en.html">JSPlotly &amp; GSPlotly - Interactive Graphics &amp; AI</a></li><li class="breadcrumb-item"><a href="../../../nivel/superior/jsplotly/jsptl_bioq_en.html">Simulations in Biophysical Chemistry</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">JSPlotly &amp; GSPlotly - Interactive Graphics &amp; AI</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jsplotly/jsplotly_geral_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">JSPlotly - Graphical simulation with JavaScript</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jsplotly/gsplotly_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GSPlotly - AI code generator for JSPlotly</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jsplotly/jsptl_bioq_en.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Simulations in Biophysical Chemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/jsplotly/jsplotly_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Simulations at School</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Jmol - 3D molecules</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jmol/index_jmol_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Atomic models in Biochemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/Curso/index_jmol_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3D molecules at School</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/Jardim/index_Jardim_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">A Molecule Garden</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">R - Maths and visualization</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/r/index_r_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R in Biophysical Chemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/ScriptRplot/scriptrplot_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">scriptRplot: elegant graphics with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/Curso/index_r_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R &amp; RStudio at School</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/R/index_OIER_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Interactive charts and maps at School</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Sisma - dynamic diagrams</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/sisma/index_sisma_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Dynamic flows in Biochemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/sisma/index_sisma_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Interactive diagrams at School</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul class="collapse">
  <li><a href="#instructions" id="toc-instructions" class="nav-link active" data-scroll-target="#instructions">Instructions</a></li>
  <li><a href="#biophysical-chemistry" id="toc-biophysical-chemistry" class="nav-link" data-scroll-target="#biophysical-chemistry"><strong>Biophysical chemistry</strong></a></li>
  <li><a href="#acid-base-equilibrium-and-buffer-system" id="toc-acid-base-equilibrium-and-buffer-system" class="nav-link" data-scroll-target="#acid-base-equilibrium-and-buffer-system"><span class="header-section-number">1</span> Acid-base equilibrium and buffer system</a></li>
  <li><a href="#ionization-state-of-biomolecule-chemical-groups" id="toc-ionization-state-of-biomolecule-chemical-groups" class="nav-link" data-scroll-target="#ionization-state-of-biomolecule-chemical-groups"><span class="header-section-number">2</span> Ionization state of biomolecule chemical groups</a></li>
  <li><a href="#net-charge-network-in-peptides" id="toc-net-charge-network-in-peptides" class="nav-link" data-scroll-target="#net-charge-network-in-peptides"><span class="header-section-number">3</span> Net charge network in peptides</a></li>
  <li><a href="#isoelectric-point-in-proteins" id="toc-isoelectric-point-in-proteins" class="nav-link" data-scroll-target="#isoelectric-point-in-proteins"><span class="header-section-number">4</span> Isoelectric Point in Proteins</a></li>
  <li><a href="#hydrophobicity-in-amino-acid-sequences" id="toc-hydrophobicity-in-amino-acid-sequences" class="nav-link" data-scroll-target="#hydrophobicity-in-amino-acid-sequences"><span class="header-section-number">5</span> Hydrophobicity in amino acid sequences</a></li>
  <li><a href="#application---experimental-simulation-for-characterizing-amino-acids-and-chains" id="toc-application---experimental-simulation-for-characterizing-amino-acids-and-chains" class="nav-link" data-scroll-target="#application---experimental-simulation-for-characterizing-amino-acids-and-chains"><span class="header-section-number">6</span> <em>Application</em> - Experimental simulation for characterizing amino acids and chains</a></li>
  <li><a href="#application---zemol-a-3d-viewer-for-molecular-models" id="toc-application---zemol-a-3d-viewer-for-molecular-models" class="nav-link" data-scroll-target="#application---zemol-a-3d-viewer-for-molecular-models"><span class="header-section-number">7</span> <em>Application</em> - ZeMol: a 3D viewer for molecular models</a></li>
  <li><a href="#interaction-of-oxygen-with-myoglobin-and-hemoglobin" id="toc-interaction-of-oxygen-with-myoglobin-and-hemoglobin" class="nav-link" data-scroll-target="#interaction-of-oxygen-with-myoglobin-and-hemoglobin"><span class="header-section-number">8</span> Interaction of oxygen with myoglobin and hemoglobin</a></li>
  <li><a href="#bohr-effect-on-hemoglobin-ph" id="toc-bohr-effect-on-hemoglobin-ph" class="nav-link" data-scroll-target="#bohr-effect-on-hemoglobin-ph"><span class="header-section-number">9</span> Bohr effect on hemoglobin (pH)</a></li>
  <li><a href="#enzyme-catalysis-and-inhibition" id="toc-enzyme-catalysis-and-inhibition" class="nav-link" data-scroll-target="#enzyme-catalysis-and-inhibition"><span class="header-section-number">10</span> Enzyme catalysis and inhibition</a></li>
  <li><a href="#linearization-of-kinetic-data-by-lineweaver-burk" id="toc-linearization-of-kinetic-data-by-lineweaver-burk" class="nav-link" data-scroll-target="#linearization-of-kinetic-data-by-lineweaver-burk"><span class="header-section-number">11</span> Linearization of kinetic data by Lineweaver-Burk</a></li>
  <li><a href="#enzyme-inhibition-diagnosis-michaelis-mentem-and-lineweaver-burk" id="toc-enzyme-inhibition-diagnosis-michaelis-mentem-and-lineweaver-burk" class="nav-link" data-scroll-target="#enzyme-inhibition-diagnosis-michaelis-mentem-and-lineweaver-burk"><span class="header-section-number">12</span> Enzyme Inhibition Diagnosis (Michaelis-Mentem and Lineweaver-Burk)</a></li>
  <li><a href="#thermodynamic-stability-of-nucleic-acids" id="toc-thermodynamic-stability-of-nucleic-acids" class="nav-link" data-scroll-target="#thermodynamic-stability-of-nucleic-acids"><span class="header-section-number">13</span> Thermodynamic stability of nucleic acids</a></li>
  <li><a href="#van-der-waals-equation-for-ideal-gases" id="toc-van-der-waals-equation-for-ideal-gases" class="nav-link" data-scroll-target="#van-der-waals-equation-for-ideal-gases"><span class="header-section-number">14</span> Van der Waals equation for ideal gases</a></li>
  <li><a href="#balance-of-atp-production-from-reagents-temperature-and-ph" id="toc-balance-of-atp-production-from-reagents-temperature-and-ph" class="nav-link" data-scroll-target="#balance-of-atp-production-from-reagents-temperature-and-ph"><span class="header-section-number">15</span> Balance of ATP production from reagents, temperature, and pH</a></li>
  <li><a href="#variation-of-gibbs-energy-with-temperature" id="toc-variation-of-gibbs-energy-with-temperature" class="nav-link" data-scroll-target="#variation-of-gibbs-energy-with-temperature"><span class="header-section-number">16</span> Variation of Gibbs energy with temperature</a></li>
  <li><a href="#electrochemical-impedance-spectroscopy-eis" id="toc-electrochemical-impedance-spectroscopy-eis" class="nav-link" data-scroll-target="#electrochemical-impedance-spectroscopy-eis"><span class="header-section-number">17</span> Electrochemical impedance spectroscopy (EIS)</a></li>
  <li><a href="#cyclic-voltammetry" id="toc-cyclic-voltammetry" class="nav-link" data-scroll-target="#cyclic-voltammetry"><span class="header-section-number">18</span> Cyclic voltammetry</a></li>
  <li><a href="#diagrams-and-flowcharts" id="toc-diagrams-and-flowcharts" class="nav-link" data-scroll-target="#diagrams-and-flowcharts"><span class="header-section-number">19</span> Diagrams and flowcharts</a></li>
  <li><a href="#data-analysis" id="toc-data-analysis" class="nav-link" data-scroll-target="#data-analysis"><strong>Data Analysis</strong></a></li>
  <li><a href="#user-data-insertion" id="toc-user-data-insertion" class="nav-link" data-scroll-target="#user-data-insertion"><span class="header-section-number">20</span> User data insertion</a></li>
  <li><a href="#loading-a-file-for-analysis" id="toc-loading-a-file-for-analysis" class="nav-link" data-scroll-target="#loading-a-file-for-analysis"><span class="header-section-number">21</span> Loading a file for analysis</a></li>
  <li><a href="#linear-data-adjustment" id="toc-linear-data-adjustment" class="nav-link" data-scroll-target="#linear-data-adjustment"><span class="header-section-number">22</span> Linear data adjustment</a></li>
  <li><a href="#polynomial-regression" id="toc-polynomial-regression" class="nav-link" data-scroll-target="#polynomial-regression"><span class="header-section-number">23</span> Polynomial regression</a></li>
  <li><a href="#multilinear-regression" id="toc-multilinear-regression" class="nav-link" data-scroll-target="#multilinear-regression"><span class="header-section-number">24</span> Multilinear regression</a></li>
  <li><a href="#response-surface-methodology-rsm" id="toc-response-surface-methodology-rsm" class="nav-link" data-scroll-target="#response-surface-methodology-rsm"><span class="header-section-number">25</span> Response Surface Methodology (RSM)</a></li>
  <li><a href="#data-smoothing---spline-and-savitzky-golay-filter" id="toc-data-smoothing---spline-and-savitzky-golay-filter" class="nav-link" data-scroll-target="#data-smoothing---spline-and-savitzky-golay-filter"><span class="header-section-number">26</span> Data smoothing - Spline and Savitzky-Golay filter</a></li>
  <li><a href="#anova-and-tukey-hsd-test" id="toc-anova-and-tukey-hsd-test" class="nav-link" data-scroll-target="#anova-and-tukey-hsd-test"><span class="header-section-number">27</span> ANOVA and Tukey-HSD test</a></li>
  <li><a href="#nonlinear-data-adjustment" id="toc-nonlinear-data-adjustment" class="nav-link" data-scroll-target="#nonlinear-data-adjustment"><span class="header-section-number">28</span> Nonlinear data adjustment</a></li>
  <li><a href="#equation-19" id="toc-equation-19" class="nav-link" data-scroll-target="#equation-19">Equation:</a></li>
  <li><a href="#application---digital-whiteboard" id="toc-application---digital-whiteboard" class="nav-link" data-scroll-target="#application---digital-whiteboard"><span class="header-section-number">29</span> Application - Digital Whiteboard</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../nivel/superior/jsplotly/jsplotly_geral_en.html">JSPlotly &amp; GSPlotly - Interactive Graphics &amp; AI</a></li><li class="breadcrumb-item"><a href="../../../nivel/superior/jsplotly/jsptl_bioq_en.html">Simulations in Biophysical Chemistry</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Interactive Simulations for Biophysical Chemistry with JSPlotly &amp; GSPlotly</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div class="line-block">To illustrate the potential use of <em>JSPlotly</em> for Biochemistry and related areas, here are some examples of <strong>simulations and data analysis</strong>, whose graphs are often found in textbooks and related content. To get the most out of each topic, try following the suggestions for <em>parametric manipulation</em> of the codes in each topic.</div>
<p><br>
</p>
<section id="instructions" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="instructions">Instructions</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Choose a topic;</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Click on the corresponding graph;</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Click on <span class="st">"Add Plot"</span>;</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Use the mouse <span class="cf">for</span> interactivity and<span class="sc">/</span>or edit the code. </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>Reminder<span class="sc">:</span> the editor uses infinite undo<span class="sc">/</span>redo <span class="cf">in</span> the <span class="fu">code</span> (Ctrl<span class="sc">+</span>Z <span class="sc">/</span> Shift<span class="sc">+</span>Ctrl<span class="sc">+</span>Z) <span class="sc">!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="biophysical-chemistry" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="biophysical-chemistry"><strong>Biophysical chemistry</strong></h2>
</section>
<section id="acid-base-equilibrium-and-buffer-system" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="acid-base-equilibrium-and-buffer-system"><span class="header-section-number">1</span> Acid-base equilibrium and buffer system</h2>
<section id="context" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context">Context:</h4>
<div class="line-block">The example illustrates the addition of a base to a system containing a weak acid and its conjugate base. The titration equation refers to a <em>triprotic acid</em>, the phosphoric acid of the buffer of the same name, but it would also apply to many others, such as those present in the Krebs cycle (citrate, isocitrate).</div>
</section>
<section id="equation" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation">Equation:</h4>
<p><span class="math display">\[
fa= \frac{1}{1 + 10^{pKa1 - pH}} + \frac{1}{1 + 10^{pKa2 - pH}} + \frac{1}{1 + 10^{pKa3 - pH}}
\]</span> <em>Where</em>, fa = fraction of acid (protonatable groups)<br>
</p>
<p><a href="Eq/jsp_acbase.html" target="_blank"><img src="Eq/tit_acbase.png" class="img-fluid"></a><br>
<!---div class="jsplotly_sup-fundo"> ---></p>
</section>
<section id="suggestion" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="st">"A. Converting the phosphate buffer (triprotic) curve to bicarbonate buffer (diprotic)"</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Change the pKa values <span class="cf">for</span> the bicarbonate buffer<span class="sc">:</span> pKa1 <span class="ot">=</span> <span class="fl">6.1</span>, and pKa2 <span class="ot">=</span> <span class="fl">10.3</span>;</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Set a very large value <span class="cf">for</span> <span class="fu">pKa3</span> (e.g., <span class="fl">1e20</span>).</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Click on <span class="st">"add plot."</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>Explanation<span class="sc">:</span> pKa is a term that represents the logarithm of a dissociation <span class="fu">constant</span> (<span class="sc">-</span>log[Ka]). With an extreme value, the denominator becomes equally large, canceling out the term that carries pKa3. In JavaScript and other languages, <span class="st">"e"</span> represents the notation <span class="cf">for</span> power of <span class="fl">10.</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="st">"B. Converting the bicarbonate buffer curve to acetate"</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Simply <span class="cf">repeat</span> the above procedure, with pKa1 <span class="ot">=</span> <span class="fl">4.75</span>, and eliminating pKa2.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="ionization-state-of-biomolecule-chemical-groups" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="ionization-state-of-biomolecule-chemical-groups"><span class="header-section-number">2</span> Ionization state of biomolecule chemical groups</h2>
<section id="context-1" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-1">Context:</h3>
<div class="line-block"><em>JSPlotly</em> also allows the observation of some concepts facilitated by <em>animation-based learning</em> together with graphical representations obtained by mathematical functions. In the following example, the interactive object illustrates the protonation state for the main chemical groups found in biomolecules.</div>
<p><br>
</p>
</section>
<section id="instruction" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="instruction">Instruction:</h3>
<ol type="1">
<li><p>After clicking on the related image below, click on “add” and observe the acid-base titration graph generated. The abscissa axis (x-axis) has a wide range of pH values, and the y-axis shows the charge that the molecule with ionization potential manifests at each pH, as a function of its pKa value;</p></li>
<li><p>Click on “Animate” to observe the displacement of a small ball along the titration curve, up to a pH value predefined in the simulator script, and as a function of its pKa;</p></li>
<li><p>Modify the type of ionizable group (in “let grupo_edit =”<em>) and/or the desired final pH (“let pH_final_edit =”</em>) in the script to evaluate the ionization state of a new group;</p></li>
<li><p>Click ‘Reset’ and “Animate” to view the new animation.</p></li>
</ol>
<p><a href="Eq/jsp_pHpKa.html" target="_blank"><img src="Eq/pHpKa.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-1" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-1">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Change the pH and the type of group you want to observe;</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Simulate the ionization state of different molecules, such as drugs with carboxylic <span class="fu">acid</span> (e.g., acetylsalicylic acid...aspirin), <span class="cf">in</span> blood pH and <span class="cf">in</span> the gastric cavity;</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Create a new ionizable group and predict its ionization state. To do this, simply change the information <span class="cf">in</span> <span class="dv">3</span> constants at the beginning of the script<span class="sc">:</span> groups, pKas, and types.</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>a.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="net-charge-network-in-peptides" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="net-charge-network-in-peptides"><span class="header-section-number">3</span> Net charge network in peptides</h2>
<section id="context-2" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-2">Context:</h4>
<div class="line-block">The code refers to the net charge network present in any sequence of amino acid residues. Here we illustrate <em>angiotensin II</em>, an important peptide for regulating blood pressure and electrolyte balance, whose converting enzyme is associated with the mechanism of cell invasibility by SARS-CoV-2.</div>
</section>
<section id="equation-1" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-1">Equation:</h4>
<div class="line-block">There is no direct equation for this simulation, since the algorithm needs to decide the charge based on the basic or acidic nature of the residue at a given pH value (see the <em>script</em>). Thus:</div>
<p><span class="math display">\[
q =
\begin{cases}
-\dfrac{1}{1 + 10^{pK_a - pH}} &amp; \text{(acid group)} \\\\
\dfrac{1}{1 + 10^{pH - pK_a}} &amp; \text{(basic group)}
\end{cases}
\]</span><br>
</p>
<p><em>Where</em>,</p>
<ul>
<li>pKa = base 10 antilogarithm value for the acid dissociation equilibrium constant, Ka (or log[Ka]).</li>
</ul>
<p><a href="Eq/jsp_cargaAA3.html" target="_blank"><img src="Eq/cargaAA3.png" class="img-fluid"></a><br>
<!---div class="jsplotly_sup-fundo"> ---></p>
</section>
<section id="suggestion-2" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-2">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Select the peptide sequence below and observe the charge distribution<span class="sc">:</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="st">"Ala,Lys,Arg,Leu,Phe,Glu,Cys,Asp,His"</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Simulate the pH condition of the <span class="fu">stomach</span> (<span class="st">"const pH = 1.5"</span>), and check the change <span class="cf">in</span> charges <span class="cf">in</span> the peptide. </span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Select a physiological <span class="fu">peptide</span> (e.g., oxytocin), observe its charge <span class="cf">in</span> <span class="fu">blood</span> (pH <span class="fl">7.5</span>), and reflect on its potential <span class="cf">for</span> electrostatic interaction with cellular components.</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="st">"Cys, Tyr, Ile, Gln, Asn, Cys, Pro, Leu, Gly"</span> <span class="sc">-</span> oxytocin</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="isoelectric-point-in-proteins" class="level2" data-number="4">
<h2 data-number="4" class="anchored" data-anchor-id="isoelectric-point-in-proteins"><span class="header-section-number">4</span> Isoelectric Point in Proteins</h2>
<section id="context-3" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-3">Context:</h4>
<div class="line-block">The <em>script</em> for this simulation is based on the distribution of the charge network for a polyelectrolyte, and the identification of the pH value at which this network is zero, i.e., the <em>isoelectric (or isoionic) point, pI</em>. The example uses the primary sequence of lysozyme, a hydrolase that acts to break down the microbial wall.</div>
<p>Equation: {.unnumbered}</p>
<p><span class="math display">\[
q_{\text{net}}(pH) = \sum_{i=1}^{N} \left[ n_i \cdot q_{B_i} + \frac{n_i}{1 + 10^{pH - pK_{a_i}}} \right]
\]</span><br>
<em>Where</em>,</p>
<ul>
<li>qnet = total net charge;</li>
<li>qB$_{i} = charge of the basic form for residue i (e.g., +1 for Lys, 0 for Asp);</li>
<li>n<span class="math inline">\(_{i}\)</span> = number of groups of residue <em>i</em>.</li>
</ul>
<p><a href="Eq/jsp_pI.html"><img src="Eq/pI.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-3" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-3">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="st">"Finding the pI for other proteins"</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> You can check the titration of any other protein or peptide sequence by simply replacing the primary sequence contained <span class="cf">in</span> the code. An assertive way to perform this replacement involves<span class="sc">:</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>a. Searching <span class="cf">for</span> the <span class="st">"FASTA"</span> sequence of the protein <span class="cf">in</span> <span class="fu">NCBI</span> (<span class="st">"https://www.ncbi.nlm.nih.gov/protein/"</span>) <span class="sc">-</span> e.g., ‘papain’;</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>b. Clicking on <span class="st">"FASTA"</span> and copying the sequence obtained <span class="cf">in</span> step <span class="dv">1</span>a;</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>c. Pasting the sequence into a website <span class="cf">for</span> residue <span class="fu">quantification</span> (e.g., <span class="st">"https://www.protpi.ch/Calculator/ProteinTool"</span>);</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Replacing the sequence <span class="cf">in</span> the code.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="hydrophobicity-in-amino-acid-sequences" class="level2" data-number="5">
<h2 data-number="5" class="anchored" data-anchor-id="hydrophobicity-in-amino-acid-sequences"><span class="header-section-number">5</span> Hydrophobicity in amino acid sequences</h2>
<section id="context-4" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-4">Context:</h3>
<div class="line-block">Observing amino acid hydrophobicity levels can be useful for studying and predicting residue sequences, such as those found in the central portion of transmembrane protein helices, as well as for identifying specific sites in enzymes, such as hydrophobic clefts and pockets.</div>
<div class="line-block">These levels are classified relatively, giving rise to indices such as the <em>Kyte–Doolittle scale</em>, which assigns each amino acid a hydrophobicity value estimated from experimental and observational data. Positive values indicate a hydrophobic tendency (e.g., Ile, Val, Leu), and negative values indicate a hydrophilic tendency (e.g., Asp, Glu, Lys, Arg). The index allows us to highlight sections prone to nonpolar environments, as well as regions exposed to the aqueous solvent.</div>
</section>
<section id="equation-2" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-2">Equation:</h3>
<p><span class="math display">\[
\tilde{y}_i \;=\; \frac{1}{w} \sum_{k=i-m}^{i+m} y_k,
\qquad w = 2m+1
\]</span><br>
</p>
<p><em>Where</em>,</p>
<span class="math display">\[\begin{aligned}
y_i &amp;:\; \text{hydrophobicity value at position $i$ of the sequence} \\
\tilde{y}_i &amp;:\; \text{smoothed value (moving average) at position $i$} \\
m &amp;:\; \text{half-window (number of neighbors on each side)} \\
w &amp;:\; \text{total window width ($w=2m+1$)} \\
i &amp;:\; \text{position of the amino acid in the sequence ($1 \leq i \leq N$)}
\end{aligned}\]</span>
<p><a href="Eq/jsp_hidrofob.html" target="_blank"><img src="Eq/hidrofob.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-4" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-4">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try changing the sequence to a known one;</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Compare a polar sequence with a nonpolar one by <span class="fu">overlaying</span> (<span class="st">"add"</span>);</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Transcribe a transmembrane helix sequence of a protein by accessing its data on the PDB website</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="application---experimental-simulation-for-characterizing-amino-acids-and-chains" class="level2" data-number="6">
<h2 data-number="6" class="anchored" data-anchor-id="application---experimental-simulation-for-characterizing-amino-acids-and-chains"><span class="header-section-number">6</span> <em>Application</em> - Experimental simulation for characterizing amino acids and chains</h2>
<section id="context-5" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-5">Context:</h3>
<div class="line-block"><em>JSPlotly</em> allows you to work with a set of interactions not necessarily linked to equations and their graphical representations. The following example illustrates its potential for simulating a classic experiment in biochemistry: <em>the characterization of amino acids, peptides, and proteins by chromogenic reactions in test tubes</em>. As the scope of this material is to present and facilitate the use of various <em>OIERs</em> (<em>interactive reproducible teaching objects</em>), users are recommended to search for sources of information related to the reactions briefly described below.</div>
<section id="simulator-instructions" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="simulator-instructions">Simulator instructions</h4>
<div class="line-block">The simulator allows two options: a) discover the nature of a sample or b) discover the color pattern of reactions for a chosen sample.</div>
<p><br>
</p>
<p><strong>Option A: discover the nature of a sample:</strong></p>
<div class="line-block">Here you can <em>“play”</em> to check whether you are dealing with a protein/peptide (chain) sample, an amino acid (AA) sample, a specific mixture of AAs, or AA(s) present in a chain. To do this:</div>
<ol type="1">
<li><p>Click on the image below and select <em>“add.”</em> A set of seven colored bars will be generated, representing the colors obtained in test tubes for seven different reactions for protein characterization: ninhydrin, biuret, Pauly, xanthoprotein, Sakagushi, Millon, and sulfur (Cys).</p></li>
<li><p>Discover the type of sample based on the color pattern of the tubes presented, providing a guess. The guess must be by type (in the <em>“Guess (type)”</em> menu - AA, chain, or mixture of AAs) and by profile (in the <em>“Guess (profile)”</em> menu - random, specific AA(s), or specific mixture);</p></li>
<li><p>After selecting the 2 menus, click <em>“Check”</em>. If you are correct, a <em>“Score”</em> just below will calculate the number of correct answers/number of samples <em>‘played’</em>;</p></li>
<li><p>If you are wrong, try again or look for an explanation of the color pattern in <em>“Explain”</em>;</p></li>
</ol>
<ol start="5" type="1">
<li><p>There are two other possibilities: <em>“Hard mode,”</em> which hides the names of the reactions while maintaining their sequence, and <em>“Shuffle,”</em> which changes the order of the tubes;</p></li>
<li><p>For a new sample, just click on <em>“New sample”</em>;</p></li>
</ol>
<p><strong>Option B: discover the color pattern of a reaction for a selected sample:</strong></p>
<ol type="1">
<li>Select a sample in <em>“Generate,”</em> and click <em>“New sample.”</em> A set of tubes will be generated with the expected reaction pattern for the sample.</li>
</ol>
<p><a href="Eq/jsp_Id-Prot-7.html" target="_blank"><img src="Eq/Id-Prot-7.png" class="img-fluid"></a></p>
</section>
</section>
<section id="suggestion-5" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-5">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try clicking on <span class="st">"New sample"</span> to check the frequency of possibilities;</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try the two <span class="st">"game"</span> operation <span class="fu">options</span> (yes, it<span class="st">'s a game... after all, it has a scoreboard!);</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="st">3. Try "hard mode" (no reference to reactions in sequence) or "shuffle" (swap the order of reactions). Obviously, however, avoid both simultaneously!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="application---zemol-a-3d-viewer-for-molecular-models" class="level2" data-number="7">
<h2 data-number="7" class="anchored" data-anchor-id="application---zemol-a-3d-viewer-for-molecular-models"><span class="header-section-number">7</span> <em>Application</em> - ZeMol: a 3D viewer for molecular models</h2>
<section id="context-6" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-6">Context:</h3>
<div class="line-block">Through the combined features of <em>JavaScript</em> and the <em>Plotly.js</em> library, it is possible to extend the application potential of <em>JSPlotly</em> to build true HTML applications. A clear example of this scope is illustrated below, for a molecular viewer similar in nature to some pre-existing ones used in teaching and research (<em>Jmol</em>, <em>Pymol</em>, etc.).</div>
<div class="line-block">Although it is obviously much smaller in scope compared to other applications, the so-called <em>“ZeMol”</em> has less than 70 kB of memory and offers features such as:</div>
<ul>
<li>models loaded online from the <a href="https://www.rcsb.org/">PDB</a> website or <a href="https://pubchem.ncbi.nlm.nih.gov/">PubChem</a>;</li>
<li>models loaded from the device’s physical memory (PDB, MOL, SDF, XYZ, TXT formats);</li>
<li>PDB file identification information (or PubChem file name);</li>
<li>header containing the number of chains, ligands, ions, and disulfide bonds;</li>
<li>identification of only the <span class="math inline">\(\alpha\)</span> carbons of each residue of the polypeptide backbone, facilitating model visualization;</li>
<li>automatic visualization of disulfide bonds;</li>
<li>identification of residues, ligands, and ions by hovering the mouse over the model;</li>
<li>checkbox for viewing the secondary structure of proteins (helix, <span class="math inline">\(\beta\)</span>-sheet);</li>
<li>checkbox for residue polarity index;</li>
<li>sliders for CA carbon size (PDB) or atoms in general, and model opacity (useful for highlighting crevices, 2a. structure, carbon chain, ligands, ions);</li>
<li>hide/show different parts of the model by clicking on its legend (chain, ions, helix/sheets, ligands);</li>
<li>Highlight AAs (individual, sequence, all of one type, predefined groups in the script);</li>
<li>Self-sufficient HTML when saving, “freezing” the model in the characteristics you want to present;</li>
<li>Zoom on mobile devices (one finger fixed on the screen and another dragging the model) and Pan (moving the model with two fingers on the screen) - actions also present in the icons above the model;</li>
<li>Optional adjustments in the script itself (sizes, colors, thicknesses, AA groups, line type and thickness, for example).</li>
</ul>
<p><a href="Eq/ZeMol.html" target="_blank"><img src="Eq/ZeMol.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-6" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-6">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Load a model with PDB code, or a molecule from the PubChem website;</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> If it is a protein, try the separate and combined effects of the sliders <span class="cf">for</span> CA size and opacity;</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Observe the polarity of protein residues <span class="cf">in</span> the checkbox of the same name;</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Try zooming <span class="cf">in</span><span class="sc">/</span>out or moving it around the screen with the mouse, or do it on a smartphone as instructed above;</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fl">5.</span> Show or hide each chain or all chains, ligands, and ions by clicking on the respective terms <span class="cf">in</span> the legend;</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fl">6.</span> Select residues of interest <span class="cf">in</span> the model, such as catalytic sites, coenzyme binding sites, or activity regulation sites. To do this, type <span class="cf">in</span> the corresponding field and click <span class="st">"highlight."</span> Some suggestions<span class="sc">:</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>a. Individual<span class="sc">:</span> Arg15, ASP117, <span class="dv">15</span>, A<span class="sc">:</span><span class="dv">15</span>, A<span class="sc">:</span>ARG15;</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>b. Ranges<span class="sc">:</span> <span class="dv">15-30</span>, A<span class="sc">:</span><span class="dv">15-30</span>, Arg15<span class="sc">-</span>Asp30;</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>c. Multiple separated by commas<span class="sc">:</span> <span class="st">"Arg15, Asp117"</span>;</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>d. Groups<span class="sc">:</span> <span class="st">"aromatic,"</span> <span class="st">"polar,"</span> <span class="st">"nonpolar,"</span> <span class="st">"basic,"</span> <span class="st">"acidic,"</span> <span class="st">"small."</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="interaction-of-oxygen-with-myoglobin-and-hemoglobin" class="level2" data-number="8">
<h2 data-number="8" class="anchored" data-anchor-id="interaction-of-oxygen-with-myoglobin-and-hemoglobin"><span class="header-section-number">8</span> Interaction of oxygen with myoglobin and hemoglobin</h2>
<section id="context-7" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-7">Context:</h4>
<div class="line-block">The oxygen molecule can combine with the <em>heme</em> group of myoglobin and hemoglobin in different ways, depending on the cooperativity exhibited in the latter due to its quaternary structure. The following model exemplifies this interaction, using the <em>Hill equation</em>.</div>
</section>
<section id="equation-3" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-3">Equation:</h4>
<p><span class="math display">\[
Y= \frac{pO_2^{nH}}{p_{50}^{nH} + pO_2^{nH}}
\]</span><br>
</p>
<p><em>Where</em></p>
<ul>
<li>Y = degree of oxygen saturation in the protein;</li>
<li>pO<span class="math inline">\(_ {2}\)</span> = oxygen pressure;</li>
<li>p<span class="math inline">\(_{50}\)</span> = oxygen pressure at 50% saturation;</li>
<li>nH = Hill coefficient for the interaction;<br>
</li>
</ul>
<p><a href="Eq/jsp_O2.html" target="_blank"><img src="Eq/O2.png" class="img-fluid"></a><br>
<!---div class="jsplotly_sup-fundo"> ---></p>
</section>
<section id="suggestion-7" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-7">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Run the <span class="fu">application</span> (<span class="st">"add plot"</span>). Note that the value of ‘nH’ <span class="cf">for</span> the Hill constant is <span class="st">"1,"</span> meaning there is no cooperativity effect.</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Now replace the value of <span class="st">"nH"</span> with the Hill coefficient <span class="cf">for</span> hemoglobin, <span class="fl">2.8</span>, and run again<span class="sc">!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="bohr-effect-on-hemoglobin-ph" class="level2" data-number="9">
<h2 data-number="9" class="anchored" data-anchor-id="bohr-effect-on-hemoglobin-ph"><span class="header-section-number">9</span> Bohr effect on hemoglobin (pH)</h2>
<section id="context-8" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-8">Context:</h4>
<div class="line-block">Some physiological or pathological conditions can alter the binding affinity of oxygen to hemoglobin, such as temperature, certain metabolites (2,3-BPG), and the hydrogen ion content of the solution.</div>
</section>
<section id="equation-4" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-4">Equation:</h4>
<p><span class="math display">\[
Y(pO_2) = \frac{{pO_2^n}}{{P_{50}^n + pO_2^n}}, \quad \text{where } P_{50} = P_{50,\text{ref}} + \alpha (pH_{\text{ref}} - pH)
\]</span><br>
<em>Where</em>,</p>
<ul>
<li>Y = hemoglobin saturation,</li>
<li>pO₂ = partial pressure of oxygen (in mmHg),</li>
<li>P₅₀ = pressure of O₂ at which hemoglobin is 50% saturated,</li>
<li>P<span class="math inline">\(_{50,ref}\)</span> = 26 mmHg (standard value),</li>
<li><span class="math inline">\(\alpha\)</span> = 50 (intensity of the Bohr effect),</li>
<li>pH<span class="math inline">\(_{ref}\)</span> = 7.4,</li>
<li>n = 2.8 = Hill coefficient for hemoglobin.</li>
</ul>
<p><a href="Eq/jspl_bohr.html"><img src="Eq/jspl_bohr.png" class="img-fluid"></a></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try changing the reference pH <span class="cf">for</span> the interaction;</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Simulate other allosteric models by changing the value of <span class="st">"n"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="enzyme-catalysis-and-inhibition" class="level2" data-number="10">
<h2 data-number="10" class="anchored" data-anchor-id="enzyme-catalysis-and-inhibition"><span class="header-section-number">10</span> Enzyme catalysis and inhibition</h2>
<section id="context-9" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-9">Context:</h3>
<div class="line-block">The following simulation aims to provide a general equation for enzyme inhibition studies, covering <em>competitive, non-competitive, and competitive (pure or mixed) models</em>, also allowing the study of enzyme catalysis in the absence of an inhibitor.</div>
<section id="equation-5" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-5">Equation:</h4>
<p><span class="math display">\[
v=\frac{Vm*S}{Km(1+\frac{I}{Kic})+S(1+\frac{I}{Kiu})}
\]</span><br>
<em>Where</em></p>
<ul>
<li>S = substrate content for reaction;</li>
<li>Vm = reaction limit speed (in books, maximum speed);</li>
<li>Km = Michaelis-Mentem constant;</li>
<li>Kic = inhibitor dissociation equilibrium constant for competitive model;</li>
<li>Kiu = inhibitor dissociation equilibrium constant for noncompetitive model<br>
</li>
</ul>
<p><a href="Eq/jsp_kin_inhib.html" target="_blank"><img src="Eq/kin_inhib.png" class="img-fluid"></a><br>
<!---div class="jsplotly_sup-fundo"> ---></p>
</section>
</section>
<section id="suggestion-8" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-8">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="st">"A. Enzymatic catalysis in the absence of inhibitor."</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Just run the application with the general equation. Note that the values <span class="cf">for</span> Kic and Kiu are <span class="fu">high</span> (<span class="fl">1e20</span>). Thus, with high <span class="st">"dissociation constants,"</span> the interaction of the inhibitor with the enzyme is irrelevant, returning the model to the classic Michaelis<span class="sc">-</span>Mentem equation.</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try changing the values of Vm and Km, comparing graphs.</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Use the geographic coordinates feature on the icon <span class="fu">bar</span> (<span class="st">"Toggle Spike Lines"</span>) to consolidate the mathematical meaning of Km, as well as to observe the effect of different values on the graph visualization.</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="st">"B. Competitive inhibition model."</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> To observe or compare the Michaelis model with the competitive inhibition model, simply replace the Kic value with a consistent <span class="fu">number</span> (e.g., <span class="at">Kic=</span> <span class="dv">3</span>).</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="st">"C. Incompetitive inhibition model."</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> The same suggestion above applies to the incompetitive model, this time replacing the value <span class="cf">for</span> Kiu.</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="st">"D. Pure non-competitive inhibition model."</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> In this model, the simulation is based on equal values <span class="cf">for</span> Kic and Kiu.</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a><span class="st">"E. Mixed non-competitive inhibition model."</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> For this model, simply allocate different values <span class="cf">for</span> Kic and Kiu.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="linearization-of-kinetic-data-by-lineweaver-burk" class="level2" data-number="11">
<h2 data-number="11" class="anchored" data-anchor-id="linearization-of-kinetic-data-by-lineweaver-burk"><span class="header-section-number">11</span> Linearization of kinetic data by Lineweaver-Burk</h2>
<section id="context-10" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-10">Context</h3>
<div class="line-block">One of the (many) difficulties in learning biochemistry is the literacy required to interpret graphs. Among these, the hyperbolic nature of the Michaelis model is treated alternatively by various transformations of the variables <em>S</em> and <em>v</em>, with linearization by <em>double reciprocal</em>, or <em>Lineweaver-Burk</em> transformation, standing out in the literature.</div>
<div class="line-block">The following example illustrates the relationship between the dependent and independent variables of the original and transformed models, using another feature of the Plotly.js library: the possibility of cross-selection of points.</div>
<div class="line-block">To observe this functionality, select a set of points in one of the subplots of the graph below, and observe their automatic selection in the other subplot.</div>
</section>
<section id="equation-6" class="level3 unnumered" data-number="11.1">
<h3 class="unnumered anchored" data-number="11.1" data-anchor-id="equation-6"><span class="header-section-number">11.1</span> Equation</h3>
<div class="line-block">The Lineweaver-Burk linear transformation is given below:</div>
<p><span class="math display">\[
\frac{1}{v} = \frac{1}{S}*\frac{Km}{Vm} + \frac{1}{Vm}
\]</span><br>
</p>
<p><a href="Eq/jsp_LB.html" target="_blank"><img src="Eq/LB.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-9" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-9">Suggestion</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try selecting the extremes of the Michaelis<span class="sc">-</span>Mentem <span class="fu">graph</span> (hyperbolic curve), and observe the same points <span class="cf">in</span> the double reciprocal. Which values are more reliable <span class="cf">in</span> the latter, the first or the last?</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Research other forms of <span class="fu">linearization</span> (e.g., Eadie<span class="sc">-</span>Hofstee; <span class="st">"v/S"</span> X <span class="st">"v"</span>), and see how the transformation and selection of points presents itself. To do this, change the constants below<span class="sc">:</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>const invS <span class="ot">=</span> <span class="fu">S.map</span>(<span class="at">s =</span><span class="sc">&gt;</span> <span class="dv">1</span><span class="sc">/</span>s);</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>const invV <span class="ot">=</span> <span class="fu">V.map</span>(<span class="at">v =</span><span class="sc">&gt;</span> <span class="dv">1</span><span class="sc">/</span>v);</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="enzyme-inhibition-diagnosis-michaelis-mentem-and-lineweaver-burk" class="level2" data-number="12">
<h2 data-number="12" class="anchored" data-anchor-id="enzyme-inhibition-diagnosis-michaelis-mentem-and-lineweaver-burk"><span class="header-section-number">12</span> Enzyme Inhibition Diagnosis (Michaelis-Mentem and Lineweaver-Burk)</h2>
<section id="context-11" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-11">Context</h3>
<div class="line-block">As presented above and in the texts in this area, it is possible to linearize the Michaelis-Mentem curve for easier observation when diagnosing enzyme inhibition models. The object below proposes this interpretation. To do so, keep the <em>control enzyme</em> without adding inhibitor in the code (<em>I=0</em>). To search for an inhibition model, enter a value for <em>I</em> and change the inhibitor dissociation equilibrium constants, <em>kic</em> and/or <em>kiu</em>.</div>
<p><a href="Eq/jsp_inib_diag.html" target="_blank"><img src="Eq/inib_diag.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-10" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-10">Suggestion</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Simulate the conditions <span class="cf">for</span> competitive inhibition by entering a value <span class="cf">for</span> the inhibitor <span class="fu">content</span> (<span class="st">"I"</span>) and changing the value of <span class="st">"Km"</span>;</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Do the same <span class="cf">for</span> a non<span class="sc">-</span>competitive model, but change the value of <span class="st">"Vm"</span>;</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Try the pure non<span class="sc">-</span>competitive model by changing <span class="st">"Km"</span> and <span class="st">"Vm"</span> to the same value;</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Test mixed competitive inhibition by entering different values <span class="cf">for</span> <span class="st">"Km"</span> and <span class="st">"Vm"</span>.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="thermodynamic-stability-of-nucleic-acids" class="level2" data-number="13">
<h2 data-number="13" class="anchored" data-anchor-id="thermodynamic-stability-of-nucleic-acids"><span class="header-section-number">13</span> Thermodynamic stability of nucleic acids</h2>
<section id="context-12" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-12">Context:</h3>
<div class="line-block">Several factors contribute to the thermodynamic stability of biopolymers, either stabilizing or destabilizing them. This simulation deals with a thermal denaturation curve for DNA in the presence or absence of some of these compounds: trehalose (stabilizing osmolite) and guanidine chloride (destabilizing).</div>
</section>
<section id="equation-7" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-7">Equation:</h3>
<p><span class="math display">\[
y(T) = \frac{1}{1 + e^{-\frac{T - T_m}{\beta}}}
\]</span></p>
<p><em>Where</em>,</p>
<ul>
<li>y(T): fraction of denatured DNA at a given temperature T;</li>
<li>Tm: transition temperature (<em>melting</em>, temperature at which 50% of molecules are double-stranded and 50% are single-stranded);</li>
<li><span class="math inline">\(\beta\)</span>: parameter that adjusts the slope of the curve (affected by trehalose and guanidine).</li>
</ul>
<p><em>Note:</em></p>
<ol type="1">
<li>Trehalose as a stabilizer (reduces <span class="math inline">\(\beta\)</span>);</li>
<li>Guanidine as a denaturant (increases <span class="math inline">\(\beta\)</span>);</li>
</ol>
<p>[<img src="Eq/denat_DNA.png" class="img-fluid">] (Eq/jsp_denat_DNA_no_osmol.html){target=“_blank”}</p>
</section>
<section id="suggestion-11" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-11">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try testing various conditions involved <span class="cf">in</span> the simulation, such as<span class="sc">:</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>a<span class="er">)</span> variation of Tm;</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>b<span class="er">)</span> variation of the <span class="sc">$</span>\beta<span class="sc">$</span> parameter;</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>c<span class="er">)</span> variation of trehalose content;</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>d<span class="er">)</span> variation of guanidine chloride content.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="van-der-waals-equation-for-ideal-gases" class="level2" data-number="14">
<h2 data-number="14" class="anchored" data-anchor-id="van-der-waals-equation-for-ideal-gases"><span class="header-section-number">14</span> Van der Waals equation for ideal gases</h2>
<section id="context-13" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-13">Context:</h3>
<div class="line-block">An adaptation that relates the thermodynamic quantities of pressure, volume, and temperature for ideal gases is the <em>van der Waals equation</em>. In this equation, coefficients are computed that estimate the existence of a volume and interparticle interactions. Thus, the van der Waals equation corrects the ideal gas equation by considering a term to compensate for intermolecular forces (a/V<span class="math inline">\(^{2}\)</span>) and the available volume, discounting the volume occupied by the gas molecules themselves.</div>
<div class="line-block">The simulation offers additional interactivity through the presence of sliders for temperature and for the finite volume coefficient (<em>b</em>) and particle interaction coefficient (<em>a</em>).</div>
</section>
<section id="equation-8" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-8">Equation:</h3>
<p><span class="math display">\[
P = \frac{RT}{V - b} - \frac{a}{V^2}
\]</span></p>
<ul>
<li>P = gas pressure (atm);</li>
<li>V = molar volume (L);</li>
<li>T = temperature (K);</li>
<li>R = 0.0821 = ideal gas constant (L·atm/mol·K);</li>
<li>a = intermolecular attraction constant (L<span class="math inline">\(^{2}\)</span>·atm/mol$^{2})</li>
<li>b = excluded volume constant (L/mol)</li>
</ul>
<p><a href="Eq/jsp_vanderWaals.html" target="_blank"><img src="Eq/vanderWaals.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-12" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-12">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try varying the parameters of the equation using the slider <span class="cf">for</span> temperature, as well as <span class="cf">for</span> coefficients <span class="st">"a and b"</span>.</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Discuss which of the coefficients has the greatest effect on the curve profile, and the reason <span class="cf">for</span> this.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="balance-of-atp-production-from-reagents-temperature-and-ph" class="level2" data-number="15">
<h2 data-number="15" class="anchored" data-anchor-id="balance-of-atp-production-from-reagents-temperature-and-ph"><span class="header-section-number">15</span> Balance of ATP production from reagents, temperature, and pH</h2>
<section id="context-14" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-14">Context:</h3>
<div class="line-block">Intracellular ATP production involves the classic chemical equilibrium relationship between reactants and products as a function of reaction temperature, adjusted to a specific pH value. By varying one or both reactant contents, or physical-chemical parameters, it is possible to quantify the product by the following thermodynamic reaction:</div>
</section>
<section id="equation-9" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-9">Equation:</h3>
<p><span class="math display">\[
\Delta G = \Delta G^{\circ'} + RT \ln\left(\frac{[\text{ADP}] \cdot [\text{P}_i]}{[\text{ATP}]}\right) + 2{,}303 \cdot RT \cdot n_H \cdot \text{pH}
\]</span></p>
<p><em>Where</em>,</p>
<ul>
<li><span class="math inline">\(\Delta\)</span>G = Gibbs energy of the reaction (positive for spontaneously unfavorable synthesis, kJ/mol);</li>
<li><span class="math inline">\(\Delta\)</span>G<span class="math inline">\(^{o'}\)</span> = 30.5 kJ/mol standard biological Gibbs energy for ATP synthesis;</li>
<li>R = 8.314 J/mol/K (universal gas constant);</li>
<li>T=310 K (physiological temperature);</li>
<li>nH<span class="math inline">\(^ {+}\)</span> = 1 (number of protons involved in the reaction);</li>
<li>[ADP], [Pi], [ATP] = molar concentrations of reactants and product</li>
</ul>
<p><a href="Eq/jsp_ATP.html" target="_blank"><img src="Eq/ATP.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-13" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-13">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Change the quantities involved <span class="cf">in</span> the expression and compare with previous visualizations. For example, temperature, pH, and ADP and Pi levels.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="variation-of-gibbs-energy-with-temperature" class="level2" data-number="16">
<h2 data-number="16" class="anchored" data-anchor-id="variation-of-gibbs-energy-with-temperature"><span class="header-section-number">16</span> Variation of Gibbs energy with temperature</h2>
<section id="context-15" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-15">Context:</h3>
<div class="line-block">The Gibbs-Helmholtz relation predicts that Gibbs energy varies nonlinearly with temperature in reactions involving a change in the heat capacity of the system (<span class="math inline">\(\Delta\)</span>Cp). This expanded form of the Gibbs equation for variable heat capacity is present in various biochemical phenomena, such as in the phase transition of biomembrane structures subjected to a challenge from compounds, or in the conformational change that accompanies the protein structure under heating. \</div>
</section>
<section id="equation-10" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-10">Equation:</h3>
<p><span class="math display">\[
\Delta G(T) = \Delta H^\circ - T\,\Delta S^\circ + \Delta C_p \left(T - T_0 - T \ln\left(\frac{T}{T_0}\right)\right)
\]</span></p>
<p><em>Where</em>,</p>
<ul>
<li><span class="math inline">\(\Delta\)</span>G(T) = Gibbs energy of the reaction at each temperature value, kJ/mol);</li>
<li><span class="math inline">\(\Delta\)</span>H<span class="math inline">\(^{o}\)</span> = standard enthalpy of the reaction at T<span class="math inline">\(_{0}\)</span>, usually 298 K (J/mol);</li>
<li><span class="math inline">\(\Delta\)</span>S<span class="math inline">\(^{o}\)</span> = standard entropy of the reaction at T<span class="math inline">\(_{0}\)</span>;</li>
<li><span class="math inline">\(\Delta\)</span>Cp = change in heat capacity of the reaction (J/mol·K), assumed to be constant with temperature;</li>
<li>T = temperature of interest (K);</li>
<li>T<span class="math inline">\(_{0}\)</span> = reference temperature, usually 298 K.</li>
<li>R = 8.314 J/mol/K (universal gas constant);</li>
</ul>
<p><a href="Eq/jsp_gibbs.html" target="_blank"><img src="Eq/gibbs.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-14" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-14">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try varying one or more parameters of the expression;</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Test the behavior of the Gibbs curve at a high reference <span class="fu">temperature</span> (simulation <span class="cf">for</span> extremophile organism);</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Simulate the situation where the variation <span class="cf">in</span> heat capacity is zero</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="electrochemical-impedance-spectroscopy-eis" class="level2" data-number="17">
<h2 data-number="17" class="anchored" data-anchor-id="electrochemical-impedance-spectroscopy-eis"><span class="header-section-number">17</span> Electrochemical impedance spectroscopy (EIS)</h2>
<section id="context-16" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-16">Context:</h3>
<div class="line-block">Electrochemical impedance spectroscopy (EIS) is an analytical technique that measures the response of an electrochemical system to a small amplitude sinusoidal disturbance (e.g., 5-10 mV), applied over a wide frequency range (e.g., 100 kHz-1mHz). EIS provides information about interfacial processes, such as charge transfer, ion diffusion, and passive film formation, through the analysis of the complex impedance of the system. It is a non-destructive technique widely used in areas such as corrosion, batteries, fuel cells, biosensors, and protective coatings.</div>
</section>
<section id="equation-11" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-11">Equation:</h3>
<div class="line-block">An electrochemical impedance (EIS) model with typical equivalent circuit components constitutes the modified Randles model: series resistor <em>Rs</em>, a resistor <em>Rp</em> in parallel with a <em>CPE</em> (Constant Phase Element), and a <em>Warburg</em> element. A common representation for EIS is the <em>Nyquist</em> plot, where the real axis represents resistance (Zre) and the imaginary axis represents negative reactance (−Zim). The formalism involved is in the equation below:</div>
<p><span class="math display">\[
Z_{\text{total}}(\omega) = R_s + \left[ \left( \frac{1}{R_p} + Q (j\omega)^n \right)^{-1} \right] + \frac{\sigma}{\sqrt{\omega}} (1 - j)
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>Z<span class="math inline">\(_{total}\)</span> = total impedance at a given frequency;</li>
<li><span class="math inline">\(\omega\)</span> = Angular frequency (rad/s), <span class="math inline">\(\omega\)</span>=2πf or <span class="math inline">\(\omega\)</span>=2πf;</li>
<li>R<span class="math inline">\(_{p}\)</span> = Ohmic resistance (resistance of the electrolytic solution, wires, contacts, etc.);</li>
<li>R<span class="math inline">\(_{p}\)</span> = Polarization resistance (associated with charge transfer processes, such as electrochemical reactions);</li>
<li><em>Q</em> = Constant associated with the constant phase element (CPE), replaces an ideal capacitor to represent non-ideal behaviors;</li>
<li><em>n</em> = CPE exponent, between 0 and 1; defines the degree of ideality of the capacitive behavior (n = 1: ideal capacitor; n &lt; 1: dispersion);</li>
<li><span class="math inline">\(\sigma\)</span> = Warburg coefficient, associated with ion diffusion in the electrochemical system;</li>
<li><em>j</em> = Imaginary unit, j<span class="math inline">\(^ {2}\)</span> = −1;</li>
</ul>
<div class="line-block">An electrochemical impedance model (EIS) with typical equivalent circuit components constitutes the modified Randles model: series resistor <em>Rs</em>, a resistor <em>Rp</em> in parallel with a <em>CPE</em> (Constant Phase Element), and a <em>Warburg</em> element. A common representation for EIS is the <em>Nyquist</em> plot, where the real axis represents resistance (Zre) and the imaginary axis represents negative reactance (−Zim).</div>
<p><a href="Eq/jsp_EIS.html" target="_blank"><img src="Eq/EIS.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-15" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-15">Suggestion</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Check the effect of Rs on the graph by canceling its <span class="fu">value</span> (solution resistance);</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Observe the deformation of the semicircle by varying the values of the constant phase <span class="fu">element</span> (e.g., <span class="at">Q =</span> <span class="fl">1e-3</span>; <span class="at">n =</span> <span class="fl">0.6</span> <span class="sc">-</span> dispersion of capacitive behavior);</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try combining other values of the code header parameters to highlight common situations <span class="cf">in</span> electroanalysis<span class="sc">:</span> Rs, Rp, Q, n, and sigma;</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Reduce the Warburg model with constant phase element to a simple Randles model, consisting of only two resistors <span class="cf">in</span> series, the second <span class="cf">in</span> parallel with an ideal capacitor, and without Warburg <span class="fu">diffusion</span> (<span class="at">sigma =</span> <span class="dv">0</span>; <span class="at">n =</span> <span class="dv">1</span>).</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="cyclic-voltammetry" class="level2" data-number="18">
<h2 data-number="18" class="anchored" data-anchor-id="cyclic-voltammetry"><span class="header-section-number">18</span> Cyclic voltammetry</h2>
<section id="context-17" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-17">Context:</h3>
<div class="line-block">Cyclic voltammetry (CV) is an electroanalytical technique for characterizing interfacial (electrode-solution) redox processes. When a standard analyte in quantitative equilibrium between the oxidized and reduced forms is used, the resulting graph acquires an unusual profile, called a <em>duck shape</em>. This results from the peculiar shape obtained from a bidirectional potentiometric scan over a set of electrodes by a <em>potentiostat</em>, resulting in cycles of oxidation and subsequent reduction (application of positive potentials followed by negative potentials). The shape, value, and distance between the resulting peaks allow the analyte to be identified and characterized, as well as the redox processes involved in the experiment.</div>
</section>
<section id="equation-12" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-12">Equation:</h3>
<div class="line-block">Although the formalism for cyclic voltammetry is complex and involves solving <em>ordinary differential equations</em> (as well as electroanalytical techniques in general), which is beyond the scope of this material, the resulting current flow can be summarized by the following <em>Buttler-Volmer equation</em>:</div>
<p><span class="math display">\[
j = j_0 \left[ \exp\left(\frac{\alpha n F (E - E^0)}{RT}\right) - \exp\left(\frac{-(1 - \alpha) n F (E - E^0)}{RT}\right) \right]
\]</span> <em>Where:</em></p>
<ul>
<li>j = current density;</li>
<li>j<span class="math inline">\(_{0}\)</span> = exchange current;</li>
<li><span class="math inline">\(\alpha\)</span> = charge transfer coefficient;</li>
<li>E = potential applied to the electrode;</li>
<li>E<span class="math inline">\(^{0}\)</span> = standard electrode potential;</li>
<li>n = number of electrons;</li>
<li>F = Faraday constant (96485 C·mol⁻¹)</li>
<li>R = universal gas constant (8.314 J·mol⁻¹·K⁻¹);</li>
<li>T = temperature<br>
</li>
</ul>
<div class="line-block">For a model of electroanalyte diffusion and accumulation at the electrode surface, the general equation for the boundary condition where the redox conversion rate is proportional to the diffusion current can be represented by <em>Fick’s 2nd Law equation </em> for mass transport:</div>
<p><span class="math display">\[
\frac{\partial C(x,t)} {\partial t} = D \frac{\partial^2 C(x,t)} {\partial x^2}
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>C(x,t) = concentration of the electroanalytical species (mol/cm³), as a function of position <em>x</em> and time <em>t</em>;</li>
<li>D = diffusion coefficient of the species (cm²/s);</li>
<li>t = time (s);</li>
<li>x = distance from the electrode surface (cm);</li>
<li><span class="math inline">\(\delta\)</span> = notation for partial derivative.<br>
</li>
</ul>
<div class="line-block">The chemical equilibrium of the redox reaction is given by the ratio between the concentrations of the oxidized and reduced forms at the electrode/solution interface when the system is in equilibrium (or quasi-equilibrium), according to the <em>Nernst equation</em>:</div>
<p>$$</p>
<p>E = E^0 + ( ) $$</p>
<div class="line-block">The resulting current is obtained by Faraday’s law:</div>
<p><span class="math display">\[
i(t) = n F A D \left. \frac{\partial C(x,t)} {\partial x} \right|_{x=0}
\]</span> <em>Where:</em></p>
<ul>
<li>A = electrode surface area (cm²);</li>
<li>i(t) = current at time <em>t</em> (A, Ampere).</li>
</ul>
</section>
<section id="cyclic-voltametry-simulator" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="cyclic-voltametry-simulator">Cyclic Voltametry Simulator</h3>
<div class="line-block">To illustrate the potential of the <em>JSPlotly/GSPlotly</em> platform for developing complex <em>virtual experiments</em>, here is a <em>cyclic voltametry simulator</em> resulting from the above procedures (as well as a few hours of prompt engineering!!).</div>
<div class="line-block">The example illustrates typical behavior in cyclic voltammetry (and the curious <em>duck shape</em>), and includes a model of diffusion and accumulation of electroactive species near the electrode surface. The code uses a <em>numerical solution</em> to obtain the current density over time, considering the effects of cyclic scanning of the applied potential, mass transport, and reaction kinetics.</div>
<div class="line-block">In short, it numerically solves Fick’s equation with dynamic boundary conditions, applies the Nernst equation at the boundary (electrode surface) to determine the conversion between species, uses the finite difference method or explicit scheme to advance in time, and obtains the current by applying Faraday’s Law, by the derivative of the flux at the interface.</div>
<div class="line-block">Did you find it complex? Then how about this: the code combines Fick (transport) and Nernst (local redox equilibrium) to accurately model the shape of the voltammogram, obtaining a physical simulation with significant realism for cyclic voltammetry.</div>
<p><br>
</p>
<p><a href="Eq/jsp_cv.html" target="_blank"><img src="Eq/cv.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-16" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-16">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Observe the number of values at the beginning of the code, tangible to <span class="st">"parametric manipulation."</span> Try to understand what they represent, and consciously vary their values, aiming to add value to the learning of the simulation. This is the essence of <span class="st">"parametric manipulation"</span> that involves <span class="st">"reproducible teaching"</span><span class="sc">!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="diagrams-and-flowcharts" class="level2" data-number="19">
<h2 data-number="19" class="anchored" data-anchor-id="diagrams-and-flowcharts"><span class="header-section-number">19</span> Diagrams and flowcharts</h2>
<section id="context-diagram" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-diagram">Context: Diagram</h3>
<div class="line-block">Due to the nature of the <em>Plotly.js</em> library, it is also possible to generate code in <em>JSPlotly</em> without involving an equation, such as for representing diagrams and flowcharts, among others. The example below illustrates a representation of the Krebs cycle, and the following is an experimental flowchart.</div>
<p><a href="Eq/jsp_krebs.html" target="_blank"><img src="Eq/krebs.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-17" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-17">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try repositioning enzymes and metabolites by clicking and dragging the terms;</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try replacing the names <span class="cf">in</span> the code to produce another metabolic cycle, such as the urea cycle.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="context-flowchart" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-flowchart">Context: Flowchart</h3>
<p><a href="Eq/jsp_fluxograma.html" target="_blank"><img src="Eq/fluxograma.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-18" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-18">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Reposition terms and connectors by dragging them with the mouse;</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> For a different flowchart <span class="cf">in</span> the content, modify the terms <span class="cf">in</span> the code;</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> For a different flowchart <span class="cf">in</span> the format, change the font and connector characteristics <span class="cf">in</span> the <span class="st">"annotations"</span> constant.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
<br></p>
</section>
</section>
<section id="data-analysis" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="data-analysis"><strong>Data Analysis</strong></h2>
</section>
<section id="user-data-insertion" class="level2" data-number="20">
<h2 data-number="20" class="anchored" data-anchor-id="user-data-insertion"><span class="header-section-number">20</span> User data insertion</h2>
<section id="context-18" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-18">Context</h3>
<div class="line-block">Sometimes it is interesting to work with discrete data itself, seeking a trend, mathematical or simply visual relationship for the relationship between variables. Here is an example of an interactive graph with user data.</div>
<p><br>
</p>
<p><a href="Eq/jsp_data.html" target="_blank"><img src="Eq/data.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-19" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-19">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try changing the data entered, overlaying the graph or not;</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try changing the data representation <span class="cf">in</span> <span class="st">"mode"</span> and <span class="st">"type"</span> to points, lines, points<span class="sc">+</span>lines, bars.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="loading-a-file-for-analysis" class="level2" data-number="21">
<h2 data-number="21" class="anchored" data-anchor-id="loading-a-file-for-analysis"><span class="header-section-number">21</span> Loading a file for analysis</h2>
<section id="context---csv-file" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context---csv-file">Context - CSV file</h3>
<div class="line-block">Sometimes, it is necessary to analyze results related to a data set in a spreadsheet. For this situation, the following example illustrates the loading of two data vectors from a <em>CSV</em> (<em>comma separated value</em>) file, a format commonly used in exporting/importing spreadsheets, and its resulting graph.</div>
<p><a href="Eq/jsp_csv.html" target="_blank">! [](Eq/csv.png)</a></p>
<p><strong>Instructions:</strong></p>
<ul>
<li>1 Click <em>add plot</em> and select a CSV file from the <em>browse</em> button at the top. Note: variable <em>X</em> in the 1st column of the file, and variable <em>Y</em> in the 2nd column;</li>
<li><ol start="2" type="1">
<li>Click <em>add plot</em> again to view the resulting graph.</li>
</ol></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try other CSV files;</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Vary the aspects of the graph, such as type, color, marker size, etc.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="linear-data-adjustment" class="level2" data-number="22">
<h2 data-number="22" class="anchored" data-anchor-id="linear-data-adjustment"><span class="header-section-number">22</span> Linear data adjustment</h2>
<section id="context-19" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-19">Context:</h3>
<div class="line-block">In various experimental research situations, it is necessary to correlate the data of a dependent variable with a predictor variable using the following equation.</div>
</section>
<section id="equation-13" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-13">Equation:</h3>
<p><span class="math display">\[
y = \alpha x + \beta + \varepsilon
\]</span> <em>Where:</em></p>
<ul>
<li>y = dependent variable;</li>
<li>x = independent variable;</li>
<li><span class="math inline">\(\alpha\)</span> = slope of the adjusted line;</li>
<li><span class="math inline">\(\beta\)</span> = intercept of the adjusted line;</li>
<li><span class="math inline">\(\epsilon\)</span> = measurement error.</li>
</ul>
<div class="line-block">The <em>slope and intercept</em> parameters can be obtained through several procedures, such as <em>least squares</em>, <em>matrix algebra</em>, or simple <em>sums</em>. In the latter case, the parameters and error can be calculated as follows:</div>
<p><span class="math display">\[
\alpha = \frac{n \sum x_i y_i - \sum x_i \sum y_i}{n \sum x_i^2 - \left( \sum x_i \right)^2}
\]</span></p>
<p><span class="math display">\[
\beta = \frac{\sum y_i - \alpha \sum x_i}{n}
\]</span></p>
<p><span class="math display">\[
\hat{y}_i = \alpha x_i + \beta
\]</span></p>
<p><span class="math display">\[
\varepsilon_i = \left| y_i - \hat{y}_i \right|
\]</span></p>
<div class="line-block">A representation for linear adjustment with arbitrary data is provided below:</div>
<p><br>
</p>
<p><a href="Eq/jsp_ajLinR2.html" target="_blank"><img src="Eq/ajLin.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-20" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-20">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Show the points overlapping or not overlapping the fit line. For overlap, choose <span class="st">"showPoints = true"</span> and <span class="st">"showLine = true"</span>;</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Change the data and perform a new fit to obtain other parameters of the line.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="polynomial-regression" class="level2" data-number="23">
<h2 data-number="23" class="anchored" data-anchor-id="polynomial-regression"><span class="header-section-number">23</span> Polynomial regression</h2>
<section id="context-20" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-20">Context:</h3>
<div class="line-block">In some situations, linear adjustment does not provide an adequate trend for modeling experimental data, which leads to the use of other models, such as <em>exponential, hyperbolic, logarithmic, power, and polynomial</em>. Polynomial trends are common in natural phenomena, with mathematical treatment of the data occurring by different methods, such as simple sums, least squares, or linear algebra. Here is an example of polynomial regression with adjustable degrees in the code.</div>
</section>
<section id="equation-14" class="level3" data-number="23.1">
<h3 data-number="23.1" class="anchored" data-anchor-id="equation-14"><span class="header-section-number">23.1</span> Equation</h3>
<div class="line-block">Error minimization for the polynomial curve is obtained by generalized <em>least squares regression</em> and application of the <em>Vandermonde matrix</em>, as follows.</div>
<div class="line-block">Let the vectors be x=[x<span class="math inline">\(_{1}\)</span>,x<span class="math inline">\(_{2}\)</span>,…,x<span class="math inline">\(_{n}\)</span>] and y=[y<span class="math inline">\(_{1}\)</span>,y<span class="math inline">\(_{2}\)</span>,…,y<span class="math inline">\({_n}\)</span>], we want to adjust the polynomial:</div>
<p><span class="math display">\[
y = \beta_0 + \beta_1 x + \beta_2 x^2 + \cdots + \beta_g x^g
\]</span> | Thus, the Vandermonde matrix is constructed as:</p>
<p><span class="math display">\[
X =
\begin{bmatrix}
1 &amp; x_1 &amp; x_1^2 &amp; \cdots &amp; x_1^g \\\\
1 &amp; x_2 &amp; x_2^2 &amp; \cdots &amp; x_2^g \\\\
\vdots &amp; \vdots &amp; \vdots &amp; &amp; \vdots \\\\
1 &amp; x_n &amp; x_n^2 &amp; \cdots &amp; x_n^g
\end{bmatrix}
\]</span></p>
<div class="line-block">To find the <em>coefficients <span class="math inline">\(\beta\)</span></em> that minimize the quadratic error, the linear system is solved in sequence:</div>
<p><span class="math display">\[
\boldsymbol{\beta} = (X^T X)^{-1} X^T \mathbf{y}
\]</span><br>
</p>
<p><em>Where:</em></p>
<ul>
<li><em>T</em> represents the transposed matrix<br>
</li>
</ul>
<p><a href="Eq/jsp_polynomial.html" target="_blank"><img src="Eq/polynomial.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-21" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-21">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try degree <span class="dv">1</span> <span class="cf">for</span> the polynomial, i.e., a reduction of the treatment to linear adjustment;</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Change the formatting of labels, colors, sizes, etc., <span class="cf">in</span> the code;</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Overlap some adjustments, edit, and reposition the legend;</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Test the code with another data vector.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="multilinear-regression" class="level2" data-number="24">
<h2 data-number="24" class="anchored" data-anchor-id="multilinear-regression"><span class="header-section-number">24</span> Multilinear regression</h2>
<section id="context-21" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-21">Context:</h3>
<div class="line-block">Multiple linear adjustment is very useful when you want to predict behavior based on two or more predictor variables. The procedure involves a <em>projection matrix</em> (or <em>projection operator</em>) similar to the <em>Vandermonde matrix</em>.</div>
</section>
<section id="equation-15" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-15">Equation:</h3>
<div class="line-block">The analytical solution for the design matrix involves solving a matrix of coefficients <span class="math inline">\(\beta\)</span> for a set of predictor values of the response vector <em>y</em>, as follows:</div>
<p><span class="math display">\[
y_i = \beta_0 + \beta_1 x_{1i} + \beta_2 x_{2i} + \cdots + \beta_p x_{pi} + \varepsilon_i
\]</span></p>
<p><span class="math display">\[
\mathbf{y} = \mathbf{X} \boldsymbol{\beta} + \boldsymbol{\varepsilon}
\]</span></p>
<p><span class="math display">\[
\hat{\boldsymbol{\beta}} = (\mathbf{X}^\top \mathbf{X})^{-1} \mathbf{X}^\top \mathbf{y}
\]</span></p>
<p><em>Where:</em> - <span class="math inline">\(\beta\)</span> = coefficient vector; - <em>y</em> = response vector; - <em>X</em> = design matrix; - <span class="math inline">\(\epsilon\)</span> = random noise</p>
<div class="line-block">Thus, predicted values and associated error are obtained by:</div>
<p><span class="math display">\[
\hat{\mathbf{y}} = \mathbf{X} \hat{\boldsymbol{\beta}}
\]</span> <span class="math display">\[
\mathbf{e} = \mathbf{y} - \hat{\mathbf{y}}
\]</span></p>
<div class="line-block">Obviously, the graphical representation for a multilinear fit is limited to two independent variables, <em>x1</em> and <em>x2</em>, in addition to the response variable (<em>3D graph</em>), although the matrix calculation allows for several predictor variables (hyperdimensional space). Tangent to the first situation, here is an example.</div>
<p><strong>Instructions</strong> - 1. Note that there is a Boolean flag (<em>showFit</em>) at the beginning of the code: <em>false</em> for data only, and <em>true</em> for the fit. - 2. You can click on <em>add plot</em> to view the data with the flag set to <em>false</em>, followed by another <em>add plot</em> with the flag set to <em>true</em>.<br>
</p>
<p><a href="Eq/jsp_multilinear.html" target="_blank"><img src="Eq/multilinear.png" class="img-fluid"></a><br>
</p>
<div class="line-block">Rotate the graph and observe the surface around the points. Unlike a linear fit, the points are distributed discontinuously in space.</div>
</section>
<section id="suggestion-22" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-22">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> <span class="fu">QSAR</span> (<span class="st">"Quantitative Structure-Activity Relationship"</span>) results use multilinear adjustment analysis to identify the strength of predictor variables. Try a dataset on the internet that has two <span class="fu">variables</span> (e.g., concentration, pH, compound A, B, etc.).</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> For more predictor variables, see the code below <span class="sc">!!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="context-multiple-linear-adjustment-with-3-or-more-predictor-variables" class="level3" data-number="24.1">
<h3 data-number="24.1" class="anchored" data-anchor-id="context-multiple-linear-adjustment-with-3-or-more-predictor-variables"><span class="header-section-number">24.1</span> Context: Multiple linear adjustment with 3 or more predictor variables</h3>
<div class="line-block">Although <em>JSPlotly</em> was designed to produce interactive graphs and maps using the <em>Plotly.js</em> library on top of the <em>JavaScript</em> language, it extends the possibilities of application beyond graphical representations. An example of this permeates the following code, for a <em>multilinear adjustment</em> for a situation with 4 predictor variables.</div>
<p><a href="Eq/jsp_multilin4Var.html" target="_blank"><img src="Eq/multilin4Var.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-23" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-23">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try varying the number of <span class="fu">predictors</span> (xi). To do this<span class="sc">:</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>a. If you want to reduce it, simply remove the desired <span class="fu">vector</span>(s) and correct its quantity <span class="cf">in</span> the line<span class="sc">:</span> <span class="st">"const X = x1.map((_, i)"</span>;</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>b. If you want to increase it, add a new vector <span class="cf">in</span> the corresponding line and update the quantity <span class="cf">in</span> the mapping of <span class="st">"const X = x1.map((_, i)"</span>.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="response-surface-methodology-rsm" class="level2" data-number="25">
<h2 data-number="25" class="anchored" data-anchor-id="response-surface-methodology-rsm"><span class="header-section-number">25</span> Response Surface Methodology (RSM)</h2>
<div class="line-block">A multilinear adjustment variation involves the <em>response surface methodology</em>, commonly linear (flat) or quadratic (curvilinear), applied in modeling and optimization of experimental processes with multiple independent variables. Its objective is to adjust a mathematical equation that represents the relationship between the experimental factors and the response variable, allowing behaviors to be predicted and optimal conditions to be identified.</div>
<div class="line-block">For experiments with two factors, a second-order quadratic model is commonly used, which allows for the representation of curved surfaces. The central idea of the experiment is to select values and their codes (-1,0,1), representing, respectively, a quantity below, an average quantity, and a quantity above, for a given predictor variable.</div>
<section id="equation-16" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-16">Equation:</h3>
<div class="line-block">The general quadratic model for 2 factors is represented by:</div>
<p><span class="math display">\[
y = \beta_0 + \beta_1 x_1 + \beta_2 x_2 + \beta_{11} x_1^2 + \beta_{22} x_2^2 + \beta_{12} x_1 x_2 + \varepsilon
\]</span></p>
<div class="line-block">With the matrix form of the model represented by:</div>
<p><span class="math display">\[
\mathbf{y} = \mathbf{X} \boldsymbol{\beta} + \boldsymbol{\varepsilon}
\]</span></p>
<div class="line-block">And the estimation of the coefficients <span class="math inline">\(\beta\)</span> by least squares, similar to multilinear adjustment, by:</div>
<p><span class="math display">\[
\hat{\boldsymbol{\beta}} = (\mathbf{X}^\top \mathbf{X})^{-1} \mathbf{X}^\top \mathbf{y}
\]</span><br>
</p>
<p><a href="Eq/jsp_msr.html" target="_blank"><img src="Eq/msr.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-24" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-24">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> As suggested above, try using data from the literature or another source whose responses are already known, replacing the respective vectors at the beginning of the code. This allows you to compare the effectiveness of using the tool presented.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="data-smoothing---spline-and-savitzky-golay-filter" class="level2" data-number="26">
<h2 data-number="26" class="anchored" data-anchor-id="data-smoothing---spline-and-savitzky-golay-filter"><span class="header-section-number">26</span> Data smoothing - Spline and Savitzky-Golay filter</h2>
<div class="line-block">In experimental research, signal detection occasionally occurs in the presence of significant levels of noise, as occurs in continuous spectral scanning under wavelength variation, or in the detection of electrical signals over time. In this case, there is no analytical equation that describes the behavior of the signal, although some digital processing tools can rely on a trend curve in data processing, significantly reducing the noise level, such as by <em>spline curve</em> and <em>Savitzky-Golay filter</em>.</div>
<section id="context---cubic-spline" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context---cubic-spline">Context - Cubic spline</h3>
<div class="line-block">A very common interpolation technique involves the use of cubic splines by degree 3 polynomials interconnecting data windows. The treatment consists of applying polynomial functions in subintervals of the points, promoting smoothing between data windows. Mathematically, the spline is applied in such a way that:</div>
<p><span class="math display">\[
S_i(x) = a_i + b_i(x - x_i) + c_i(x - x_i)^2 + d_i(x - x_i)^3
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>S<span class="math inline">\(_{i}\)</span>(x) = cubic spline function in the interval <em>[xi,xi+1]</em>;</li>
<li>a<span class="math inline">\(_{i, b\)</span><em>{i}<span class="math inline">\(, c\)</span></em>{i}<span class="math inline">\(, d\)</span>_{i}$ = specific coefficients for segment <em>i</em>;</li>
<li>x<span class="math inline">\(_{i}\)</span> = starting point of the interval;</li>
<li><em>x</em> = independent variable.</li>
</ul>
</section>
<section id="context---savitzky-golay-filter" class="level3" data-number="26.1">
<h3 data-number="26.1" class="anchored" data-anchor-id="context---savitzky-golay-filter"><span class="header-section-number">26.1</span> Context - Savitzky-Golay filter</h3>
<div class="line-block">The Savitzky-Golay filter smoothing technique applies a local polynomial regression to a moving window, using least squares and a predetermined degree, preserving peak, valley, and band signals. Unlike the cubic spline, the filter produces a continuous global curve, rather than a set of interconnected curves. Mathematically:</div>
<p><span class="math display">\[
\tilde{y}_i = \sum_{j=-m}^{m} c_j \cdot y_{i+j}
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>y<span class="math inline">\(_{i}\)</span> = smoothed value at position <em>i</em>;</li>
<li>y<span class="math inline">\(_ {i+j}\)</span> = actual values of the series within the window;</li>
<li>c<span class="math inline">\(_{i}\)</span> = filter coefficients derived from a polynomial regression;</li>
<li><em>m</em> = number of points on each side of the central window.</li>
</ul>
<div class="line-block">Here is an example of code for <em>JSplotly</em> that illustrates the use of the two interpolation methods.</div>
<p><a href="Eq/jsp_splineGolay.html" target="_blank"><img src="Eq/splineGolay.png" class="img-fluid"></a><br>
</p>
<p><strong>Instructions:</strong></p>
<ul>
<li><ol type="1">
<li>The code has two <em>true/false flags</em>, one for “useGolay” and another for “useSpline”;</li>
</ol></li>
<li><ol start="2" type="1">
<li>View the raw data by setting both flags to <em>false</em>;</li>
</ol></li>
<li><ol start="3" type="1">
<li>To overlay a cubic spline, change its constant to <em>true</em>;</li>
</ol></li>
<li><ol start="4" type="1">
<li>To overlay the Savitzky-Golay filter, change its constant to <em>true</em>;</li>
</ol></li>
<li><ol start="5" type="1">
<li>To adjust the filter, modify its parameters in the constants <em>window</em> (sliding window) and/or <em>degree</em> (degree of the polynomial).</li>
</ol></li>
</ul>
</section>
<section id="suggestion-25" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-25">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Compare the smoothing effects of the two polynomial interpolation treatments;</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Apply only the filter, adjusting the moving window and polynomial degree parameters;</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Try interpolation with other raw data <span class="cf">in</span> the code, replacing the constants <span class="st">"x_values"</span> and <span class="st">"y_raw"</span> constants with numerical vectors;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="anova-and-tukey-hsd-test" class="level2" data-number="27">
<h2 data-number="27" class="anchored" data-anchor-id="anova-and-tukey-hsd-test"><span class="header-section-number">27</span> ANOVA and Tukey-HSD test</h2>
<section id="context-22" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-22">Context:</h3>
<div class="line-block">In research, it is common to analyze a data set by comparing the variance between groups with that obtained within each group, or <em>Analysis of Variance (ANOVA)</em>. Once statistical information has been obtained showing that there is a difference between groups greater than the internal difference, a <em>post-hoc</em> test is performed to compare means, such as the <em>Tukey-HSD test</em> (<em>Honest Significant Difference</em>).</div>
</section>
<section id="equation-17" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-17">Equation:</h3>
<div class="line-block">The equation that characterizes a Tukey-HSD test is provided below (correction for unbalanced groups):</div>
<p><span class="math display">\[
\text{HSD}_{ij} = q_{\alpha} \cdot \sqrt{\frac{\text{MS}_{\text{within}}}{2} \left( \frac{1}{n_i} + \frac{1}{n_j} \right)}
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>HSD = minimum value of the difference between two means for it to be considered significant;</li>
<li>q<span class="math inline">\(_ {\alpha}\)</span> = critical value of the studentized distribution <em>q</em> (Tukey distribution), depending on the number of groups and the significance level <span class="math inline">\(\alpha\)</span>;</li>
<li>MS<span class="math inline">\(_{within}\)</span> = mean square within groups (error) obtained from ANAVA;</li>
<li>n: number of observations per group (<em>i</em> and <em>j</em> if there are groups of different sizes).<br>
</li>
</ul>
<p><a href="Eq/jsp_tukey.html" target="_blank"><img src="Eq/tukey.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-26" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-26">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> You can click and drag the minimum difference <span class="fu">markers</span> (a, b, etc.) above the graph <span class="cf">for</span> better visual adjustment, as well as the ANOVA table and the statistical test result;</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Change the vector values, redo the simultaneous analysis and graph, and notice the difference <span class="cf">in</span> the values obtained. Note that, as stated at the bottom, you must refresh the website page to clear previous data retained <span class="cf">in</span> the cache;</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Enter other data <span class="cf">in</span> the respective vectors, either your own or from another source;</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Enter new data vectors, or delete some, <span class="cf">for</span> a customized calculation and graph.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="nonlinear-data-adjustment" class="level2" data-number="28">
<h2 data-number="28" class="anchored" data-anchor-id="nonlinear-data-adjustment"><span class="header-section-number">28</span> Nonlinear data adjustment</h2>
<div class="line-block">Modeling is a common area of employment in the natural sciences when mathematical prediction or correlation of experimental data is desired. Models can be linear or nonlinear. When nonlinear, they can be applied empirically to the data (e.g., quadratic, logarithmic, exponential, or power functions), or from a mathematical function that explains the model.</div>
<div class="line-block">In this case, techniques for nonlinear adjustment of the data to the model are used, such as <em>Simplex</em>, <em>Gauss</em>, <em>Newton-Raphson</em>, <em>Levenberg-Marquardt</em>, or <em>least squares</em>, for example. Whatever the method, nonlinear adjustment differs from linear adjustment in its assumptions and treatment of data (see a quick explanation <a href="https://bioquanti.netlify.app/pt/nivel/superior/r/enzimas#ajuste-n%C3%A3o-linear">here</a>.</div>
<section id="context-ligand-protein-interaction-grid-search-algorithm" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-ligand-protein-interaction-grid-search-algorithm">Context: Ligand-protein interaction (<em>grid search</em> algorithm)</h3>
<div class="line-block">The following example illustrates the application of a nonlinear adjustment of molecular interaction data using the Langmuir model and treatment by <em>quadratic error minimization by manual scanning</em>. In this treatment, the <em>sum of quadratic errors (RMSE)</em> is calculated for combinations of <em>a</em> and <em>b</em>, selecting the pair with the lowest calculated error. In this case, for simplicity, although <em>JSPlotly</em> works with more specific libraries, the nonlinear regression is performed by least squares, but carried out by <em>grid search</em>, and not by classical iterative methods. Thus, a set of values is tested for the pair (a, b) in a previously defined grid (table of combinations), allowing the identification of the parameters that best describe the experimental data according to the minimum error criterion.</div>
<p><br>
</p>
</section>
<section id="equation-18" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-18">Equation:</h3>
<div class="line-block">The Langmuir equation is represented below:</div>
<p><span class="math display">\[
Y = \frac{aX}{K_d + X}
\]</span> <em>Where:</em></p>
<ul>
<li>Y = fraction of occupied sites;</li>
<li>X = concentration of free ligand;</li>
<li><em>a</em> = saturation value (sites of the same affinity completely occupied);</li>
<li><em>Kd</em> = equilibrium dissociation constant of the complex.</li>
</ul>
<div class="line-block">In this case, the nonlinear least squares function is:</div>
<p><span class="math display">\[
SSR(a, K_d) = \sum_{i=1}^n \left( y_i - \frac{a x_i}{K_d + x_i} \right)^2
\]</span></p>
<p><em>Where:</em></p>
<p><em>SSR</em> = sum of the squares of the residuals (inversely proportional to the quality of the fit)<br>
</p>
<p><a href="Eq/jsp_ajNonLin.html" target="_blank"><img src="Eq/ajNonLin.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-27" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-27">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> As <span class="cf">in</span> the previous example, alternately display the points and the adjustment curve using the <span class="st">"Boolean"</span> <span class="fu">operation</span> (<span class="st">"showPoints = true/false"</span>; <span class="st">"showCurve = true/false"</span>);</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Change the data and perform a new fit to obtain other parameters of the <span class="fu">equation</span> (Kd, SSR, R², adjusted R²). Note<span class="sc">:</span> Adjusted R² corresponds to the value of R² corrected <span class="cf">for</span> the number of model <span class="fu">parameters</span> (<span class="cf">in</span> this case, <span class="st">"2"</span>), directly affecting the <span class="st">"degrees of freedom"</span> <span class="cf">for</span> the fit.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
<section id="context---competitive-inhibition-in-enzyme-kinetics-gauss-newton-algorithm" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context---competitive-inhibition-in-enzyme-kinetics-gauss-newton-algorithm">Context - Competitive inhibition in enzyme kinetics (Gauss-Newton algorithm)</h3>
<div class="line-block">The <em>Gauss-Newton</em> method iteratively adjusts nonlinear models to experimental data sets. It is especially efficient when the model is <em>quasi-linear</em> with respect to the parameters and the residuals are moderately small. The goal is to find the parameters <span class="math inline">\(\theta\)</span> that minimize the sum of the squares of the residuals between the observed data <em>yi</em> and the values predicted by the model f(xi,<span class="math inline">\(\theta\)</span>. Unlike the full Newton-Raphson method, Gauss-Newton avoids the explicit calculation of the Hessian by using an approximation based on the Jacobian of the residuals.</div>
</section>
</section>
<section id="equation-19" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="equation-19">Equation:</h2>
<div class="line-block">This approximation results in a linear system that can be solved at each iteration to update the model parameters. The central formula of the method consists of multiplying the transposed Jacobian <em>JT</em> by the Jacobian <em>J</em> itself, forming a symmetric matrix that approximates the Hessian. The residual vector <em>r</em> is then used to determine the update direction <span class="math inline">\(\Delta\theta\)</span>, with the following equation:</div>
<p><span class="math display">\[
\theta_{k+1} = \theta_k + \left(J^T J\right)^{-1} J^T \cdot r
\]</span></p>
<div class="line-block">To illustrate the potential of the application in iterative nonlinear fitting, here is an example for enzyme kinetics using a <em>competitive inhibition model</em>.</div>
<div class="line-block">The code allows you to adjust the <em>seeds</em>, <em>tolerance</em>, and number of <em>iterations</em>, and provides the graph (points/adjusted line, manually), and statistical parameters of the regression.</div>
<p><br>
</p>
<p><strong>Instructions</strong></p>
<ul>
<li><ol type="1">
<li>Note that the code has a flag for <em>true/false</em> in the <em>fit</em> constant. Leave it at <em>false</em> for data representation only (<em>x</em> and <em>y</em> vectors, or <em>S</em> and <em>v</em>);</li>
</ol></li>
<li><ol start="2" type="1">
<li>In the <em>Plot</em> area of the code, set the <em>trace</em> constant so that it also includes the data vectors;</li>
</ol></li>
<li><ol start="3" type="1">
<li>Click <em>add plot</em> to view the experimental points;</li>
</ol></li>
<li><ol start="4" type="1">
<li>To superimpose the fitted curve, update the constants <em>fit</em> and <em>trace</em> (vectors <em>S_fit</em> and <em>v_fit</em>), and choose <em>lines</em> in <em>mode</em>.</li>
</ol></li>
</ul>
<p><a href="Eq/jsp_ajNonLinGauss3.html" target="_blank"><img src="Eq/ajNonLinGauss2.png" class="img-fluid"></a><br>
</p>
</section>
<section id="application---digital-whiteboard" class="level2" data-number="29">
<h2 data-number="29" class="anchored" data-anchor-id="application---digital-whiteboard"><span class="header-section-number">29</span> Application - Digital Whiteboard</h2>
<div class="line-block">The nature of the object below needs no introduction. It is yet another of the many applications for simulating a whiteboard or digital chalkboard/pen. However, it differs from its competitors in the following ways:</div>
<ol type="1">
<li>It can be exported as a self-sufficient HTML file using a button on the whiteboard itself, allowing the generated image to be shared and edited further.</li>
<li>It allows customization of the code that produces it, allowing you to insert/change pens, thicknesses, and different actions not provided for in the source code.<br>
</li>
</ol>
<p><a href="Eq/lousa.html" target="_blank"><img src="Eq/lousa.png" class="img-fluid"></a><br>
</p>
<div class="line-block">In addition to its features, the digital whiteboard allows you to:</div>
<ul>
<li>Write with 7 colors and 3 different thicknesses;</li>
<li>Access 4 different colored backgrounds;</li>
<li>Overlay the plane with a grid;</li>
<li>Use an eraser;</li>
<li>Use a laser pointer to locate what you want to highlight;</li>
<li>Save the image as a PNG, indicating the date and time;</li>
<li>Undo and clean commands for the board;</li>
<li>Keyboard shortcuts: E=eraser; L=laser; Z undo, X clean, H save HTML</li>
</ul>
<section id="suggestion-28" class="level3" data-number="29.1">
<h3 data-number="29.1" class="anchored" data-anchor-id="suggestion-28"><span class="header-section-number">29.1</span> Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> You can change the thickness of the pens <span class="cf">in</span> <span class="st">"var baseWidth = 1;"</span> ;</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> The <span class="dv">1</span>x, <span class="dv">2</span>x, <span class="dv">4</span>x buttons are multipliers; edit the button creation lines to change the label<span class="sc">/</span><span class="fu">multiplier</span> (e.g., <span class="st">"addSize(‘3x’, 3, true)"</span>).</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>
</section>

<a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../../../nivel/superior/jsplotly/gsplotly_en.html" class="pagination-link" aria-label="GSPlotly - AI code generator for JSPlotly">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">GSPlotly - AI code generator for JSPlotly</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../../nivel/basico/jsplotly/jsplotly_bas_en.html" class="pagination-link" aria-label="Simulations at School">
        <span class="nav-page-text">Simulations at School</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>